<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2022-03-20" />

<title>TF2KEGG_network</title>

<script src="site_libs/header-attrs-2.13/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>



<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div id="header">



<h1 class="title toc-ignore">TF2KEGG_network</h1>
<h4 class="date">2022-03-20</h4>

</div>


<p>#Set libaries</p>
<pre class="r"><code>library(ngsReports)
library(tidyverse)
library(magrittr)
library(edgeR)
library(AnnotationHub)
library(ensembldb)
library(scales)
library(pander)
library(cqn)
library(ggrepel)
library(pheatmap)
library(RColorBrewer)
library(corrplot)
library(UpSetR)
library(cowplot)
library(msigdbr)
library(GO.db)
library(goseq)
library(kableExtra)
library(plotly)
library(reshape2)
library(DT)
library(ComplexHeatmap)
library(viridis)
library(igraph)
library(ggraph)</code></pre>
<pre class="r"><code>theme_set(theme_bw())
panderOptions(&quot;table.split.table&quot;, Inf)
panderOptions(&quot;table.style&quot;, &quot;rmarkdown&quot;)
panderOptions(&quot;big.mark&quot;, &quot;,&quot;)</code></pre>
<pre class="r"><code>ah &lt;- AnnotationHub() %&gt;%
    subset(species == &quot;Homo sapiens&quot;) %&gt;%
    subset(rdataclass == &quot;EnsDb&quot;)
ensDB &lt;- ah[[&quot;AH75011&quot;]]
grTrans &lt;- transcripts(ensDB)
trLengths &lt;- exonsBy(ensDB, &quot;tx&quot;) %&gt;%
    width() %&gt;%
    vapply(sum, integer(1))
mcols(grTrans)$length &lt;- trLengths[names(grTrans)]
gcGene &lt;- grTrans %&gt;%
  mcols() %&gt;%
  as.data.frame() %&gt;%
  dplyr::select(gene_id, tx_id, gc_content, length) %&gt;%
  as_tibble() %&gt;%
  group_by(gene_id) %&gt;%
  summarise(
    gc_content = sum(gc_content*length) / sum(length),
    length = ceiling(median(length))
  )
grGenes &lt;- genes(ensDB)
mcols(grGenes) %&lt;&gt;%
  as.data.frame() %&gt;%
  left_join(gcGene) %&gt;%
  as.data.frame() %&gt;%
  DataFrame()</code></pre>
<p>Raw read counts were read in and an overall <code>DGElist</code>
object was created.</p>
<pre class="r"><code>filename &lt;- here::here(&quot;data&quot;, &quot;filename.txt&quot;) %&gt;%
  read.table(header = TRUE, sep = &quot;\t&quot;,stringsAsFactors = FALSE) %&gt;% 
  as_tibble()%&gt;%
  mutate_at(vars(one_of(c(&quot;CELL&quot;, &quot;TREAT&quot;,&quot;CONTR&quot;, &quot;TIME&quot;))), as.factor) %&gt;%
  mutate(TREATED = ifelse(CONTR == 0,1,0))
dgeList &lt;- here::here(&quot;data&quot;, &quot;2_alignedData&quot;, &quot;featureCounts&quot;, &quot;genes.out&quot;) %&gt;%
  read_delim(delim = &quot;\t&quot;) %&gt;%
  set_names(basename(names(.))) %&gt;%
  as.data.frame() %&gt;%
  column_to_rownames(&quot;Geneid&quot;) %&gt;%
  as.matrix() %&gt;% 
  set_colnames(str_remove(colnames(.), &quot;Aligned.sortedByCoord.out.bam&quot;)) %&gt;%
  DGEList(
    samples = tibble(sample = colnames(.)) %&gt;%
      left_join(dplyr::filter(filename, read == &quot;R1&quot;)),
    genes = grGenes[rownames(.)] %&gt;%
      as.data.frame() %&gt;%
      dplyr::select(
        chromosome = seqnames, start, end, 
        gene_id, gene_name, gene_biotype, description, 
        entrezid, gc_content, length
      )
  ) %&gt;%
  .[!grepl(&quot;rRNA&quot;, .$genes$gene_biotype),] %&gt;%
  calcNormFactors()</code></pre>
<div id="retrieve-tft-gene-sets" class="section level1">
<h1>Retrieve TFT gene-sets</h1>
<pre class="r"><code>entrezGenes &lt;- dgeList$genes %&gt;%
  unnest(entrezid) %&gt;%
  dplyr::rename(entrez_gene = entrezid)
# save(entrezGenes, file = here::here(&quot;output/Simplified/entrezGenes.Rda&quot;))
tf &lt;- msigdbr(&quot;Homo sapiens&quot;, category = &quot;C3&quot;) %&gt;%
  dplyr::filter(gs_subcat %in% c(&quot;TFT:GTRD&quot;, &quot;TFT:TFT_Legacy&quot;)) %&gt;%
  dplyr::rename(gene_id = ensembl_gene) %&gt;%
  dplyr::filter(!is.na(gene_id)) %&gt;%
  distinct(gs_name, gene_id, .keep_all = TRUE)
tfByGene &lt;- tf  %&gt;%
  split(f = .$gene_id) %&gt;%
  lapply(extract2, &quot;gs_name&quot;)
tfByID &lt;- tf  %&gt;%
  split(f = .$gs_name) %&gt;%
  lapply(extract2, &quot;gene_symbol&quot;)</code></pre>
</div>
<div id="tf-kegg-network" class="section level1">
<h1>TF-KEGG network</h1>
<p>Knowing what TFT gene-sets and KEGG pathways were enriched among the
data-set. We then hypothesized that the enriched TFs were the driver of
gene signatures detected while the significantly regulated KEGG were the
reflection of gene signatures on pathway level.</p>
<p>We wanted to infer a regulatory network between the TFs and KEGG,
which will allow us to integrate the two pieces of information.</p>
<p>To achieve that, KEGG pathways perturbed due to the deferentially
expressed target genes of enriched TFs were tested through modified
SPIA.</p>
<pre class="r"><code>load(here::here(&quot;output/goseqTF_DE.Rda&quot;))
goseqTF_DE_sig &lt;- goseqTF_DE %&gt;%
  lapply(dplyr::filter, FDR &lt; 0.05) %&gt;%
  lapply(pull, gs_name)
goseq_all &lt;- goseqTF_DE_sig %&gt;%
  Reduce(intersect,.)
goseq_SY5Y &lt;- goseqTF_DE_sig[str_subset(names(goseqTF_DE_sig), &quot;SH&quot;)] %&gt;%
  Reduce(intersect,.) %&gt;%
  setdiff(., goseqTF_DE_sig[str_subset(names(goseqTF_DE_sig), &quot;SH&quot;, negate = TRUE)] %&gt;%
            unname() %&gt;%
            unlist())
goseq_EBSS &lt;- goseqTF_DE_sig[str_subset(names(goseqTF_DE_sig), &quot;Starvation&quot;)] %&gt;%
  Reduce(intersect,.) %&gt;%
  setdiff(., goseqTF_DE_sig[str_subset(names(goseqTF_DE_sig), &quot;Starvation&quot;, negate = TRUE)] %&gt;%
            unname() %&gt;%
            unlist())</code></pre>
<div id="consistent-tfs" class="section level2">
<h2>Consistent TFs</h2>
<p>30 TFs were consistently enriched under all conditions with an FDR
&lt; 0.05. DEGs that were also target genes of those TFs were extracted
and used as input of SPIA. Observed PCs were derived locally while the
permutation was performed on phoenix</p>
<pre class="r"><code>DE_genes &lt;- readRDS(here::here(&quot;output/DE_genes.rds&quot;))
names(DE_genes) &lt;- c(&quot;HEK293_AZD8055&quot;,&quot;HEK293_Starvation&quot;,&quot;SH-SY5Y_AZD8055&quot;, &quot;SH-SY5Y_Starvation&quot;,&quot;HeLa_AZD8055&quot;,&quot;HeLa_Starvation&quot;)
consTF_DE &lt;- sapply(goseq_all, function(y){
  sapply(names(DE_genes), function(x){
    DE_genes[[x]] %&gt;%
      dplyr::filter(gene_name %in% tfByID[[y]]) %&gt;%
      left_join(grGenes %&gt;%
                  as.data.frame() %&gt;%
                  distinct(gene_id, .keep_all = TRUE) %&gt;%
                  dplyr::select(gene_id, gene_name))
    
  }, simplify = FALSE)
}, simplify = FALSE)</code></pre>
<pre class="r"><code># load(&quot;/Users/wenjunliu/Induced_autophagy/output/BminsI.Rda&quot;)
# consTF_expressed &lt;- sapply(names(consTF_DE), function(y){
#   sapply(names(DE_genes), function(x){
#     DE_genes[[x]] %&gt;%
#       dplyr::filter(gene_name %in% consTF_DE[[y]][[x]]$gene_name) %&gt;%
#       left_join(entrezGenes) %&gt;%
#       drop_na() %&gt;%
#       mutate(entrez_gene = paste(&quot;ENTREZID:&quot;, entrez_gene, sep = &quot;&quot;)) %&gt;%
#       dplyr::select(entrez_gene, aveFC, gene_name) %&gt;%
#       .[!duplicated(.$entrez_gene),]
#   }, simplify = FALSE)
# }, simplify = FALSE)
# consTF_expressed &lt;- sapply(names(consTF_expressed), function(y){
#   sapply(names(consTF_expressed[[y]]), function(x){
#     consTF_expressed[[y]][[x]] %&gt;%
#       extract2(&quot;aveFC&quot;) %&gt;%
#       set_names(consTF_expressed[[y]][[x]]$entrez_gene)
#   }, simplify = FALSE)
# }, simplify = FALSE)
# spia_consTF &lt;- sapply(names(consTF_expressed), function(y){
#   sapply(names(consTF_expressed[[y]]), function(x){
#     spia_my(de = consTF_expressed[[y]][[x]], BminsI = BminsI)
#   }, simplify = FALSE)
# }, simplify = FALSE)
# save(spia_consTF, file = here::here(&quot;output/TF_KEGG/spia_consTF.Rda&quot;))
load(here::here(&quot;output/TF_KEGG/spia_consTF.Rda&quot;))
load(here::here(&quot;output/TF_KEGG/spia_consTF_permut.Rda&quot;))</code></pre>
<pre class="r"><code> spia_consTF_sum &lt;- sapply(names(spia_consTF), function(x){
  sapply(names(spia_consTF_permut[[x]]), function(y){
    spia_consTF_permut[[x]][[y]] %&gt;%
      split(f = .$gs_name) %&gt;%
      lapply(dplyr::mutate, 
             MAD = mad(tA), 
             MEDIAN = median(tA)) %&gt;%
      lapply(dplyr::select,gs_name, MAD, 
             MEDIAN) %&gt;%
      lapply(unique) %&gt;%
      bind_rows() %&gt;%
      dplyr::filter(MAD != 0) %&gt;%
      left_join(spia_consTF[[x]][[y]]) %&gt;%
      mutate(
             Z_mad = (tA - MEDIAN)/MAD) %&gt;% 
      mutate(
             p_mad = 2*pnorm(-abs(Z_mad)),
             bonf_mad = p.adjust(p_mad, method = &quot;bonferroni&quot;),
             FDR_mad = p.adjust(p_mad, method = &quot;fdr&quot;),
             Sig = ifelse(FDR_mad &lt; 0.05, TRUE, FALSE)) %&gt;%
      .[order(.$FDR_mad),] %&gt;%
      mutate(Rank_spia = row_number()) 
  # %&gt;%
  #     dplyr::filter(!Z_mad %in% c(&quot;-Inf&quot;, &quot;Inf&quot;))
  }, simplify = FALSE) 
}, simplify = FALSE)
spia_consTF_sig &lt;- sapply(names(spia_consTF_sum), function(x){
  sapply(names(spia_consTF_sum[[x]]), function(y){
    spia_consTF_sum[[x]][[y]] %&gt;%
      dplyr::filter(Sig) %&gt;%
      mutate(TF = str_split(x, &quot;_&quot;)[[1]][1], 
             condition = y) %&gt;%
      dplyr::select(TF, gs_name, condition, tA)
  }, simplify = FALSE) %&gt;%
    bind_rows()
}, simplify = FALSE) %&gt;%
    bind_rows() %&gt;%
  split(f = .$condition)</code></pre>
</div>
<div id="sy5y-tfs" class="section level2">
<h2>SY5Y TFs</h2>
<p>The same method was applied onto SY5Y-specific TFs, while the only
difference was that DE TFTs of the TFs belonged to E2F family and highly
overlapped were merged into a “E2F_merge” group.</p>
<pre class="r"><code>sy5yTF_DE &lt;- sapply(goseq_SY5Y, function(y){
  sapply(str_subset(names(DE_genes), &quot;SH&quot;), function(x){
    DE_genes[[x]] %&gt;%
      dplyr::filter(gene_name %in% tfByID[[y]]) %&gt;%
      left_join(grGenes %&gt;%
                  as.data.frame() %&gt;%
                  distinct(gene_id, .keep_all = TRUE) %&gt;%
                  dplyr::select(gene_id, gene_name))
    
  }, simplify = FALSE)
}, simplify = FALSE)
notCombine &lt;- c(&quot;HSD17B8_TARGET_GENES&quot;, &quot;MIER1_TARGET_GENES&quot;)
temp_ls &lt;- list()
temp_ls[[&quot;E2F_merge&quot;]] &lt;- sapply(setdiff(goseq_SY5Y, notCombine), function(y){
  sapply(names(sy5yTF_DE[[y]]), function(x){
    sy5yTF_DE[[y]][[x]] %&gt;%
      mutate(condition = x)
  }, simplify = FALSE) %&gt;%
    bind_rows()
}, simplify = FALSE) %&gt;%
    bind_rows() %&gt;%
  split(f =.$condition)
sy5yTF_DE &lt;- c(temp_ls, 
               sy5yTF_DE[notCombine])</code></pre>
<pre class="r"><code># sy5yTF_expressed &lt;- sapply(names(sy5yTF_DE), function(y){
#   sapply(str_subset(names(DE_genes), &quot;SH&quot;), function(x){
#     DE_genes[[x]] %&gt;%
#       dplyr::filter(gene_name %in% sy5yTF_DE[[y]][[x]]$gene_name) %&gt;%
#       left_join(entrezGenes) %&gt;%
#       drop_na() %&gt;%
#       mutate(entrez_gene = paste(&quot;ENTREZID:&quot;, entrez_gene, sep = &quot;&quot;)) %&gt;%
#       dplyr::select(entrez_gene, aveFC, gene_name) %&gt;%
#       .[!duplicated(.$entrez_gene),]
#   }, simplify = FALSE)
# }, simplify = FALSE)
# sy5yTF_expressed &lt;- sapply(names(sy5yTF_expressed), function(y){
#   sapply(names(sy5yTF_expressed[[y]]), function(x){
#     sy5yTF_expressed[[y]][[x]] %&gt;%
#       extract2(&quot;aveFC&quot;) %&gt;%
#       set_names(sy5yTF_expressed[[y]][[x]]$entrez_gene)
#   }, simplify = FALSE)
# }, simplify = FALSE)
# spia_sy5yTF &lt;- sapply(names(sy5yTF_expressed), function(y){
#   sapply(names(sy5yTF_expressed[[y]]), function(x){
#     spia_my(de = sy5yTF_expressed[[y]][[x]], BminsI = BminsI)
#   }, simplify = FALSE)
# }, simplify = FALSE)
# save(spia_sy5yTF, file = here::here(&quot;output/TF_KEGG/spia_sy5yTF.Rda&quot;))
load(here::here(&quot;output/TF_KEGG/spia_sy5yTF.Rda&quot;))
load(here::here(&quot;output/TF_KEGG/spia_sy5yTF_permut.Rda&quot;))</code></pre>
<pre class="r"><code>spia_sy5yTF_sum &lt;- sapply(names(spia_sy5yTF), function(x){
  sapply(names(spia_sy5yTF_permut[[x]]), function(y){
    spia_sy5yTF_permut[[x]][[y]] %&gt;%
      split(f = .$gs_name) %&gt;%
      lapply(dplyr::mutate, 
             MAD = mad(tA), 
             MEDIAN = median(tA)) %&gt;%
      lapply(dplyr::select,gs_name, MAD, 
             MEDIAN) %&gt;%
      lapply(unique) %&gt;%
      bind_rows() %&gt;%
      dplyr::filter(MAD != 0) %&gt;%
      left_join(spia_sy5yTF[[x]][[y]]) %&gt;%
      mutate(
        Z_mad = (tA - MEDIAN)/MAD) %&gt;% 
      mutate(
        p_mad = 2*pnorm(-abs(Z_mad)),
        bonf_mad = p.adjust(p_mad, method = &quot;bonferroni&quot;),
        FDR_mad = p.adjust(p_mad, method = &quot;fdr&quot;),
        Sig = ifelse(FDR_mad &lt; 0.05, TRUE, FALSE)) %&gt;%
      .[order(.$FDR_mad),] %&gt;%
      mutate(Rank_spia = row_number()) 
    # %&gt;%
    #     dplyr::filter(!Z_mad %in% c(&quot;-Inf&quot;, &quot;Inf&quot;))
  }, simplify = FALSE) 
}, simplify = FALSE)
spia_sy5yTF_sig &lt;- sapply(names(spia_sy5yTF_sum), function(x){
  sapply(names(spia_sy5yTF_sum[[x]]), function(y){
    spia_sy5yTF_sum[[x]][[y]] %&gt;%
      dplyr::filter(Sig) %&gt;%
      mutate(TF = str_split(x, &quot;_&quot;)[[1]][1], 
             condition = y) %&gt;%
      dplyr::select(TF, gs_name, condition, tA)
  }, simplify = FALSE) %&gt;%
    bind_rows()
}, simplify = FALSE) %&gt;%
    bind_rows() %&gt;%
  split(f = .$condition)</code></pre>
</div>
<div id="summary" class="section level2">
<h2>Summary</h2>
<p>Significantly perturbed KEGG pathways and TFs regulating them were
combined into regulatory pairs and intersections between pairs defined
under each condition was visualized.</p>
<pre class="r"><code>spia_sum &lt;- c(spia_consTF_sum %&gt;%
                lapply(function(x){lapply(x, mutate, category = &quot;Consistent&quot;)}), 
              spia_sy5yTF_sum %&gt;%
                lapply(function(x){lapply(x, mutate, category = &quot;SY5Y-specific&quot;)}))
TF_KEGG &lt;- sapply(names(spia_sum), function(x){
  sapply(names(spia_sum[[x]]), function(y){
    spia_sum[[x]][[y]] %&gt;%
      # dplyr::filter(gs_name %in%  spia_net_sig[[y]]) %&gt;%
      mutate(TF = str_split(x, &quot;_&quot;)[[1]][1], 
             condition = y, 
             Regulation = ifelse(Z_mad &lt; 0, &quot;Inhibition&quot;, &quot;Activation&quot;)) %&gt;%
      dplyr::select(TF, gs_name, condition, Z_mad, Sig,p_mad, FDR_mad)
  }, simplify = FALSE) %&gt;%
    bind_rows()
}, simplify = FALSE) %&gt;%
    bind_rows() %&gt;%
  split(f = .$condition)
TF_KEGG %&gt;%
  lapply(dplyr::filter, Sig) %&gt;%
  lapply(mutate, pair = paste(TF, gs_name, sep = &quot;-&quot;)) %&gt;%
  lapply(pull, pair) %&gt;%
  fromList() %&gt;%
  upset(sets = colnames(.), 
        nintersects = NA, 
        keep.order = TRUE, 
         queries = list(
        list(query = intersects,
             params = list(&quot;HEK293_AZD8055&quot;,&quot;HEK293_Starvation&quot;, &quot;SH-SY5Y_AZD8055&quot;, &quot;SH-SY5Y_Starvation&quot;, &quot;HeLa_AZD8055&quot;,&quot;HeLa_Starvation&quot;),
             color = &quot;red&quot;,
             active = T),
        list(query = intersects,
             params = list(&quot;SH-SY5Y_AZD8055&quot;, &quot;SH-SY5Y_Starvation&quot;),
             color = &quot;blue&quot;,
             active = T)
        )
        )</code></pre>
<div class="figure">
<img src="-TF2KEGG_network_files/figure-html/spia_sum-1.png" alt="*Intersection between TF-KEGG regulatory pairs that were significant in each condition: (red) all conditions, (blue) exclusively in SH-SY5Y cell lines*" width="672" />
<p class="caption">
<em>Intersection between TF-KEGG regulatory pairs that were significant
in each condition: (red) all conditions, (blue) exclusively in SH-SY5Y
cell lines</em>
</p>
</div>
<p>10 pairs of regulatory relationships were significant under all
conditions.</p>
<pre class="r"><code>tfKEGG10 &lt;- TF_KEGG %&gt;%
  lapply(dplyr::filter, Sig) %&gt;%
  lapply(mutate, pair = paste(TF, gs_name, sep = &quot;-&quot;)) %&gt;%
  lapply(pull, pair) %&gt;%
  Reduce(intersect,.)</code></pre>
<pre class="r"><code>temp &lt;- TF_KEGG %&gt;%
  lapply(mutate, pair = paste(TF, gs_name, sep = &quot;-&quot;)) %&gt;%
  lapply(dplyr::filter, pair %in% tfKEGG10) %&gt;%
  bind_rows() %&gt;%
  mutate(
    Z_mad = round(Z_mad, 2),
    Z_label = ifelse(Sig, paste(Z_mad, &quot;*&quot;, sep = &quot;&quot;), Z_mad),
    neg_extreme = ifelse(Z_mad &lt; -10, TRUE, FALSE), 
    pos_extreme = ifelse(Z_mad &gt; 10, TRUE, FALSE), 
    Z_label = ifelse(pos_extreme, &quot;&gt;10&quot;, Z_label),
    )
temp %&gt;%
  ggplot(aes(condition, pair, fill = Z_mad)) +
  geom_tile(data = .%&gt;%
              dplyr::filter(!neg_extreme, !pos_extreme),
            aes(fill = Z_mad), fontsize =3) +
  scale_fill_distiller(palette = &quot;RdYlBu&quot;, 
                      limit = c(-10, 10),
                       name = &quot;Normalised Perturbation Score&quot;
                       ) +
  geom_tile(data = .%&gt;%
              dplyr::filter(pos_extreme),
            fill = &quot;firebrick&quot;, fontsize =3) +
  geom_text(aes(label = Z_label)) +
  theme(
    axis.title=element_blank(),
    panel.grid = element_blank(),
    # axis.ticks = element_blank(),
    legend.title = element_text(size= 8),
    legend.key.size = unit(3, &#39;mm&#39;),
    axis.text = element_text(size = 13)
  ) +
  ggtitle(&quot;TF-KEGG regulatory pairs conserverd across all conditions&quot;)</code></pre>
<div class="figure">
<img src="-TF2KEGG_network_files/figure-html/tfKEGG10_hp-1.png" alt="TF-KEGG regulatory pairs that were significant in all conditions*" width="1152" />
<p class="caption">
TF-KEGG regulatory pairs that were significant in all conditions*
</p>
</div>
<pre class="r"><code>temp &lt;- TF_KEGG %&gt;%
  lapply(mutate, pair = paste(TF, gs_name, sep = &quot;-&quot;)) %&gt;%
  lapply(dplyr::filter, pair %in% tfKEGG10) %&gt;%
  bind_rows() %&gt;%
  dplyr::select(TF, gs_name, Z_mad) %&gt;%
  group_by(TF, gs_name) %&gt;%
  summarise(Mean_mad = mean(Z_mad)) %&gt;%
  mutate(color = ifelse(Mean_mad &gt; 0 ,&quot;firebrick&quot;, &quot;midnightblue&quot;),
         type = TF, 
         gs_name = str_replace_all(gs_name, &quot; &quot;, &quot;\n&quot;) %&gt;%
           str_to_title(), 
         gs_name = ifelse(gs_name == &quot;Pi3k-Akt\nSignaling\nPathway&quot;, &quot;PI3K-AKT\nSignaling\nPathway&quot;, gs_name)) %&gt;%
  dplyr::rename(from = TF, 
                to = gs_name) 
g &lt;- temp %&gt;%
  .[,c(&quot;from&quot;, &quot;to&quot;, &quot;color&quot;, &quot;Mean_mad&quot;)] %&gt;%
  graph.data.frame()
TFtoGS &lt;- temp %&gt;%
  #dplyr::select(gs_name, gene_name) %&gt;%
    split(f = .$from)
n &lt;- length(TFtoGS)
totV &lt;- length(V(g))
V(g)$type &lt;- &quot;kegg&quot;
V(g)$type[1:n] &lt;- &quot;TF&quot;
 g %&gt;%
  ggraph(layout = &quot;lgl&quot;) +
  geom_edge_arc(alpha = 0.9,
                color = &quot;tomato3&quot;, 
                show.legend = FALSE,
                strength = 0.4, 
                linemitre = 5, 
                width = 2.3,
                arrow = arrow(type = &quot;closed&quot;, length = unit(3, &#39;mm&#39;))
                ) +
  geom_node_point(
    fill = &quot;lightyellow1&quot;,
    data = . %&gt;% dplyr::filter(type == &quot;TF&quot;),
    size = 5,
    shape = 21,
    repel = TRUE,
    stroke = 0.5,
    show.legend = FALSE
  ) + 
  geom_node_label(
    aes_(label = ~name
         ),
    fill = &quot;lightyellow1&quot;,
    data = . %&gt;% dplyr::filter(type == &quot;TF&quot;),
    repel = TRUE,
    size = 7,
    force = 0.2,
    label.padding = unit(0.8, &quot;lines&quot;), 
    label.size = 0,
    label.r = 0.3,
    show.legend = FALSE
  ) +
  geom_node_point(
    fill = &quot;lightblue&quot;,
    data = . %&gt;% dplyr::filter(type == &quot;kegg&quot;),
    shape = 21,
    size = 8,
    stroke = 0.5,
    repel = TRUE,
    show.legend = FALSE,
    alpha = 0.7, 
  ) +
  geom_node_text(
    aes(label = name),
    fill = &quot;lightblue&quot;,
    size = 6,
    data = . %&gt;% dplyr::filter(type == &quot;kegg&quot;),
    repel = TRUE
  ) +
  theme(panel.border = element_blank(), 
         panel.background = element_rect(fill = &quot;transparent&quot;), # bg of the panel
  plot.background = element_rect(fill = &quot;transparent&quot;, color = NA)
  )</code></pre>
<div class="figure">
<img src="-TF2KEGG_network_files/figure-html/unnamed-chunk-1-1.png" alt="Network plot of TF-KEGG regulatory pairs that were significant in all conditions*" width="672" />
<p class="caption">
Network plot of TF-KEGG regulatory pairs that were significant in all
conditions*
</p>
</div>
<p>From the upset plot, it’s found that 32 regulatory relationship were
exclusive to SY5Y under both treatments. Those were extracted.</p>
<pre class="r"><code>pairToExtract &lt;- TF_KEGG[str_subset(names(TF_KEGG), &quot;SH&quot;)] %&gt;%
  lapply(dplyr::filter, Sig) %&gt;%
  lapply(dplyr::select, TF, gs_name) %&gt;%
  Reduce(inner_join,.) %&gt;%
  mutate(pair = paste(TF, gs_name, sep = &quot;-&quot;)) %&gt;%
  pull(pair) %&gt;%
  setdiff(.,TF_KEGG[str_subset(names(TF_KEGG), &quot;SH&quot;, negate = TRUE)] %&gt;%
            lapply(dplyr::filter, Sig) %&gt;%
            lapply(dplyr::select, TF, gs_name) %&gt;%
            Reduce(full_join,.) %&gt;%
            mutate(pair = paste(TF, gs_name, sep = &quot;-&quot;)) %&gt;%
            pull(pair)
  )  
temp2 &lt;- TF_KEGG[str_subset(names(TF_KEGG), &quot;SH&quot;)] %&gt;%
  lapply(mutate, pair = paste(TF, gs_name, sep = &quot;-&quot;)) %&gt;%
  lapply(dplyr::filter, pair %in% pairToExtract) %&gt;%
  bind_rows() %&gt;%
  mutate(
    Z_mad = round(Z_mad, 2),
    Z_label = ifelse(Sig, paste(Z_mad, &quot;*&quot;, sep = &quot;&quot;), Z_mad),
    neg_extreme = ifelse(Z_mad &lt; -16, TRUE, FALSE), 
    Z_label = ifelse(neg_extreme, &quot;&lt;16&quot;, Z_label),
    )
temp2 %&gt;%
  bind_rows() %&gt;%
  ggplot(aes(condition, pair, fill = Z_mad)) +
  geom_tile(data = . %&gt;% dplyr::filter(!neg_extreme),
            aes(fill = Z_mad), fontsize =3) +
  scale_fill_distiller(palette = &quot;RdYlBu&quot;, 
                      limit = c(-15, 15),
                       name = &quot;Normalised Perturbation Score&quot;
                       ) +
  geom_tile(data = .%&gt;%
              dplyr::filter(neg_extreme),
            fill = &quot;royalblue3&quot;, fontsize =3) +
  geom_text(aes(label = Z_label)) +
  theme(
    axis.title=element_blank(),
    panel.grid = element_blank(),
    # axis.ticks = element_blank(),
    legend.title = element_text(size= 8),
    legend.key.size = unit(3, &#39;mm&#39;),
    #   axis.text.y = element_text(colour = col)
  )  +
  ggtitle(&quot;TF-KEGG regulatory pairs unique to SY5Y cell line&quot;)</code></pre>
<p><img src="-TF2KEGG_network_files/figure-html/tf-keggSH-1.png" width="1152" /></p>
</div>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
