<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Wenjun Liu" />


<title>5_SHEcell</title>

<script src="site_libs/header-attrs-2.13/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.21/datatables.js"></script>
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script>
<link href="site_libs/nouislider-7.0.10/jquery.nouislider.min.css" rel="stylesheet" />
<script src="site_libs/nouislider-7.0.10/jquery.nouislider.min.js"></script>
<link href="site_libs/selectize-0.12.0/selectize.bootstrap3.css" rel="stylesheet" />
<script src="site_libs/selectize-0.12.0/selectize.min.js"></script>
<link href="site_libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Induced_autophagy</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Quality Check
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="HeLa_QC.html">HeLa</a>
    </li>
    <li>
      <a href="SH-SY5Y.html">SH-SY5Y</a>
    </li>
    <li>
      <a href="HEK293.html">HEK293</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    DE
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="DE_analysis.html">Analysis</a>
    </li>
    <li>
      <a href="DE_summary">Summary</a>
    </li>
  </ul>
</li>
<li>
  <a href="TFT_enrichment.html">TFT ORA</a>
</li>
<li>
  <a href="Pathway_analysis(KEGG).html">Pathway Analysis (KEGG)</a>
</li>
<li>
  <a href="TF2KEG_network.html">TF-KEGG regulation inference</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">5_SHEcell</h1>
<h4 class="author">Wenjun Liu</h4>
<h4 class="date">14/05/2020</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-03-21
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>Induced_autophagy/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220317code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20220317)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220317code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220317)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomWenjunLiuInducedautophagytree3d30821bbcd1820dd54f77d9953aa474df721c52targetblank3d30821a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/Wenjun-Liu/Induced_autophagy/tree/3d30821bbcd1820dd54f77d9953aa474df721c52" target="_blank">3d30821</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomWenjunLiuInducedautophagytree3d30821bbcd1820dd54f77d9953aa474df721c52targetblank3d30821a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/Wenjun-Liu/Induced_autophagy/tree/3d30821bbcd1820dd54f77d9953aa474df721c52" target="_blank">3d30821</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    -TF2KEGG_network_files/
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    Induced_autophagy.Rproj
    Ignored:    Pathway_analysis-KEGG-_files/

Untracked files:
    Untracked:  -TF-KEGG_network.Rmd
    Untracked:  analysis/ TF2KEGG_network.Rmd
    Untracked:  analysis/DE_summary.Rmd
    Untracked:  analysis/Pathway_analysis(KEGG).Rmd
    Untracked:  analysis/TFT_enrichment.Rmd
    Untracked:  data/2_alignedData/featureCounts/rawCounts.rds
    Untracked:  output/BminsI.Rda
    Untracked:  output/DE_genes.rds
    Untracked:  output/TF_KEGG/
    Untracked:  output/dgeByCon.rds
    Untracked:  output/goseqTF_DE.Rda
    Untracked:  output/goseq_kegg.rds
    Untracked:  output/kgToGene.rds
    Untracked:  output/spia_netPer.Rda
    Untracked:  output/spia_net_permut.Rda
    Untracked:  output/topTables/
    Untracked:  site_libs/

Unstaged changes:
    Modified:   README.md
    Deleted:    analysis/12_metaDEcopy.Rmd
    Modified:   analysis/DE_analysis.Rmd
    Modified:   analysis/_site.yml
    Deleted:    analysis/about.Rmd
    Modified:   analysis/index.Rmd
    Deleted:    data/dgeList.rda

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/SHSY5Y_QC.Rmd</code>) and HTML
(<code>docs/SHSY5Y_QC.html</code>) files. If you’ve configured a remote
Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/Wenjun-Liu/Induced_autophagy/blob/3d30821bbcd1820dd54f77d9953aa474df721c52/analysis/SHSY5Y_QC.Rmd" target="_blank">3d30821</a>
</td>
<td>
Wenjun-Liu
</td>
<td>
2022-03-18
</td>
<td>
half way through tidying up the DE analysis
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>#Set libaries</p>
<pre class="r"><code>library(ngsReports)
library(tidyverse)
library(magrittr)
library(edgeR)
library(AnnotationHub)
library(ensembldb)
library(scales)
library(pander)
library(cqn)
library(ggrepel)
library(pheatmap)
library(RColorBrewer)
library(corrplot)
library(UpSetR)
library(cowplot)
library(msigdbr)
library(GO.db)
library(goseq)
library(kableExtra)
library(plotly)
library(reshape2)
library(DT)</code></pre>
<pre class="r"><code>theme_set(theme_bw())
panderOptions(&quot;table.split.table&quot;, Inf)
panderOptions(&quot;table.style&quot;, &quot;rmarkdown&quot;)
panderOptions(&quot;big.mark&quot;, &quot;,&quot;)</code></pre>
<pre class="r"><code>ah &lt;- AnnotationHub() %&gt;%
    subset(species == &quot;Homo sapiens&quot;) %&gt;%
    subset(rdataclass == &quot;EnsDb&quot;)
ensDB &lt;- ah[[&quot;AH75011&quot;]]
grTrans &lt;- transcripts(ensDB)
trLengths &lt;- exonsBy(ensDB, &quot;tx&quot;) %&gt;%
    width() %&gt;%
    vapply(sum, integer(1))
mcols(grTrans)$length &lt;- trLengths[names(grTrans)]</code></pre>
<pre class="r"><code>gcGene &lt;- grTrans %&gt;%
  mcols() %&gt;%
  as.data.frame() %&gt;%
  dplyr::select(gene_id, tx_id, gc_content, length) %&gt;%
  as_tibble() %&gt;%
  group_by(gene_id) %&gt;%
  summarise(
    gc_content = sum(gc_content*length) / sum(length),
    length = ceiling(median(length))
  )
grGenes &lt;- genes(ensDB)
mcols(grGenes) %&lt;&gt;%
  as.data.frame() %&gt;%
  left_join(gcGene) %&gt;%
  as.data.frame() %&gt;%
  DataFrame()</code></pre>
<pre class="r"><code>geneLabeller &lt;- structure(grGenes$gene_name, names = grGenes$gene_id) %&gt;%
  as_labeller()</code></pre>
<p>Metadata of all SH-SY5Y samples was extracted and could be explored
through the table below:</p>
<pre class="r"><code>metadata &lt;- here::here(&quot;data&quot;, &quot;metaData.txt&quot;) %&gt;%
  read.table(header = TRUE, sep = &quot;\t&quot;,stringsAsFactors = FALSE) %&gt;% 
  as_tibble() %&gt;%
  mutate_at(vars(one_of(c(&quot;CELL&quot;, &quot;TREAT&quot;,&quot;CONTR&quot;, &quot;TIME&quot;))), as.factor)
filename &lt;- here::here(&quot;data&quot;, &quot;filename.txt&quot;) %&gt;%
  read.table(header = TRUE, sep = &quot;\t&quot;,stringsAsFactors = FALSE) %&gt;% 
  as_tibble() %&gt;%
  mutate_at(vars(one_of(c(&quot;CELL&quot;, &quot;TREAT&quot;,&quot;CONTR&quot;, &quot;TIME&quot;))), as.factor)
#subset SH cells
SH_metadata &lt;- metadata %&gt;%
  dplyr::filter(CELL == &quot;SH-SY5Y.tfLC3&quot;)  
treatCols &lt;- SH_metadata$TREAT %&gt;%
  levels() %&gt;%
  length() %&gt;%
  brewer.pal(&quot;Set1&quot;) %&gt;%
  setNames(levels(SH_metadata$TREAT))
SH_metadata %&gt;%
  dplyr::select(
    `Cell line` = CELL, 
    Treatment = TREAT, 
    Time = TIME, 
    sample, 
    RG.ratio
  ) %&gt;%
  datatable(filter = &quot;top&quot;)</code></pre>
<div id="htmlwidget-7f1368cff9d23946c839" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-7f1368cff9d23946c839">{"x":{"filter":"top","vertical":false,"filterHTML":"<tr>\n  <td><\/td>\n  <td data-type=\"disabled\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;AZD8055&quot;,&quot;DMEM&quot;,&quot;DMSO&quot;,&quot;EBSS&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"factor\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"width: 100%; display: none;\">\n      <select multiple=\"multiple\" style=\"width: 100%;\" data-options=\"[&quot;0&quot;,&quot;1&quot;,&quot;15&quot;,&quot;30&quot;]\"><\/select>\n    <\/div>\n  <\/td>\n  <td data-type=\"character\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n  <\/td>\n  <td data-type=\"number\" style=\"vertical-align: top;\">\n    <div class=\"form-group has-feedback\" style=\"margin-bottom: auto;\">\n      <input type=\"search\" placeholder=\"All\" class=\"form-control\" style=\"width: 100%;\"/>\n      <span class=\"glyphicon glyphicon-remove-circle form-control-feedback\"><\/span>\n    <\/div>\n    <div style=\"display: none;position: absolute;width: 200px;opacity: 1\">\n      <div data-min=\"1.52912019826518\" data-max=\"3.5186170212766\" data-scale=\"14\"><\/div>\n      <span style=\"float: left;\"><\/span>\n      <span style=\"float: right;\"><\/span>\n    <\/div>\n  <\/td>\n<\/tr>","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36"],["SH-SY5Y.tfLC3","SH-SY5Y.tfLC3","SH-SY5Y.tfLC3","SH-SY5Y.tfLC3","SH-SY5Y.tfLC3","SH-SY5Y.tfLC3","SH-SY5Y.tfLC3","SH-SY5Y.tfLC3","SH-SY5Y.tfLC3","SH-SY5Y.tfLC3","SH-SY5Y.tfLC3","SH-SY5Y.tfLC3","SH-SY5Y.tfLC3","SH-SY5Y.tfLC3","SH-SY5Y.tfLC3","SH-SY5Y.tfLC3","SH-SY5Y.tfLC3","SH-SY5Y.tfLC3","SH-SY5Y.tfLC3","SH-SY5Y.tfLC3","SH-SY5Y.tfLC3","SH-SY5Y.tfLC3","SH-SY5Y.tfLC3","SH-SY5Y.tfLC3","SH-SY5Y.tfLC3","SH-SY5Y.tfLC3","SH-SY5Y.tfLC3","SH-SY5Y.tfLC3","SH-SY5Y.tfLC3","SH-SY5Y.tfLC3","SH-SY5Y.tfLC3","SH-SY5Y.tfLC3","SH-SY5Y.tfLC3","SH-SY5Y.tfLC3","SH-SY5Y.tfLC3","SH-SY5Y.tfLC3"],["DMSO","DMSO","DMSO","DMSO","DMSO","AZD8055","DMSO","DMSO","DMSO","DMSO","AZD8055","AZD8055","AZD8055","AZD8055","AZD8055","DMEM","DMEM","DMEM","AZD8055","AZD8055","AZD8055","DMEM","DMEM","DMEM","DMEM","EBSS","EBSS","EBSS","EBSS","EBSS","DMEM","DMEM","EBSS","EBSS","EBSS","EBSS"],["0","0","15","30","30","1","0","15","15","30","1","1","15","15","30","0","15","15","15","30","30","0","0","15","30","1","1","15","30","30","30","30","1","15","15","30"],["F01_S6","F02_S14","F03_S22","F04_S30","F05_S38","F06_S46","F07_S54","F08_S62","F09_S70","F10_S78","F11_S86","F12_S94","G01_S7","G02_S15","G03_S23","G04_S31","G05_S39","G06_S47","G07_S55","G08_S63","G09_S71","G10_S79","G11_S87","G12_S95","H01_S8","H02_S16","H03_S24","H04_S32","H05_S40","H06_S48","H07_S56","H08_S64","H09_S72","H10_S80","H11_S88","H12_S96"],[1.62890625,1.59879518072289,1.62515413070284,1.52912019826518,1.54619883040936,1.59558823529412,1.62064825930372,1.65104166666667,1.62767295597484,1.54206730769231,1.60896637608966,1.59610705596107,2.91283292978208,2.87323943661972,3.48924731182796,1.79329004329004,1.81113320079523,1.8116079923882,2.93516209476309,3.5186170212766,3.44854881266491,1.75609756097561,1.77706323687031,1.81656804733728,1.89084895259096,1.87833140208575,1.89651162790698,2.82297551789077,3.10444444444444,3.09006928406467,1.89197530864198,1.89130434782609,1.86613272311213,2.8271375464684,2.84060721062619,3.03964757709251]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Cell line<\/th>\n      <th>Treatment<\/th>\n      <th>Time<\/th>\n      <th>sample<\/th>\n      <th>RG.ratio<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":5},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"orderCellsTop":true}},"evals":[],"jsHooks":[]}</script>
<p>There are in total 36 SH-SY5Y samples. Study design is illustrated
through the figure embedded in the home page.</p>
<p>The same quality control pipeline as developed<a
href="HeLa_QC.html">using HeLa cell line</a> was applied on SH-SY5Y
samples with minor change.</p>
<div id="raw-data" class="section level1">
<h1>Raw Data</h1>
<div id="library-sizes" class="section level2">
<h2>Library Sizes</h2>
<pre class="r"><code>rawFqc &lt;- here::here(&quot;data&quot;, &quot;0_rawData&quot;, &quot;FastQC&quot;) %&gt;%
  list.files( pattern = &quot;zip&quot;, full.names = TRUE) %&gt;%
  FastqcDataList()
labels &lt;- structure(
  filename$sample,
  names = filename$Filename
)
SHSequence &lt;- subset(filename, CELL == &quot;SH-SY5Y.tfLC3&quot;)$Filename</code></pre>
<pre class="r"><code>SH &lt;- grepl(paste(SHSequence, collapse=&quot;|&quot;),  fqName(rawFqc)) 
r1 &lt;- grepl(&quot;R1&quot;, fqName(rawFqc))
SH_r1 &lt;- r1 &amp; SH
SH_r2 &lt;- SH != SH_r1
plotReadTotals(rawFqc[SH_r1], labels = labels[SH_r1])</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/SHSY5Y_QC.Rmd/plotLibSizes-1.png" alt="*Library Sizes for SH Cell Line samples before any processing was undertaken.*" width="672" />
<p class="caption">
<em>Library Sizes for SH Cell Line samples before any processing was
undertaken.</em>
</p>
</div>
<p>Library Sizes for the raw, unprocessed dataset ranged between
<em>88,435,668</em> and <em>148,842,501</em> reads.</p>
</div>
<div id="over-represented-sequences" class="section level2">
<h2>Over-represented Sequences</h2>
<p>The top 30 Overrepresented sequences were analysed using
<code>blastn</code>.</p>
<pre class="r"><code>getModule(rawFqc[SH], &quot;Overrep&quot;) %&gt;% 
    group_by(Sequence, Possible_Source) %&gt;% 
    summarise(`Found In` = n(), `Highest Percentage` = max(Percentage)) %&gt;% 
    arrange(desc(`Highest Percentage`), desc(`Found In`)) %&gt;% 
    ungroup() %&gt;% 
    dplyr::slice(1:30) %&gt;%
    mutate(`Highest Percentage` = percent_format(0.01)(`Highest Percentage`/100)) %&gt;%
    pander(
        justify = &quot;llrr&quot;,
        caption = paste(
            &quot;*Top&quot;, nrow(.),&quot;Overrepresented sequences.&quot;,
            &quot;The number of samples they were found in is shown,&quot;,
            &quot;along with the percentage of the most &#39;contaminated&#39; sample.*&quot;
            )
    )</code></pre>
<table>
<caption><em>Top 30 Overrepresented sequences. The number of samples
they were found in is shown, along with the percentage of the most
‘contaminated’ sample.</em></caption>
<colgroup>
<col width="52%" />
<col width="17%" />
<col width="10%" />
<col width="20%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">Sequence</th>
<th align="left">Possible_Source</th>
<th align="right">Found In</th>
<th align="right">Highest Percentage</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">CTTAGGCAACCTGGTGGTCCCCCGCTCCCGGGAGGTCACCATATTGATGC</td>
<td align="left">No Hit</td>
<td align="right">33</td>
<td align="right">1.30%</td>
</tr>
<tr class="even">
<td align="left">CCTTAGGCAACCTGGTGGTCCCCCGCTCCCGGGAGGTCACCATATTGATG</td>
<td align="left">No Hit</td>
<td align="right">36</td>
<td align="right">1.17%</td>
</tr>
<tr class="odd">
<td align="left">CTGGAGTCTTGGAAGCTTGACTACCCTACGTTCTCCTACAAATGGACCTT</td>
<td align="left">No Hit</td>
<td align="right">36</td>
<td align="right">1.05%</td>
</tr>
<tr class="even">
<td align="left">CTCAGACCGCGTTCTCTCCCTCTCACTCCCCAATACGGAGAGAAGAACGA</td>
<td align="left">No Hit</td>
<td align="right">33</td>
<td align="right">0.86%</td>
</tr>
<tr class="odd">
<td align="left">CCCCGAGCCACCTTCCCCGCCGGGCCTTCCCAGCCGTCCCGGAGCCGGTC</td>
<td align="left">No Hit</td>
<td align="right">11</td>
<td align="right">0.62%</td>
</tr>
<tr class="even">
<td align="left">CTCCGTTTCCGACCTGGGCCGGTTCACCCCTCCTTAGGCAACCTGGTGGT</td>
<td align="left">No Hit</td>
<td align="right">36</td>
<td align="right">0.56%</td>
</tr>
<tr class="odd">
<td align="left">CGTCCCCCGACCGGCGACCGGCCGCCGCCGGGCGCATTTCCACCGCGGCG</td>
<td align="left">No Hit</td>
<td align="right">9</td>
<td align="right">0.53%</td>
</tr>
<tr class="even">
<td align="left">CTGGCTGCGACATCTGTCACCCCATTGATCGCCAGGGTTGATTCGGCTGA</td>
<td align="left">No Hit</td>
<td align="right">36</td>
<td align="right">0.47%</td>
</tr>
<tr class="odd">
<td align="left">CTCATTTGGATGTGTCTGGAGTCTTGGAAGCTTGACTACCCTACGTTCTC</td>
<td align="left">No Hit</td>
<td align="right">36</td>
<td align="right">0.44%</td>
</tr>
<tr class="even">
<td align="left">CTGGAGTGCAGTGGCTATTCACAGGCGCGATCCCACTACTGATCAGCACG</td>
<td align="left">No Hit</td>
<td align="right">18</td>
<td align="right">0.40%</td>
</tr>
<tr class="odd">
<td align="left">GTCTGGAGTCTTGGAAGCTTGACTACCCTACGTTCTCCTACAAATGGACC</td>
<td align="left">No Hit</td>
<td align="right">36</td>
<td align="right">0.39%</td>
</tr>
<tr class="even">
<td align="left">CCCCTCCTTAGGCAACCTGGTGGTCCCCCGCTCCCGGGAGGTCACCATAT</td>
<td align="left">No Hit</td>
<td align="right">18</td>
<td align="right">0.36%</td>
</tr>
<tr class="odd">
<td align="left">CGGTGGCGCGTGCCTGTAGTCCCAGCTACTCGGGAGGCTGAGGCTGGAGG</td>
<td align="left">No Hit</td>
<td align="right">15</td>
<td align="right">0.35%</td>
</tr>
<tr class="even">
<td align="left">GGTCGACCCGTGCGGAGGAGCGAGGAGGAAGGACGCGCGAGGGCCGGGAC</td>
<td align="left">No Hit</td>
<td align="right">1</td>
<td align="right">0.34%</td>
</tr>
<tr class="odd">
<td align="left">CCCGAGCCACCTTCCCCGCCGGGCCTTCCCAGCCGTCCCGGAGCCGGTCG</td>
<td align="left">No Hit</td>
<td align="right">3</td>
<td align="right">0.34%</td>
</tr>
<tr class="even">
<td align="left">CCCTCCTTAGGCAACCTGGTGGTCCCCCGCTCCCGGGAGGTCACCATATT</td>
<td align="left">No Hit</td>
<td align="right">18</td>
<td align="right">0.32%</td>
</tr>
<tr class="odd">
<td align="left">GGGCGATCTGGCTGCGACATCTGTCACCCCATTGATCGCCAGGGTTGATT</td>
<td align="left">No Hit</td>
<td align="right">15</td>
<td align="right">0.32%</td>
</tr>
<tr class="even">
<td align="left">CTCCTTAGGCAACCTGGTGGTCCCCCGCTCCCGGGAGGTCACCATATTGA</td>
<td align="left">No Hit</td>
<td align="right">18</td>
<td align="right">0.31%</td>
</tr>
<tr class="odd">
<td align="left">CTTCGATGTCGGCTCTTCCTATCATTGTGAAGCAGAATTCACCAAGCGTT</td>
<td align="left">No Hit</td>
<td align="right">8</td>
<td align="right">0.31%</td>
</tr>
<tr class="even">
<td align="left">CTTGAGTCCAGGAGTTCTGGGCTGTAGTGCGCTATGCCGATCGGGTGTCC</td>
<td align="left">No Hit</td>
<td align="right">36</td>
<td align="right">0.29%</td>
</tr>
<tr class="odd">
<td align="left">CCTCCTTAGGCAACCTGGTGGTCCCCCGCTCCCGGGAGGTCACCATATTG</td>
<td align="left">No Hit</td>
<td align="right">18</td>
<td align="right">0.27%</td>
</tr>
<tr class="even">
<td align="left">CCAGGCTGGAGTGCAGTGGCTATTCACAGGCGCGATCCCACTACTGATCA</td>
<td align="left">No Hit</td>
<td align="right">33</td>
<td align="right">0.26%</td>
</tr>
<tr class="odd">
<td align="left">CGCGTGCCTGTAGTCCCAGCTACTCGGGAGGCTGAGGCTGGAGGATCGCT</td>
<td align="left">No Hit</td>
<td align="right">32</td>
<td align="right">0.26%</td>
</tr>
<tr class="even">
<td align="left">CGGTGGCGCGTGCCTGTAGTCCCAGCTACTCGGGAGGCTGAGGTGGGAGG</td>
<td align="left">No Hit</td>
<td align="right">15</td>
<td align="right">0.26%</td>
</tr>
<tr class="odd">
<td align="left">CTCGTATACCCTTGACCGAAGACCGGTCCTCCTCTATCGGGGATGGTCGT</td>
<td align="left">No Hit</td>
<td align="right">36</td>
<td align="right">0.25%</td>
</tr>
<tr class="even">
<td align="left">CTGGGCTGTAGTGCGCTATGCCGATCGGGTGTCCGCACTAAGTTCGGCAT</td>
<td align="left">No Hit</td>
<td align="right">36</td>
<td align="right">0.25%</td>
</tr>
<tr class="odd">
<td align="left">CATCTGTCACCCCATTGATCGCCAGGGTTGATTCGGCTGATCTGGCTGGC</td>
<td align="left">No Hit</td>
<td align="right">15</td>
<td align="right">0.24%</td>
</tr>
<tr class="even">
<td align="left">GTTCTGGGCTGTAGTGCGCTATGCCGATCGGGTGTCCGCACTAAGTTCGG</td>
<td align="left">No Hit</td>
<td align="right">26</td>
<td align="right">0.24%</td>
</tr>
<tr class="odd">
<td align="left">CGACCATCCCCGATAGAGGAGGACCGGTCTTCGGTCAAGGGTATACGAGT</td>
<td align="left">No Hit</td>
<td align="right">34</td>
<td align="right">0.23%</td>
</tr>
<tr class="even">
<td align="left">GGGAAGCTCATCAGTGGGGCCACGAGCTGAGTGCGTCCTGTCACTCCACT</td>
<td align="left">No Hit</td>
<td align="right">15</td>
<td align="right">0.23%</td>
</tr>
</tbody>
</table>
<pre class="r"><code>#overRep2Fasta(rawFqc, &quot;top30.fa&quot;)</code></pre>
</div>
<div id="gc-content" class="section level2">
<h2>GC Content</h2>
<p>To determine if rRNA depletion was done succesfully in this total RNA
dataset, GC contents for R1 and R2 read and percentage of each library
containing &gt;70% GC contents were plotted.</p>
<p>Plots for R1 and R2 are almost idential. Both of the graphs show a
large peak and a small spike in GC content at around 65% and 83%
respectively, wich might resulted from incomplete rRNA depletion. Read 1
also showed a small spike at around 50% GC content, which didn’t appear
in other two cell lines.</p>
<pre class="r"><code>gcPlots &lt;- list(
    r1 = plotGcContent(
        x = rawFqc[SH_r1], 
        labels = labels[SH_r1],
        plotType = &quot;line&quot;,
        gcType = &quot;Transcriptome&quot;,
        species = &quot;Hsapiens&quot;
    ),
    r2 = plotGcContent(
        x = rawFqc[SH_r2], 
        labels = labels[SH_r2],
        plotType = &quot;line&quot;,
        gcType = &quot;Transcriptome&quot;,
        species = &quot;Hsapiens&quot;
    )
)
lg &lt;- get_legend(gcPlots$r2 + theme(legend.position = &quot;bottom&quot;)) 
plot_grid(
    plot_grid(
        r1 = gcPlots$r1 + 
            ggtitle(&quot;R1: GC Distribution&quot;, subtitle = c()) +
            theme(legend.position = &quot;none&quot;),
        r2 = gcPlots$r2 + 
            ggtitle(&quot;R2: GC Distribution&quot;, subtitle = c()) +
            theme(legend.position = &quot;none&quot;)
    ),
    lg = lg,
    nrow = 2,
    rel_heights = c(5,2)
)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/SHSY5Y_QC.Rmd/gcPlots-1.png" alt="*GC content for SH Cells R1 and R2 reads" width="672" />
<p class="caption">
*GC content for SH Cells R1 and R2 reads
</p>
</div>
<pre class="r"><code>gc &lt;- getModule(rawFqc[SH], &quot;Per_sequence_GC&quot;) 
rawGC &lt;- gc %&gt;% 
    group_by(Filename) %&gt;% 
    mutate(Freq = Count / sum(Count)) %&gt;%
    dplyr::filter(GC_Content &gt; 70) %&gt;% 
    summarise(Freq = sum(Freq)) %&gt;% 
    arrange(desc(Freq)) %&gt;%
    left_join(filename) 
rawGC %&gt;%
    ggplot(aes(sample, Freq, fill = read)) +
    geom_bar(stat = &quot;identity&quot;, position = &quot;dodge&quot;) +
    scale_y_continuous(labels = percent) +
    facet_wrap(~CELL, scales = &quot;free_x&quot;) +
    labs(x = &quot;Sample&quot;, y = &quot;Percent of Total&quot;)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/SHSY5Y_QC.Rmd/gcPerc-1.png" alt="*Percentages of each library which contain &gt;70% GC. Using the known theoretical distribution, this should be 3.65% of the total library.*" width="768" />
<p class="caption">
<em>Percentages of each library which contain &gt;70% GC. Using the
known theoretical distribution, this should be 3.65% of the total
library.</em>
</p>
</div>
</div>
</div>
<div id="aligned-data" class="section level1">
<h1>Aligned Data</h1>
<p>Trimmed reads were aligned to the human genome assembly GRCh38.p13
from <code>Ensembl Release 98</code> using <code>STAR 2.7.0d</code> and
summarised to each gene using <code>featureCounts</code>. These read
counts are read into a DGEList object for gene-level analysis.</p>
<pre class="r"><code>dgeList &lt;- here::here(&quot;data&quot;, &quot;2_alignedData&quot;, &quot;featureCounts&quot;, &quot;genes.out&quot;) %&gt;%
  read_delim(delim = &quot;\t&quot;) %&gt;%
  set_names(basename(names(.))) %&gt;%
  as.data.frame() %&gt;%
  column_to_rownames(&quot;Geneid&quot;) %&gt;%
  as.matrix() %&gt;% 
  set_colnames(str_remove(colnames(.), &quot;Aligned.sortedByCoord.out.bam&quot;)) %&gt;%
  DGEList(
    samples = tibble(sample = colnames(.)) %&gt;%
      left_join(dplyr::filter(filename, read == &quot;R1&quot;)),
    genes = grGenes[rownames(.)] %&gt;%
      as.data.frame() %&gt;%
      dplyr::select(
        chromosome = seqnames, start, end, 
        gene_id, gene_name, gene_biotype, description, 
        entrezid, gc_content, length
      )
  ) %&gt;%
  .[!grepl(&quot;rRNA&quot;, .$genes$gene_biotype),] %&gt;%
  calcNormFactors()
SH_cell &lt;- grepl(paste(unique(SH_metadata$sample), collapse=&quot;|&quot;),dgeList$samples$sample)  
SH_dgeList &lt;- dgeList[,SH_cell]
minCPM &lt;- 1.5
minSamples &lt;- 3
SH_dgeList %&lt;&gt;% .[rowSums(cpm(.) &gt;= minCPM) &gt;= minSamples,]</code></pre>
<p>At this step, genes were removed if:</p>
<ul>
<li>They were not considered as detectable (CPM &lt; 1.5 in &gt; 33
samples). This translates to &gt; 58 reads assigned a gene in all
samples from one or more of the treatment groups</li>
<li>The <code>gene_biotype</code> was any type of
<code>rRNA</code>.</li>
</ul>
<p>These filtering steps returned gene-level counts for 13,941 genes,
with total library sizes between <em>38,497,645</em> and
<em>84,710,972</em> reads assigned to genes.</p>
<pre class="r"><code>cpm(SH_dgeList, log = TRUE) %&gt;%
  as.data.frame() %&gt;%
  pivot_longer(
    cols = everything(),
    names_to = &quot;sample&quot;,
    values_to = &quot;logCPM&quot;
  ) %&gt;%
  split(f = .$sample) %&gt;%
  lapply(function(x){
    d &lt;- density(x$logCPM)
    tibble(
      sample = unique(x$sample),
      x = d$x,
      y = d$y
    )
  }) %&gt;%
  bind_rows() %&gt;%
  left_join(metadata) %&gt;%
  ggplot(aes(x, y, colour = TREAT, group = sample)) +
  geom_line() +
  scale_colour_manual(
    values = treatCols
  ) +
  labs(
    x = &quot;logCPM&quot;,
    y = &quot;Density&quot;,
    colour = &quot;TREAT&quot;
  )</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/SHSY5Y_QC.Rmd/plotDensities-1.png" alt="*Expression density plots for all samples after filtering, showing logCPM values.*" width="480" />
<p class="caption">
<em>Expression density plots for all samples after filtering, showing
logCPM values.</em>
</p>
</div>
<p>##GC Content</p>
<pre class="r"><code>gcInfo &lt;- SH_dgeList$counts%&gt;%
    as.data.frame() %&gt;%
    rownames_to_column(&quot;gene_id&quot;)%&gt;%
  as_tibble() %&gt;%
    pivot_longer(
      cols = ends_with(as.character(c(0:9))),
        names_to = &quot;sample&quot;,
        values_to = &quot;counts&quot;
    ) %&gt;%
    left_join(
        mcols(grTrans) %&gt;% as.data.frame()
    ) %&gt;%
    dplyr::select(
        gene_id, sample, counts, gc_content, length
    ) %&gt;%
    split(f = .$sample) %&gt;%
    lapply(function(x){
        DataFrame(
            gc = Rle(x$gc_content/100, x$counts),
            logLen = Rle(log10(x$length), x$counts)
        )
    }
    ) 
gcSummary &lt;- gcInfo %&gt;%
    vapply(function(x){
        c(mean(x$gc), sd(x$gc), mean(x$logLen), sd(x$logLen))
        }, numeric(4)
        ) %&gt;%
    t() %&gt;%
    set_colnames(
        c(&quot;mn_gc&quot;, &quot;sd_gc&quot;, &quot;mn_logLen&quot;, &quot;sd_logLen&quot;)
    ) %&gt;%
    as.data.frame() %&gt;%
    rownames_to_column(&quot;sample&quot;) %&gt;%
    as_tibble() %&gt;%
    left_join(dplyr::filter(filename, read == &quot;R1&quot;)) %&gt;%
    dplyr::select(sample, CELL, TREAT, TIME, CONTR, contains(&quot;_&quot;))</code></pre>
<pre class="r"><code>gcDev &lt;- gc %&gt;%
    left_join(filename) %&gt;%
    group_by(sample, Filename) %&gt;%
    mutate(Freq = Count / sum(Count)) %&gt;%
    left_join(
        getGC(gcTheoretical, &quot;Hsapiens&quot;, &quot;Trans&quot;)) %&gt;%
    dplyr::rename(actual = Hsapiens) %&gt;%
    mutate(res = Freq - actual) %&gt;%
    summarise(ss = sum(res^2), n = n()) %&gt;%
    ungroup() %&gt;%
    mutate(sd = sqrt(ss / (n - 1)))</code></pre>
<pre class="r"><code>gcCors &lt;- rawGC %&gt;%
    dplyr::filter(read == &quot;R1&quot;) %&gt;%
    dplyr::select(CELL, TREAT, TIME, sample, Freq, read) %&gt;%
    left_join(gcSummary) %&gt;%
    left_join(gcDev) %&gt;%
    dplyr::select(Freq, mn_gc, mn_logLen, sd) %&gt;%
    cor()</code></pre>
<p>A <em>run length encoded</em> (RLE) vector was formed for each sample
taking the number of reads for each transcript as the run lengths, and
both the GC content and length of each transcript as the values.
Transcript lengths were transformed to the log<sub>10</sub> scale due to
the wide variety of lengths contained in the transcriptome.</p>
<p>From these RLEs, the mean GC and mean length was calculated for each
sample, and these values were compared to the proportion of raw reads
with &gt; 70% GC, taking these values from the R1 libraries only.</p>
<pre class="r"><code>a &lt;- gcSummary %&gt;%
    left_join(rawGC) %&gt;%
    dplyr::filter(read == &quot;R1&quot;) %&gt;%
    ggplot(aes(Freq, mn_logLen)) +
    geom_point(aes(colour = TREAT), size = 3) +
    geom_smooth(method = &quot;lm&quot;) +
    scale_shape_manual(values = c(19, 1)) +
    labs(
        x = &quot;Proportion of initial library with &gt; 70% GC&quot;,
        y = &quot;Mean log(length)&quot;,
        colour = &quot;Treatment&quot;
    ) 
b &lt;- gcSummary %&gt;%
    left_join(rawGC) %&gt;%
    dplyr::filter(read == &quot;R1&quot;) %&gt;%
    ggplot(aes(Freq, mn_gc)) +
    geom_point(aes(colour = TREAT), size = 3) +
    geom_smooth(method = &quot;lm&quot;) +
    scale_shape_manual(values = c(19, 1)) +
    scale_y_continuous(labels = percent) +
    labs(
        x = &quot;Proportion of initial library with &gt; 70% GC&quot;,
        y = &quot;Mean GC Content&quot;,
        colour = &quot;Treatment&quot;
    ) 
c &lt;- gcSummary %&gt;%
    left_join(gcDev) %&gt;%
    ggplot(aes(sd, mn_logLen)) +
    geom_point(aes(colour = TREAT), size = 3) +
    geom_smooth(method = &quot;lm&quot;) +
    scale_shape_manual(values = c(19, 1)) +
    scale_y_continuous(breaks = seq(3.2, 3.5, by = 0.02)) +
    labs(
        x = &quot;SD (GC - Theoretical)&quot;,
        y = &quot;Mean log(length)&quot;,
        colour = &quot;Treatment&quot;
    ) 
d &lt;- gcSummary %&gt;%
    left_join(gcDev) %&gt;%
    ggplot(aes(sd, mn_gc)) +
    geom_point(aes(colour = TREAT), size = 3) +
    geom_smooth(method = &quot;lm&quot;) +
    scale_shape_manual(values = c(19, 1)) +
    scale_y_continuous(labels = percent) +
    labs(
        x = &quot;SD (GC - Theoretical)&quot;,
        y = &quot;Mean GC Content&quot;,
        colour = &quot;Treatment&quot;
    ) 
plot_grid(
    plot_grid(
        a + theme(legend.position = &quot;none&quot;),
        b + theme(legend.position = &quot;none&quot;),
        c + theme(legend.position = &quot;none&quot;),
        d + theme(legend.position = &quot;none&quot;),
        nrow = 2
    ),
    get_legend(b),
    nrow = 1,
    rel_widths = c(8,1)
)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/SHSY5Y_QC.Rmd/plotBias-1.png" alt="*Comparison of bias introduced by incomplete rRNA removal. Regression lines are shown along with standard error bands for each comparison.*" width="768" />
<p class="caption">
<em>Comparison of bias introduced by incomplete rRNA removal. Regression
lines are shown along with standard error bands for each
comparison.</em>
</p>
</div>
<div id="initial-pca" class="section level2">
<h2>Initial PCA</h2>
<p>An initial PCA was performed to check the correlations between the
first three components and measured variable, including average GC
content,proportion of libraries with GC content &gt; 70%, mean
log<sub>10</sub> gene length, treatment and time point.</p>
<pre class="r"><code>SH_pca &lt;- cpm(SH_dgeList, log = TRUE) %&gt;%
    t() %&gt;%
    prcomp()
SH_pcaVars &lt;- percent_format(0.1)(summary(SH_pca)$importance[&quot;Proportion of Variance&quot;,])</code></pre>
<pre class="r"><code>SH_pca$x %&gt;%
    as.data.frame() %&gt;%
    rownames_to_column(&quot;sample&quot;) %&gt;%
    left_join(gcSummary) %&gt;%
    as_tibble() %&gt;% 
    left_join(
        dplyr::filter(rawGC, read == &quot;R1&quot;)
    ) %&gt;%
    left_join(gcDev) %&gt;%
    dplyr::select(
        PC1, PC2, PC3,
        Mean_GC = mn_gc,
        Mean_Length = mn_logLen,
        Initial_GC70 = Freq,
        SD = sd,
        TREAT,
        TIME
    ) %&gt;%
    mutate(TREAT = as.numeric(as.factor(TREAT)),
           TIME = as.numeric(as.factor(TIME))) %&gt;%
    cor() %&gt;%
    corrplot(
        type = &quot;lower&quot;,
        diag = FALSE,
        addCoef.col = 1, addCoefasPercent = TRUE
    )</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/SHSY5Y_QC.Rmd/pcaCorrs-1.png" alt="*Correlations between the first three principal components and measured variables. Cell type, time and treatments were converted to an ordered categorical variable for the purposes of visualisation*" width="672" />
<p class="caption">
<em>Correlations between the first three principal components and
measured variables. Cell type, time and treatments were converted to an
ordered categorical variable for the purposes of visualisation</em>
</p>
</div>
<pre class="r"><code>overall &lt;- SH_pca$x %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(&quot;sample&quot;) %&gt;%
  left_join(SH_metadata) %&gt;%
  as_tibble() %&gt;%
  ggplot(aes(PC1, PC2, color = TIME, fill = TIME, shape = TREAT)) +
  geom_point() +
  scale_colour_manual(values = c(&quot;gray80&quot;, &quot;skyblue1&quot;, &quot;dodgerblue1&quot;, &quot;navyblue&quot;)) +
  scale_shape_manual(values = c(17,1,2,19)) +
  coord_cartesian(ylim = c(-30,25), 
                  xlim = c(-55, 40)) +
  guides(fill = FALSE) %&gt;%
  labs(
    x = paste0(&quot;PC1 (&quot;, SH_pcaVars[[&quot;PC1&quot;]], &quot;)&quot;),
    y = paste0(&quot;PC2 (&quot;, SH_pcaVars[[&quot;PC2&quot;]], &quot;)&quot;),
    colour = &quot;TIME&quot;
  )
# Control cells
contr &lt;- SH_pca$x %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(&quot;sample&quot;) %&gt;%
  left_join(SH_metadata) %&gt;%
  dplyr::filter(TREAT %in% c(&quot;DMSO&quot;, &quot;DMEM&quot;)) %&gt;%
  as_tibble() %&gt;%
  ggplot(aes(PC1, PC2, shape = TREAT)) +
  geom_point(aes(color = TIME)) +
  scale_colour_manual(values = c(&quot;gray80&quot;,&quot;dodgerblue1&quot;, &quot;navyblue&quot;)) +
  scale_shape_manual(values = c(1,2)) +
  coord_cartesian(ylim = c(-30,25), 
                  xlim = c(-55, 40)) +
  guides(fill = FALSE) %&gt;%
  labs(
    x = paste0(&quot;PC1 (&quot;, SH_pcaVars[[&quot;PC1&quot;]], &quot;)&quot;),
    y = paste0(&quot;PC2 (&quot;, SH_pcaVars[[&quot;PC2&quot;]], &quot;)&quot;),
    colour = &quot;TIME&quot;
  )
# EBSS vs DMEM
ebss &lt;- SH_pca$x %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(&quot;sample&quot;) %&gt;%
  left_join(SH_metadata) %&gt;%
  dplyr::filter(TREAT %in% c(&quot;EBSS&quot;, &quot;DMEM&quot;)) %&gt;%
  as_tibble() %&gt;%
  ggplot(aes(PC1, PC2, color = TIME, fill = TIME, shape = TREAT)) +
  geom_point() +
  scale_colour_manual(values = c(&quot;gray80&quot;, &quot;skyblue1&quot;, &quot;dodgerblue1&quot;, &quot;navyblue&quot;)) +
  scale_shape_manual(values = c(1,19)) +
  coord_cartesian(ylim = c(-30,25), 
                  xlim = c(-55, 40)) +
  guides(fill = FALSE) %&gt;%
  labs(
    x = paste0(&quot;PC1 (&quot;, SH_pcaVars[[&quot;PC1&quot;]], &quot;)&quot;),
    y = paste0(&quot;PC2 (&quot;, SH_pcaVars[[&quot;PC2&quot;]], &quot;)&quot;),
    colour = &quot;TIME&quot;
  )
# AZD vs DMSO
azd8055 &lt;- SH_pca$x %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(&quot;sample&quot;) %&gt;%
  left_join(SH_metadata) %&gt;%
  dplyr::filter(TREAT %in% c(&quot;AZD8055&quot;, &quot;DMSO&quot;)) %&gt;%
  as_tibble() %&gt;%
  ggplot(aes(PC1, PC2, color = TIME, fill = TIME, shape = TREAT)) +
  geom_point() +
  scale_colour_manual(values = c(&quot;gray80&quot;, &quot;skyblue1&quot;, &quot;dodgerblue1&quot;, &quot;navyblue&quot;)) +
  scale_shape_manual(values = c(17,2)) +
  coord_cartesian(ylim = c(-30,25), 
                  xlim = c(-55, 40)) +
  guides(fill = FALSE) %&gt;%
  labs(
    x = paste0(&quot;PC1 (&quot;, SH_pcaVars[[&quot;PC1&quot;]], &quot;)&quot;),
    y = paste0(&quot;PC2 (&quot;, SH_pcaVars[[&quot;PC2&quot;]], &quot;)&quot;),
    colour = &quot;TIME&quot;
  )
plot_grid(
  plot_grid(
  overall + theme(legend.position=&quot;none&quot;),
  contr+ theme(legend.position=&quot;none&quot;),
  ebss+ theme(legend.position=&quot;none&quot;), 
  azd8055+ theme(legend.position=&quot;none&quot;)
),
get_legend(overall),
rel_widths = c(8,1)
)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/SHSY5Y_QC.Rmd/plotPCA2-1.png" alt="*PCA of (A) all SH-SY5Y samples, (B) only control SH-SY5Y samples, (C) starved samples and corresponding controls and (D) mTOR-inhibited samples and corresponding controls. Plots were colored by time point while shape represent shapes*" width="768" />
<p class="caption">
<em>PCA of (A) all SH-SY5Y samples, (B) only control SH-SY5Y samples,
(C) starved samples and corresponding controls and (D) mTOR-inhibited
samples and corresponding controls. Plots were colored by time point
while shape represent shapes</em>
</p>
</div>
<p>An initial Principal Component Analysis (PCA) was performed using
logCPM values for each SH cell sample. Although 15 hrs and 30 hrs DMEM
didn’t closely cluster with other control cells, in general control
cells at all time points and 1 hrs treated cells are at the same plot in
the first PCA plot. 15 and 30 hrs cells of two treatment groups
clustered at two distinct spots.</p>
<p>In the pervious quality control step, it was identified that
gene-length is strongly correlated with PC1. Therefore,
<em>conditional-quantile normalisation</em> was performed using the
<code>cqn</code> package, which adds <code>glm.offset</code> values to
the origianl <code>dgeList</code> object and calculates all dispersion
estimates.</p>
</div>
<div id="conditional-quantile-normalisation" class="section level2">
<h2>Conditional quantile normalisation</h2>
<pre class="r"><code>SH_gcCqn &lt;- cqn(
  counts = SH_dgeList$counts,
  x = SH_dgeList$genes$gc_content,
  lengths = SH_dgeList$genes$length,
  sizeFactors = SH_dgeList$samples$lib.size
)
# Offset was added to original dgelist
SH_dgeList$offset &lt;- SH_gcCqn$glm.offset </code></pre>
<pre class="r"><code>SH_cpmPostNorm &lt;- SH_gcCqn %&gt;%
  with(y + offset)</code></pre>
</div>
<div id="data-quality-checking" class="section level2">
<h2>Data quality checking</h2>
<p>It is expected that none or negligibal number of genes will appear to
be differentially expressed between two control groups (DMSO &amp;
DMEM). In addition, there shouldn’t be any gene expression changes
between 0hr control cells, 15hrs contorl cells and the 30hrs control
cell. Hence, a 2 x 2 factorial model was built to test the effects of
different media and time.</p>
<pre class="r"><code>d &lt;- model.matrix(
  ~ (TREAT + TIME)^2,
  data = dplyr::filter(SH_dgeList$samples, TREAT %in% c(&quot;DMEM&quot;, &quot;DMSO&quot;)) %&gt;% 
    droplevels()
)</code></pre>
<pre class="r"><code>SHContr_cell &lt;- grepl(paste(dplyr::filter(SH_metadata, TREAT %in% c(&quot;DMEM&quot;, &quot;DMSO&quot;))$sample, collapse=&quot;|&quot;),SH_dgeList$samples$sample)
SHContro_dgeList &lt;- SH_dgeList[,SHContr_cell]
SHContro_dgeList%&lt;&gt;% estimateDisp(design = d)</code></pre>
</div>
<div id="model-fitting" class="section level2">
<h2>Model Fitting</h2>
<pre class="r"><code>minLfc &lt;- log2(2)
alpha &lt;- 0.01
fit &lt;- glmFit(SHContro_dgeList)
topTables_SHContr &lt;- colnames(d)[2:6] %&gt;%
  sapply(function(x){
    glmLRT(fit, coef = x) %&gt;%
      topTags(n = Inf) %&gt;%
      .[[&quot;table&quot;]] %&gt;%
      as_tibble() %&gt;%
      arrange(PValue) %&gt;%
      dplyr::select(
        gene_id, gene_name, logFC, logCPM, PValue, FDR, everything()  
      ) %&gt;%
      mutate(
        coef = x,
        bonfP = p.adjust(PValue, &quot;bonf&quot;),
        DE = case_when(
          bonfP &lt; alpha ~ TRUE,
          FDR &lt; alpha &amp; abs(logFC) &gt; minLfc ~ TRUE
        ),
        DE = ifelse(is.na(DE), FALSE, DE)
      )
  }, simplify = FALSE)</code></pre>
<p>Models were fit using the negative-binomial approaches of
<code>glmFit()</code>. Top Tables were then obtained using
likelihood-ratio tests in <code>glmLRT()</code>. These test the standard
<span class="math inline">\(H_0\)</span> that the true value of the
estimated model coefficient is zero. These model coefficients
effectively compares: 1. 0hr DMSO treated cells vs 0H DMEM treated cells
2. 15 hrs DMSO treated cells vs 0H DMEM treated cells 3. 30 hrs DMSO
treated cells vs 0H DMEM treated cells 4. Difference between 0 and 15
hrs DMSO treated cells vs differences between 0 and 15 hrs DMEM treated
cells 3. Difference between 0 and 30 hrs DMSO treated cells vs
differences between 0 and 30 hrs DMEM treated cells</p>
<p>For enrichment testing, genes were initially considered to be DE
using: a. A Bonferroni-adjusted p-value &lt; 0.01 b. An FDR-adjusted
p-value &lt; 0.01 along with an estimated logFC outside of the range
<span class="math inline">\(\pm \log_2(2)\)</span>.</p>
<p>Using these criteria, the following initial DE gene-sets were
defined:</p>
<pre class="r"><code>topTables_SHContr %&gt;%
  lapply(dplyr::filter, DE) %&gt;% 
  vapply(nrow, integer(1)) %&gt;%
  pander()</code></pre>
<table>
<thead>
<tr class="header">
<th align="center">TREATDMSO</th>
<th align="center">TIME15</th>
<th align="center">TIME30</th>
<th align="center">TREATDMSO:TIME15</th>
<th align="center">TREATDMSO:TIME30</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">55</td>
<td align="center">348</td>
<td align="center">268</td>
<td align="center">254</td>
<td align="center">254</td>
</tr>
</tbody>
</table>
<p>All genes came up to be differentially expressed in this model will
be excluded from downstream analysis.</p>
<pre class="r"><code># contro_deGenes &lt;- topTables_SHContr %&gt;%
#   lapply(dplyr::filter, DE) %&gt;%
#   lapply(magrittr::extract2, &quot;gene_id&quot;) %&gt;%
#   unlist(use.names = FALSE) %&gt;%
#   unique()
# contro_deGenes %&gt;% 
#   as.data.frame() %&gt;%
#   write_csv(
#     path = here::here(&quot;output/DEoutput/DEremove_SH.csv&quot;)
#   )</code></pre>
</div>
</div>
<div id="auophagy-flux" class="section level1">
<h1>Auophagy flux</h1>
<p>As part of the study, the autophagic flux of cells were measured with
an tandem-fluorescent LC3 assay (tf-LC3), which output a RG ratio for
each sample and higher ratio implies higher autophagic activity.</p>
<p>In both control arms, autophagic activities of cells barely changed
through the 30hr time course. mTOR-inhibited cells’ autophagic activity
increased steadily from 1hr to 15hr but plateaued after that. Starved
cells, on the other hand, experienced dramatic rise in autophagic flux
for the entire duration.</p>
<pre class="r"><code>SH_metadata %&gt;%
  ggplot(aes(as.numeric(as.character(TIME)), RG.ratio, colour = TREAT)) +
  stat_summary(
        fun = mean,
        geom=&#39;line&#39;,
        aes(color=TREAT)) +
    stat_summary(
        fun=mean,
        geom=&#39;point&#39;) +
    stat_summary(
        fun.data=mean_cl_boot,
        geom=&#39;errorbar&#39;,
        width=0.5) +
  scale_color_manual(values = treatCols, 
                     name = &quot;Treatment&quot;) + 
  ylab(&quot;Time (hrs)&quot;)</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/SHSY5Y_QC.Rmd/RGplotPCA-1.png" alt="*Plot of Hela cells' RG ratio (autophagic flux)*" width="576" />
<p class="caption">
<em>Plot of Hela cells’ RG ratio (autophagic flux)</em>
</p>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.3 (2022-03-10)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur/Monterey 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_AU.UTF-8/en_AU.UTF-8/en_AU.UTF-8/C/en_AU.UTF-8/en_AU.UTF-8

attached base packages:
[1] splines   stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] DT_0.21                 reshape2_1.4.4          plotly_4.10.0          
 [4] kableExtra_1.3.4        goseq_1.46.0            geneLenDataBase_1.30.0 
 [7] BiasedUrn_1.07          GO.db_3.14.0            msigdbr_7.4.1          
[10] cowplot_1.1.1           UpSetR_1.4.0            corrplot_0.92          
[13] RColorBrewer_1.1-2      pheatmap_1.0.12         ggrepel_0.9.1          
[16] cqn_1.40.0              quantreg_5.88           SparseM_1.81           
[19] preprocessCore_1.56.0   nor1mix_1.3-0           mclust_5.4.9           
[22] pander_0.6.4            scales_1.1.1            ensembldb_2.18.3       
[25] AnnotationFilter_1.18.0 GenomicFeatures_1.46.5  AnnotationDbi_1.56.2   
[28] Biobase_2.54.0          GenomicRanges_1.46.1    GenomeInfoDb_1.30.1    
[31] IRanges_2.28.0          S4Vectors_0.32.3        AnnotationHub_3.2.2    
[34] BiocFileCache_2.2.1     dbplyr_2.1.1            edgeR_3.36.0           
[37] limma_3.50.1            magrittr_2.0.2          forcats_0.5.1          
[40] stringr_1.4.0           dplyr_1.0.8             purrr_0.3.4            
[43] readr_2.1.2             tidyr_1.2.0             tidyverse_1.3.1        
[46] ngsReports_1.10.0       tibble_3.1.6            ggplot2_3.3.5          
[49] BiocGenerics_0.40.0     workflowr_1.7.0        

loaded via a namespace (and not attached):
  [1] utf8_1.2.2                    tidyselect_1.1.2             
  [3] RSQLite_2.2.10                htmlwidgets_1.5.4            
  [5] grid_4.1.3                    BiocParallel_1.28.3          
  [7] munsell_0.5.0                 withr_2.5.0                  
  [9] colorspace_2.0-3              filelock_1.0.2               
 [11] highr_0.9                     knitr_1.37                   
 [13] rstudioapi_0.13               MatrixGenerics_1.6.0         
 [15] labeling_0.4.2                git2r_0.29.0                 
 [17] GenomeInfoDbData_1.2.7        farver_2.1.0                 
 [19] bit64_4.0.5                   rprojroot_2.0.2              
 [21] vctrs_0.3.8                   generics_0.1.2               
 [23] xfun_0.30                     R6_2.5.1                     
 [25] locfit_1.5-9.5                bitops_1.0-7                 
 [27] cachem_1.0.6                  DelayedArray_0.20.0          
 [29] assertthat_0.2.1              vroom_1.5.7                  
 [31] promises_1.2.0.1              BiocIO_1.4.0                 
 [33] nnet_7.3-17                   gtable_0.3.0                 
 [35] processx_3.5.2                rlang_1.0.2                  
 [37] MatrixModels_0.5-0            systemfonts_1.0.4            
 [39] rtracklayer_1.54.0            lazyeval_0.2.2               
 [41] checkmate_2.0.0               broom_0.7.12                 
 [43] BiocManager_1.30.16           yaml_2.3.5                   
 [45] modelr_0.1.8                  crosstalk_1.2.0              
 [47] backports_1.4.1               httpuv_1.6.5                 
 [49] Hmisc_4.6-0                   tools_4.1.3                  
 [51] ellipsis_0.3.2                jquerylib_0.1.4              
 [53] ggdendro_0.1.23               Rcpp_1.0.8.2                 
 [55] plyr_1.8.6                    base64enc_0.1-3              
 [57] progress_1.2.2                zlibbioc_1.40.0              
 [59] RCurl_1.98-1.6                ps_1.6.0                     
 [61] prettyunits_1.1.1             rpart_4.1.16                 
 [63] zoo_1.8-9                     cluster_2.1.2                
 [65] SummarizedExperiment_1.24.0   haven_2.4.3                  
 [67] fs_1.5.2                      here_1.0.1                   
 [69] data.table_1.14.2             reprex_2.0.1                 
 [71] whisker_0.4                   ProtGenerics_1.26.0          
 [73] matrixStats_0.61.0            hms_1.1.1                    
 [75] mime_0.12                     evaluate_0.15                
 [77] xtable_1.8-4                  XML_3.99-0.9                 
 [79] jpeg_0.1-9                    readxl_1.3.1                 
 [81] gridExtra_2.3                 compiler_4.1.3               
 [83] biomaRt_2.50.3                crayon_1.5.0                 
 [85] htmltools_0.5.2               mgcv_1.8-39                  
 [87] later_1.3.0                   tzdb_0.2.0                   
 [89] Formula_1.2-4                 lubridate_1.8.0              
 [91] DBI_1.1.2                     MASS_7.3-55                  
 [93] rappdirs_0.3.3                babelgene_21.4               
 [95] Matrix_1.4-0                  cli_3.2.0                    
 [97] parallel_4.1.3                pkgconfig_2.0.3              
 [99] getPass_0.2-2                 GenomicAlignments_1.30.0     
[101] foreign_0.8-82                xml2_1.3.3                   
[103] svglite_2.1.0                 bslib_0.3.1                  
[105] webshot_0.5.2                 XVector_0.34.0               
[107] rvest_1.0.2                   callr_3.7.0                  
[109] digest_0.6.29                 Biostrings_2.62.0            
[111] rmarkdown_2.13                cellranger_1.1.0             
[113] htmlTable_2.4.0               restfulr_0.0.13              
[115] curl_4.3.2                    shiny_1.7.1                  
[117] Rsamtools_2.10.0              rjson_0.2.21                 
[119] lifecycle_1.0.1               nlme_3.1-155                 
[121] jsonlite_1.8.0                viridisLite_0.4.0            
[123] fansi_1.0.2                   pillar_1.7.0                 
[125] lattice_0.20-45               survival_3.3-1               
[127] KEGGREST_1.34.0               fastmap_1.1.0                
[129] httr_1.4.2                    interactiveDisplayBase_1.32.0
[131] glue_1.6.2                    png_0.1-7                    
[133] BiocVersion_3.14.0            bit_4.0.4                    
[135] stringi_1.7.6                 sass_0.4.0                   
[137] blob_1.2.2                    latticeExtra_0.6-29          
[139] memoise_2.0.1                </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
