---
title: "12_metaDE"
author: "Wenjun Liu"
date: "29/01/2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(
    autodep = TRUE,
	echo = TRUE,
	warning = FALSE,
	message = FALSE
)
```

#Set libaries
```{r loadPackages}
library(ngsReports)
library(plotly)
library(textworks)
library(gtools)
library(tidyverse)
library(magrittr)
library(edgeR)
library(AnnotationHub)
library(ensembldb)
library(scales)
library(pander)
library(ggrepel)
library(pheatmap)
library(RColorBrewer)
library(corrplot)
library(UpSetR)
library(cowplot)
library(msigdbr)
library(GO.db)
library(goseq)
library(kableExtra)
library(gplots)
library(ggalluvial)
library(Numero)
library(igraph)
library(ggraph)
library(ComplexHeatmap)
library(splitstackshape)
source(here::here("smallFunction/enrich_network.R"))
source(here::here("smallFunction/enrich_network2.R"))
source(here::here("smallFunction/overlapGroups.R"))
source(here::here("smallFunction/create_igraph.R"))
source(here::here("smallFunction/spia_my.R"))
library(WGCNA)
library(WebGestaltR)
library(wordcloud)
library(tidytext)
library(circlize)
library(RColorBrewer)
library(scatterpie)
library(SPIA)
library(graphite)
library(gridExtra)
library(poolr)
library(cqn)
library(viridis)
library(openxlsx)
library(colorspace)
```

```{r setOpts}
theme_set(theme_bw())
panderOptions("table.split.table", Inf)
panderOptions("table.style", "rmarkdown")
panderOptions("big.mark", ",")
```

```{r annotationSetup}
ah <- AnnotationHub() %>%
	subset(species == "Homo sapiens") %>%
	subset(rdataclass == "EnsDb")
ensDB <- ah[["AH75011"]]
grTrans <- transcripts(ensDB)
trLengths <- exonsBy(ensDB, "tx") %>%
	width() %>%
	vapply(sum, integer(1))
mcols(grTrans)$length <- trLengths[names(grTrans)]
gcGene <- grTrans %>%
  mcols() %>%
  as.data.frame() %>%
  dplyr::select(gene_id, tx_id, gc_content, length) %>%
  as_tibble() %>%
  group_by(gene_id) %>%
  summarise(
    gc_content = sum(gc_content*length) / sum(length),
    length = ceiling(median(length))
  )
grGenes <- genes(ensDB)
mcols(grGenes) %<>%
  as.data.frame() %>%
  left_join(gcGene) %>%
  as.data.frame() %>%
  DataFrame()
```

```{r dgeList}
dgeList <- here::here("data/dgeList.rds") %>% read_rds()
# dgeList %<>% .[rowSums(cpm(.) >= 1.5) >= 3,]
metadata <- here::here("data", "metaData.txt") %>%
  read.table(header = TRUE, sep = "\t",stringsAsFactors = FALSE) %>% 
  as_tibble() %>%
  mutate_at(vars(one_of(c("CELL", "TREAT","CONTR", "TIME"))), as.factor)
pdf(file = here::here("Manuscript/Figure/newest/autophagic_flux.pdf"), 
    height = 6, 
    width = 4)
metadata %>%
  mutate(Time = as.numeric(as.character(TIME)), 
         TREAT = factor(TREAT, levels = c("DMSO", "DMEM", "AZD8055", "EBSS")),
         Group = ifelse(TREAT %in% c("DMSO", "AZD8055") , "AZD8055", "Starvation")) %>%
  ggplot(aes(TIME, RG.ratio, color = TREAT)) +
  geom_line(aes(group = TREAT)) +
  scale_color_manual(values = c( "gold2","skyblue1", "darkorange1","navyblue" ), 
                     name = "Treatment") +
  geom_point() +
  facet_grid (CELL ~ Group) +
  labs(x = "Time (hs)", 
       y = "Autophagic Flux") +
  theme(legend.position = "top")
dev.off()
```


```{r PCA_3cells, fig.width=10, fig.height=6}
# sampleByCell <- metadata %>%
#   split(f = .$CELL) %>%
#   lapply(pull, sample)
# Hek_cell <- grepl(paste(unique(metadata %>%
#                                  dplyr::filter(CELL = )), collapse="|"),dgeList$samples$sample)  
# Hek_dgeList <- dgeList[,Hek_cell]
# 
# pca_pl <- sapply(names(sampleByCell), function(x){
#   samToKeep <- grepl(paste(unique(sampleByCell[[x]]), collapse="|"),dgeList$samples$sample)  
#   dgeList_sub <- dgeList[,samToKeep]
#   pca <-  cpm(dgeList_sub, log = TRUE) %>%
#     t() %>%
#     prcomp()
#  pcaVars <- percent_format(0.1)(summary(pca)$importance["Proportion of Variance",])
#  pca$x %>%
#   as.data.frame() %>%
#   rownames_to_column("sample") %>%
#   left_join(metadata) %>%
#   as_tibble() %>%
#   ggplot(aes(PC1, PC2, color = TIME,  shape = TREAT)) +
#   geom_point(size= 3) +
#   scale_colour_manual(values = c("gray80", "skyblue1", "dodgerblue1", "navyblue"), 
#                       name = "Time"
#                       ) +
#   scale_shape_manual(values = c(17,1,2,19)) +
#   # coord_cartesian(ylim = c(-50,25), 
#   #                 xlim = c(-40, 65)) +
#   # guides(fill = FALSE, 
#   #        color = FALSE) %>%
#   labs(
#     x = paste0("PC1 (", pcaVars[["PC1"]], ")"),
#     y = paste0("PC2 (", pcaVars[["PC2"]], ")"),
#     shape = "Treatment"
#     # colour = "Time"
#   ) +
#    ggtitle(str_remove(x, ".tfLC3"))
# }, simplify = FALSE)
# # pdf(file = "/Users/wenjunliu/RNA_seq_autophagicflux/Manuscript/Figure/new_draft/PCA_allCells.pdf", 
# #     height = 5, 
# #     width = 9)
# plot_grid(
#   plot_grid(
#    pca_pl$HEK293.tfLC3 + theme(legend.position = "none"),
#    pca_pl$HeLa.tfLC3 + theme(legend.position = "none"),
#    pca_pl$`SH-SY5Y.tfLC3` + theme(legend.position = "none"), 
#    nrow = 1
#   ), 
#   get_legend(pca_pl$HEK293.tfLC3), 
#   ncol = 2, 
#   rel_widths = c(9,1)
#   
# )
# # dev.off()
```


## Read in DE output 
Output from DE analyses were loaded. 

```{r topTables2}
temp = here::here("output/newTop") %>%
    list.files( pattern = "topTables", full.names = TRUE)
list2env(lapply(setNames(temp, gsub("/Users/wenjunliu/RNA_seq_autophagicflux/output/newTop/|.rds","",temp)), 
         read_rds), envir = .GlobalEnv)
topTables2 <- list(
  Hek_AZD = topTables_hekAZD, 
  Hek_EBSS = topTables_hekEBSS,
  SH_AZD = topTables_SHAZD, 
  SH_EBSS = topTables_SHEBSS,
  Hela_AZD = topTables_helaAZD,
  Hela_EBSS = topTables_helaEBSS
)
# calculate average number of genes kept
# topTables2[str_subset(names(topTables2), "AZD")] %>%
#   lapply(function(x)x[["AZD1vsContr"]]) %>%
#   lapply(nrow) %>% 
#   unlist() %>% 
#   mean()
# sapply(names(topTables2), function(x){
#   topTables2[[x]][str_subset(names(topTables2[[x]]), "AZD15|AZD30|EBSS15|EBSS30")]
#   }, simplify = FALSE) %>%
#   saveRDS(file = here::here("output/Simplified/fullDE_output.rds"))
```

```{r}
# genePassedQC <- topTables2[str_subset(names(topTables2), "AZD")] %>%
#   lapply(function(x)x[["AZD1vsContr"]]) %>%
#   lapply(pull, gene_name) 

DE_15and30 <- sapply(names(topTables2), function(x){
  sapply(str_subset(names(topTables2[[x]]), "AZD15|AZD30|EBSS15|EBSS30"), function(y){
    topTables2[[x]][[y]] %>%
      dplyr::filter(FDR < 0.05)
  }, simplify = FALSE) %>%
    lapply(nrow) %>%
    unlist() %>%
    t() %>%
    as.data.frame() %>%
    set_colnames(c("Total15", "Total30")) %>%
    mutate(condition = x)
}, simplify = FALSE) %>%
  bind_rows()
overlap1530 <- sapply(names(topTables2), function(x){
  sapply(str_subset(names(topTables2[[x]]), "AZD15|AZD30|EBSS15|EBSS30"), function(y){
    topTables2[[x]][[y]] %>%
      dplyr::filter(FDR < 0.05) %>%
      pull(gene_name)
  }, simplify = FALSE) %>%
    Reduce(intersect,.) %>%
    length()
}, simplify = FALSE) %>%
    unlist() %>%
    as.data.frame() %>%
  set_colnames("Overlapped") %>%
  rownames_to_column("condition") %>%
  left_join(DE_15and30) %>%
  mutate("15Specific" = Total15 - Overlapped, 
         "30Specific" = Total30 - Overlapped)
# wb <- createWorkbook()
# addWorksheet(wb=wb, sheetName = "Sheet1")
# writeData(wb, sheet = 1,  overlap1530)
# saveWorkbook(wb, here::here("output/Simplified/15&30hrs_comparision.xlsx"), overwrite = TRUE)
```

## Ranking all genes

Following performing pair-wise differential analyses to compare treated cells against their corresponding baseline control cells, result derived from individual DE analyses were merged through generating an overall ranking for all genes. 

Only results from 15 and 30 hrs comparisons were used because significant smaller numbers of DE genes were detected at 1 hr across all conditions. 

Genes were ranked within each DE test first based on their P-values, where genes with lower p-values received higher rankings. If a gene was considered to be undetectable and removed in a cell line, it was assigned a ranking of 0. Rankings were then scaled to a range of 0 to 1 and the scaled rankings were multiplied by sign of logFCs within each DE testing.   

Average of 12 scaled-rankings were taken for each gene and the absolute values of the average ranks were again ranked to obtain the final rankings. Genes with smaller final rankings are more significant.

Rankings of all genes were output into a csv. file.

```{r All_rank}
All_rank <- sapply(names(topTables2),function(x){
  sapply( str_subset(names(topTables2[[x]]), "AZD15|AZD30|EBSS15|EBSS30"), function(y){
    topTables2[[x]][[y]] %>%
      .[!duplicated(.$gene_name),] %>%
      .[order(.$PValue, decreasing = TRUE),] %>%
      mutate(
        rank = 1:nrow(.),
        coeff = paste(str_split(x,"_")[[1]][1], y, sep = "_"), 
        scaled_rank = rescale(rank) * sign(logFC),
        ) %>%
      dplyr::select(gene_name,coeff, scaled_rank) 
  }, simplify = FALSE) %>%
    bind_rows()
}, simplify = FALSE) %>%
    bind_rows() %>%
  pivot_wider(names_from = coeff, 
              values_from = scaled_rank) %>%
  replace(is.na(.),0) %>%
  mutate(aveRank = rowMeans(.[-1],)) %>%
  .[order(abs(.$aveRank), decreasing = TRUE),] %>%
  mutate(rank = 1:nrow(.))

All_rank <- sapply(names(topTables2),function(x){
  sapply( str_subset(names(topTables2[[x]]), "AZD15|AZD30|EBSS15|EBSS30"), function(y){
    topTables2[[x]][[y]] %>%
      mutate(
        coeff = paste(str_split(x,"_")[[1]][1], y, sep = "_")) %>%
      dplyr::select(gene_name, FDR, coeff) %>%
      .[!duplicated(.$gene_name),]
  }, simplify = FALSE) %>%
    bind_rows()
}, simplify = FALSE) %>%
  bind_rows() %>%
  pivot_wider(names_from = coeff, 
              values_from = FDR) %>%
  mutate(max_FDR = apply(.[-1], MARGIN = 1, FUN = max, na.rm = TRUE)) %>%
  dplyr::select(gene_name,max_FDR) %>%
  left_join(All_rank)%>%
  .[order(.$rank),]
```

In each individual testing, genes were also ranked ascendingly by pvalues. Individual rankings, along with the overall rankings were saved into an excel spreadsheet "DE_withRank0206.xlsx"
```{r}
# wb <- createWorkbook()
sheets12 <- sapply(names(topTables2),function(x){
  sapply( str_subset(names(topTables2[[x]]), "AZD15|AZD30|EBSS15|EBSS30"), function(y){
    topTables2[[x]][[y]] %>%
      .[!duplicated(.$gene_name),] %>%
      .[order(.$PValue, decreasing = TRUE),] %>%
      mutate(
        rank = 1:nrow(.),
        coeff = paste(str_split(x,"_")[[1]][1], y, sep = "_"),
        scaled_rank = rescale(rank) * sign(logFC),
        ) %>%
      .[order(.$PValue, decreasing = FALSE),] %>%
      mutate(Sigrank = 1:nrow(.)) %>%
      dplyr::select(gene_name,coeff, scaled_rank, Sigrank, PValue, logFC)
  }, simplify = FALSE) %>%
    bind_rows()
}, simplify = FALSE) %>%
    bind_rows()  %>%
  split(f = .$coeff)
temp1 <- sheets12 %>%
  lapply(pull, gene_name) %>%
  Reduce(union,.) %>%
  unique()
# sheets12$"Overall_rank" <- All_rank
# lapply(seq_along(sheets12), function(i){
#   addWorksheet(wb=wb, sheetName = names(sheets12[i]))
#   writeData(wb, sheet = i, sheets12[[i]])
# })
# saveWorkbook(wb, here::here("output/Simplified/DE_withRank1006.xlsx"), overwrite = TRUE)
```


Within each condition, extract DEGs by selecting genes that had FDR < 0.05 at both time points and moved in the same direction between two time points. Numbers of DEGs under each conditions are:

```{r DE_op}
DE_genes <- sapply(names(topTables2),function(x){
  sapply( str_subset(names(topTables2[[x]]), "AZD15|AZD30|EBSS15|EBSS30"), function(y){
    topTables2[[x]][[y]] %>%
      dplyr::filter(FDR < 0.05) %>%
      mutate(dir = ifelse(logFC < 0, "down", "up")) %>%
      dplyr::select(gene_name, gene_id, dir) %>%
      group_by(gene_name) %>%
      dplyr::filter(n() ==1) %>%
      ungroup()
  }, simplify = FALSE) %>%
    Reduce(inner_join,.) %>%
    .[!duplicated(.$gene_name),]
}, simplify = FALSE)
DE_genes %>%
  lapply(nrow) %>%
  pander()
# save(DE_genes, file = here::here("output/Simplified/DE_genes.Rda"))
```
```{r}
temp2 <- topTables2 %>%
  lapply(function(x)x[[1]]) %>%
  lapply(pull, gene_name) %>%
  Reduce(union,.) %>%
  unique() 
DE_genes[str_subset(names(DE_genes), "EBSS")] %>%
  lapply(function(x)x[!duplicated(x$gene_name),]) %>%
  lapply(pull, gene_name) %>%
  Reduce(union,.) %>%
  unique() %>%
  length()
DE_genes[str_subset(names(DE_genes), "AZD")] %>%
  lapply(pull, gene_name) %>%
  Reduce(union,.) %>%
  unique() %>%
  length()

```

A list of 891 genes that were known to be linked to autophagy endolysosomal network was read in. Their changes in expression at 1 hr were expected and less than 20 AEL genes were significantly differentiallky expressed under each condition. 

```{r}
autophagyGene <- read.csv(file = here::here("data/autophagic_endolysosomal_gene_list.txt")) %>%
  dplyr::rename(gene_name = Gene)
sapply(names(topTables2),function(x){
    topTables2[[x]][[str_subset(names(topTables2[[x]]), "AZD1vs|EBSS1vs")]] %>%
    dplyr::filter(FDR < 0.05, gene_name %in% autophagyGene$gene_name) 
  }, simplify = FALSE)
```
## DEGs

### Consistent DEGs

From the defined DEGs, I firstly captured the gene that were consistently differentially expressed across the entire data-set. 63 genes met the criteria

```{r Consist_genes}
Consist_genes <- DE_genes %>%
  lapply(dplyr::select, "gene_name", "gene_id") %>%
  Reduce(inner_join,.) 
```

56 out of 63 consistent DEGs also showed directional consistency across the entire dataset. 
```{r consist_dir}
consist_dir <- sapply(names(topTables2),function(x){
  sapply( str_subset(names(topTables2[[x]]), "AZD15|AZD30|EBSS15|EBSS30"), function(y){
    topTables2[[x]][[y]] %>%
      dplyr::filter(gene_name %in% Consist_genes$gene_name) %>%
      mutate(
        coeff = paste(str_split(x,"_")[[1]][1], y, sep = "_"),
        dir = ifelse(logFC < 0, -1, 1)) %>%
      dplyr::select(gene_name, dir, coeff) %>%
      .[!duplicated(.$gene_name),]
  }, simplify = FALSE) %>%
    bind_rows()
}, simplify = FALSE) %>%
  bind_rows() %>%
  pivot_wider(names_from = coeff,
              values_from = dir) %>%
  mutate(OverallDir = abs(rowSums(.[-1], na.rm = TRUE))) %>%
  dplyr::filter(OverallDir == 12) %>%
  extract2("gene_name")
consist_nodir <- setdiff(Consist_genes$gene_name, consist_dir)
```

Numbers of up- and down-regualted genes among the 63 consistent DEGs are:

```{r consist_numb, fig.cap="*Numbers of up- and down-regualted genes among the consistent DEGs.*"}

labels <- c("Discordant","Down-regulated","Up-regulated")
temp <- sapply(names(topTables2),function(x){
  sapply(str_subset(names(topTables2[[x]]), "AZD15|AZD30|EBSS15|EBSS30"), function(y){
    topTables2[[x]][[y]] %>%
      dplyr::filter(gene_name %in% consist_dir) %>%
      mutate(
        coeff = paste(str_split(x,"_")[[1]][1], y, sep = "_"),
        dir = ifelse(logFC < 0, -1, 1)) %>%
      dplyr::select(gene_name, dir) %>%
      .[!duplicated(.$gene_name),]
  }, simplify = FALSE) %>%
    bind_rows()
}, simplify = FALSE) %>%
  bind_rows() %>%
  unique() %>%
  extract2("dir") %>% 
  table() %>%
  as.data.frame() %>%
  set_colnames(c("Direction", "Genes")) %>%
  add_row(Direction = "0", Genes = 7) 
temp$Direction <- factor(as.character(temp$Direction), levels = c("0", "-1", "1"))
# pdf(file = "/Users/wenjunliu/RNA_seq_autophagicflux/Manuscript/Figure/new_draft/Consist_numbers.pdf",
#     width = 4,
#     height = 6)
temp %>%
  ggplot(aes(Direction, Genes, fill = Direction))  +
  geom_bar(stat="identity") +
  geom_text(aes(label=Genes), vjust=1.9, size=3.5, color = "white")+
  scale_x_discrete(labels = labels) +
  scale_fill_manual(values = c("mediumpurple2","skyblue2","firebrick")) +
  ylab("Number of genes") +
  theme(legend.position = "none", 
        axis.title.x=element_blank(), 
        panel.border = element_blank()) +
  ggtitle("Consistent DEGs")
# dev.off()
```

Top 25 most significant ones among the 56 directional consistent DEGs were extracted by ranking the maximum FDR (among the 12 comparisons) of those genes. Genes with lowest max_FDR were considered to be more significant. 

```{r top_genes}
top25 <- sapply(names(topTables2),function(x){
  sapply( str_subset(names(topTables2[[x]]), "AZD15|AZD30|EBSS15|EBSS30"), function(y){
    topTables2[[x]][[y]] %>%
      dplyr::filter(gene_name %in% consist_dir) %>%
      mutate(
        coeff = paste(str_split(x,"_")[[1]][1], y, sep = "_")) %>%
      dplyr::select(gene_name, FDR, coeff) %>%
      .[!duplicated(.$gene_name),]
  }, simplify = FALSE) %>%
    bind_rows()
}, simplify = FALSE) %>%
  bind_rows() %>%
  pivot_wider(names_from = coeff,
              values_from = FDR) %>%
  mutate(max_FDR = apply(.[-1], MARGIN = 1, FUN = max, na.rm = TRUE)) %>%
  dplyr::select(gene_name,max_FDR) %>%
  .[order(.$max_FDR),] %>%
  .[1:25,] %>%
  extract2("gene_name")
top10 <- sapply(names(topTables2),function(x){
  sapply( str_subset(names(topTables2[[x]]), "AZD15|AZD30|EBSS15|EBSS30"), function(y){
    topTables2[[x]][[y]] %>%
      dplyr::filter(gene_name %in% consist_dir) %>%
      mutate(
        coeff = paste(str_split(x,"_")[[1]][1], y, sep = "_")) %>%
      dplyr::select(gene_name, FDR, coeff) %>%
      .[!duplicated(.$gene_name),]
  }, simplify = FALSE) %>%
    bind_rows()
}, simplify = FALSE) %>%
  bind_rows() %>%
  pivot_wider(names_from = coeff,
              values_from = FDR) %>%
  mutate(max_FDR = apply(.[-1], MARGIN = 1, FUN = max, na.rm = TRUE)) %>%
  dplyr::select(gene_name,max_FDR) %>%
  .[order(.$max_FDR),] %>%
  .[1:10,] %>%
  extract2("gene_name")
# consistOrdered <- sapply(names(topTables2),function(x){
#   sapply( str_subset(names(topTables2[[x]]), "AZD15|AZD30|EBSS15|EBSS30"), function(y){
#     topTables2[[x]][[y]] %>%
#       dplyr::filter(gene_name %in% consist_dir) %>%
#       mutate(
#         coeff = paste(str_split(x,"_")[[1]][1], y, sep = "_")) %>%
#       dplyr::select(gene_name, FDR, coeff) %>%
#       .[!duplicated(.$gene_name),]
#   }, simplify = FALSE) %>%
#     bind_rows()
# }, simplify = FALSE) %>%
#   bind_rows() %>%
#   pivot_wider(names_from = coeff,
#               values_from = FDR) %>%
#   mutate(max_FDR = apply(.[-1], MARGIN = 1, FUN = max, na.rm = TRUE)) %>%
#   dplyr::select(gene_name,max_FDR) %>%
#   .[order(.$max_FDR),] %>%
#   mutate(rank = row_number()) %>%
#   left_join(sapply(names(topTables2),function(x){
#   sapply( str_subset(names(topTables2[[x]]), "AZD15|AZD30|EBSS15|EBSS30"), function(y){
#     topTables2[[x]][[y]] %>%
#       dplyr::filter(gene_name %in% consist_dir) %>%
#       mutate(
#         time = str_remove(y, "vsContr"),
#         cell = str_split(x, "_")[[1]][1],
#         cell = case_when(cell == "Hek" ~ "HEK293",
#                          cell == "SH" ~ "SH-SY5Y",
#                          cell == "Hela" ~ "HeLa"),
#         coeff = paste(cell, time, sep = "_"),
#         zscore = -sign(logFC) * qnorm(PValue/2),
#         coeff = str_replace(coeff, "_", "+"),
#         coeff = str_replace(coeff, "EBSS", "Star"),
#         coeff = str_replace(coeff, "15", " (15hr)"),
#         coeff = str_replace(coeff, "30", " (30hr)")
#         ) %>%
#       dplyr::select(gene_name,  zscore,coeff)
#   }, simplify = FALSE) %>%
#     bind_rows()
# }, simplify = FALSE) %>%
#   bind_rows() %>%
#   pivot_wider(names_from = coeff, values_from = zscore)) %>%
#   left_join( DE_genes %>%
#                lapply(dplyr::filter, gene_name %in% consist_dir) %>%
#                Reduce(inner_join,.) %>%
#                dplyr::select(-gene_id) %>%
#                dplyr::rename(Direction = dir))
# 
# wb <- createWorkbook()
# addWorksheet(wb=wb, sheetName = "Consistent")
# writeData(wb, sheet = 1, consistOrdered)
```

```{r}
# top10_Average <- sapply(names(topTables2),function(x){
#   sapply( str_subset(names(topTables2[[x]]), "AZD15|AZD30|EBSS15|EBSS30"), function(y){
#     topTables2[[x]][[y]] %>%
#       dplyr::filter(gene_name %in% top10) %>%
#       mutate(
#         time = str_remove(y, "vsContr"), 
#         cell = str_split(x, "_")[[1]][1],
#         coeff = paste(cell, time, sep = "_"), 
#         zscore = -sign(logFC) * qnorm(PValue/2)
#         ) %>%
#       dplyr::select(gene_name,  zscore,coeff)
#   }, simplify = FALSE) %>%
#     bind_rows() %>%
#     pivot_wider(names_from = coeff, values_from = zscore) %>%
#     mutate(aveFC = rowMeans(.[-1]), 
#            coeff = x, 
#            coeff = str_replace(coeff, "_", "+"), 
#            coeff = str_replace(coeff, "EBSS", "Star"),
#            coeff = str_replace(coeff, "Hek", "HEK293"),
#            coeff = str_replace(coeff, "SH", "SH-SY5Y"),
#            coeff = str_replace(coeff, "Hela", "HeLa")) %>%
#     dplyr::select(gene_name, coeff, aveFC)
# }, simplify = FALSE) %>%
#   bind_rows() %>%
#   pivot_wider(names_from = coeff, values_from = aveFC) %>%
#   column_to_rownames("gene_name") 
# top10_Average <- top10_Average %>%
#   Heatmap(
#     cluster_columns = FALSE,
#     column_title = NULL,
#     # show_heatmap_legend = FALSE,
#     heatmap_legend_param = list(title = "Average\nLogFCs", 
#                                 legend_height = unit(2.5, "cm")),
#     column_names_rot = 45, 
#     column_names_gp = gpar(fontsize = 8),
#     col = colorRampPalette(rev(brewer.pal(n = 7, name =
#                                             "RdYlBu")))(100),
#     column_split = rep(c(1,2,3),each = 2),
#     row_km = 2,
#     cluster_column_slices = FALSE,
#     row_title = c("", ""),
#     height = unit(0.5, "cm") * nrow(.),
#     width = unit(1, "cm") * ncol(.)) 
# 
# 
# pdf(file = "/Users/wenjunliu/Desktop/HDR/conferences/BioInfo 2021/DEGs.pdf",
#     width = 8,
#     height = 4,
#     bg  = "transparent")
# draw(top10_Average, heatmap_legend_side = "left", legend_title_gp = gpar(fontsize = 6))
# dev.off()
```

```{r top25, fig.height=6, fig.cap="*LogFCs of the top 25 directional consistent DEGs. Changes in gene expression in those genes were relatively consistent across the dataset.*"}
top25_hp <- sapply(names(topTables2),function(x){
  sapply( str_subset(names(topTables2[[x]]), "AZD15|AZD30|EBSS15|EBSS30"), function(y){
    topTables2[[x]][[y]] %>%
      dplyr::filter(gene_name %in% top25) %>%
      mutate(
        time = str_remove(y, "vsContr"), 
        cell = str_split(x, "_")[[1]][1], 
        cell = case_when(cell == "Hek" ~ "HEK293", 
                         cell == "SH" ~ "SH-SY5Y", 
                         cell == "Hela" ~ "HeLa"),
        coeff = paste(cell, time, sep = "_"), 
        zscore = -sign(logFC) * qnorm(PValue/2), 
        coeff = str_replace(coeff, "_", "+"), 
        coeff = str_replace(coeff, "EBSS", "Star"),
        coeff = str_replace(coeff, "15", " (15hr)"),
        coeff = str_replace(coeff, "30", " (30hr)")
        ) %>%
      dplyr::select(gene_name,  zscore,coeff)
  }, simplify = FALSE) %>%
    bind_rows() 
}, simplify = FALSE) %>%
  bind_rows() %>%
  pivot_wider(names_from = coeff, values_from = zscore) %>%
  column_to_rownames("gene_name") %>%
  Heatmap(
    cluster_columns = FALSE,
    column_title = "Top25 consistent DEGs",
    # show_heatmap_legend = FALSE,
    heatmap_legend_param = list(title = "Z-score"),
    column_names_rot = 90, 
    col = colorRampPalette(rev(brewer.pal(n = 7, name =
                                                          "RdYlBu")))(100),
    column_split = rep(c(1,2,3),each = 4),
  row_km = 2,
  cluster_column_slices = FALSE,
  row_title = c("Upregulated", "Downregulated"),
    height = unit(0.5, "cm") * nrow(.))

# pdf(file = "/Users/wenjunliu/RNA_seq_autophagicflux/Manuscript/Figure/newest/top25.pdf",
#     width = 9,
#     height = 9)
draw(top25_hp)
# dev.off()
# lg <- Legend(col_fun = colorRamp2(c(-10,-5,0,5,10), colorRampPalette(rev(brewer.pal(n = 7, name =
#                                                          "RdYlBu")))(5)),
#              title = "Z-score")
# pdf(file = "/Users/wenjunliu/RNA_seq_autophagicflux/Manuscript/Figure/new_draft/legend1.pdf",
#     width = 2,
#     height =2)
# draw(lg)
# dev.off()
```


```{r top25_Average, fig.height=6, fig.cap="*LogFCs of the top 25 directional consistent DEGs. Changes in gene expression in those genes were relatively consistent across the dataset.*"}
top25_Average <- sapply(names(topTables2),function(x){
  sapply( str_subset(names(topTables2[[x]]), "AZD15|AZD30|EBSS15|EBSS30"), function(y){
    topTables2[[x]][[y]] %>%
      dplyr::filter(gene_name %in% top25) %>%
      mutate(
        time = str_remove(y, "vsContr"), 
        cell = str_split(x, "_")[[1]][1],
        coeff = paste(cell, time, sep = "_"), 
        zscore = -sign(logFC) * qnorm(PValue/2)
        ) %>%
      dplyr::select(gene_name,  zscore,coeff)
  }, simplify = FALSE) %>%
    bind_rows() %>%
    pivot_wider(names_from = coeff, values_from = zscore) %>%
    mutate(aveFC = rowMeans(.[-1]), 
           coeff = x, 
           coeff = str_replace(coeff, "_", "+"), 
           coeff = str_replace(coeff, "EBSS", "Star"),
           coeff = str_replace(coeff, "Hek", "HEK293"),
           coeff = str_replace(coeff, "SH", "SH-SY5Y"),
           coeff = str_replace(coeff, "Hela", "HeLa")) %>%
    dplyr::select(gene_name, coeff, aveFC)
}, simplify = FALSE) %>%
  bind_rows() %>%
  pivot_wider(names_from = coeff, values_from = aveFC) %>%
  column_to_rownames("gene_name") 
top25_Average <- top25_Average %>%
  Heatmap(
    cluster_columns = FALSE,
    column_title = NULL,
    # show_heatmap_legend = FALSE,
    heatmap_legend_param = list(title = "Average logFCs"),
    column_names_rot = 90, 
    col = colorRampPalette(rev(brewer.pal(n = 7, name =
                                                          "RdYlBu")))(100),
    column_split = rep(c(1,2,3),each = 2),
  row_km = 2,
  cluster_column_slices = FALSE,
  row_title = c("", ""),
    height = unit(0.5, "cm") * nrow(.),
    width = unit(1, "cm") * ncol(.)) 


# pdf(file = "/Users/wenjunliu/RNA_seq_autophagicflux/Manuscript/Figure/newest/top25(average).pdf",
#     width = 12,
#     height = 9)
draw(top25_Average, heatmap_legend_side = "left")
# dev.off()
lg <- Legend(col_fun = colorRamp2(c(-10,-5,0,5,10), colorRampPalette(rev(brewer.pal(n = 7, name =
                                                         "RdYlBu")))(5)),
             title = "Z-score")
# pdf(file = "/Users/wenjunliu/RNA_seq_autophagicflux/Manuscript/Figure/newest/legend1.pdf",
#     width = 4,
#     height =4)
draw(lg)
# dev.off()
```



### Treatment specific effects

```{r overlapGroups}
# function for extracting set information for upset plot
overlapGroups <- function (fromListInput, sort = TRUE) {
  fromList2 <- function (input) {
    elements <- unique(unlist(input))
    data <- unlist(lapply(input, function(x) {
      x <- as.vector(match(elements, x))
    }))
    data[is.na(data)] <- as.integer(0)
    data[data != 0] <- as.integer(1)
    data <- data.frame(matrix(data, ncol = length(input), byrow = F))
    data <- data[which(rowSums(data) != 0), ]
    names(data) <- names(input)
    row.names(data) <- elements
    return(data)
  }
  fromListOutput <- fromList2(fromListInput)
  listInputunique <- unique(fromListOutput)
  grouplist <- list()
  for (i in 1:nrow(listInputunique)) {
    currentRow <- listInputunique[i,]
    myelements <- which(apply(fromListOutput,1,function(x) all(x == currentRow))) %>%
      names()
    attr(myelements, "degree") <- currentRow %>% sum(na.rm = TRUE)
    Name <- colnames(listInputunique)[currentRow == 1]
    attrCell <- sapply(str_split(Name, "_"), "[", 1) %>% unique()
    if (length(attrCell) == 1){
      attr(myelements, "Cell") <- attrCell
    } else {
      attr(myelements, "Cell") <- paste(attrCell, collapse = ";")
    }
    attr(myelements, "Num_cell") <- length(attrCell)
    attrDir <- sapply(str_split(Name, "_"), "[", 3) %>% unique()
    if (length(attrDir) == 1){
      attr(myelements, "Dir") <- attrDir
    } else {
      
      attr(myelements, "Dir") <- "Up&Down"
    }
    attrTreat <- sapply(str_split(Name, "_"), "[", 2) %>% unique()
    if (length(attrTreat) == 1){
      attr(myelements, "Treat") <- attrTreat
    } else {
      attr(myelements, "Treat") <- "Both"
    }
    grouplist[[paste(colnames(listInputunique)[currentRow == 1], collapse = ":")]] <- myelements
  }
  if (sort) {
    grouplist <- grouplist[order(sapply(grouplist, function(x) length(x)), decreasing = TRUE)]
  }
  return(grouplist)
  # save element list to facilitate access using an index in case rownames are not named
}
```



From the same list of DEGs defined above, I also extracted the DEGs that were differetially expressed either in all AZD groups (SH_AZD, Hek_AZD and Hela_AZD) or in all EBSS groups (SH_EBSS, Hek_EBSS and Hela_EBSS) but not in any other groups. 

25 genes were unique to AZD8055, and 125 were unique to starvation where 106 of them were also directionally concordant. 

```{r OneTreat_only}
## DEGs that were observed in all cell lines under AZD8055
AZD_all <- DE_genes[str_subset(names(DE_genes), "AZD")] %>%
  lapply(pull, gene_name) %>%
  Reduce(intersect, .) 
## DEGs that were observed in all cell lines under starvation
EBSS_all <- DE_genes[str_subset(names(DE_genes), "EBSS")] %>%
  lapply(pull, gene_name) %>%
  Reduce(intersect, .) 
### DEGs that were specific to AZD8055
AZD_only <- setdiff(AZD_all, DE_genes[str_subset(names(DE_genes), "EBSS")] %>%
  lapply(pull, gene_name) %>%
  Reduce(union, .) )
### DEGs that were specific to starvation
EBSS_only <- setdiff(EBSS_all,DE_genes[str_subset(names(DE_genes), "AZD")] %>%
  lapply(pull, gene_name) %>%
  Reduce(union, .))
### AZD8055-specific DEGs that are directionally concordant
AZD_only_con <- DE_genes[str_subset(names(DE_genes), "AZD")] %>%
  lapply(dplyr::filter, gene_name %in% AZD_only) %>%
  Reduce(inner_join,.) %>%
  pull(gene_name)
### Starvation-specific DEGs that are directionally concordant
EBSS_only_con <- DE_genes[str_subset(names(DE_genes), "EBSS")] %>%
  lapply(dplyr::filter, gene_name %in% EBSS_only) %>%
  Reduce(inner_join,.) %>%
  pull(gene_name)

```


```{r oneTreat_numb, fig.cap="*Numbers of up- and down-regulated genes among AZD8055- and starvation-specific DEGS.*"}
# pdf(file = "/Users/wenjunliu/RNA_seq_autophagicflux/Manuscript/Figure/new_draft/oneTreat_numbers.pdf",
#     # width = 4,
#     height = 4)
labels <- c( "Discordant","Down-regulated","Up-regulated")
EBSS_numb <- sapply(str_subset(names(topTables2), "EBSS"),function(x){
    topTables2[[x]]$EBSS30 %>%
      dplyr::filter(gene_name %in% EBSS_only) %>%
      mutate(dir = ifelse(logFC < 0, -1, 1)) %>%
      dplyr::select(gene_name, dir) %>%
       .[!duplicated(.$gene_name),]

    # group_by(gene_name) %>%
    # filter(n() == 1) %>%
    # ungroup()
}, simplify = FALSE) %>%
  bind_rows() %>%
  chop("dir") %>%
  mutate(overDir = vapply(.$dir, sum, numeric(1)), 
         Direction = case_when(overDir == 3 ~"Upregulated",
                               overDir == -3 ~"Downregulated", 
                               abs(overDir) != 3 ~ "Discordant")) %>%
  group_by(Direction) %>%
  mutate(Genes = n()) %>%
  dplyr::select(Direction, Genes) %>%
  unique() %>%
  ggplot(aes(Direction, Genes, fill = Direction))  +
  geom_bar(stat="identity") +
  geom_text(aes(label=Genes), vjust=1.9, size=3.5, color = "white")+
  scale_x_discrete(labels = labels) +
  scale_fill_manual(values = c("mediumpurple2","skyblue2","firebrick")) +
  ylab("Number of genes") +
  theme(legend.position = "none", 
        axis.title.x=element_blank(), 
        panel.border = element_blank()) +
  ggtitle("Starvation-specific")

labels <- c( "Down-regulated","Up-regulated")
AZD_numb <- sapply(str_subset(names(topTables2), "AZD"),function(x){
    topTables2[[x]]$AZD30 %>%
      dplyr::filter(gene_name %in% AZD_only) %>%
      mutate(dir = ifelse(logFC < 0, -1, 1)) %>%
      dplyr::select(gene_name, dir) %>%
    group_by(gene_name) %>%
    dplyr::filter(n() == 1) %>%
    ungroup()
}, simplify = FALSE) %>%
  bind_rows() %>%
  chop("dir") %>%
  mutate(overDir = vapply(.$dir, sum, numeric(1)), 
         Direction = case_when(overDir == 3 ~"Upregulated",
                               overDir == -3 ~"Downregulated", 
                               abs(overDir) != 3 ~ "Discordant")) %>%
  group_by(Direction) %>%
  mutate(Genes = n()) %>%
  dplyr::select(Direction, Genes) %>%
  unique() %>%
  ggplot(aes(Direction, Genes, fill = Direction))  +
  geom_bar(stat="identity") +
  geom_text(aes(label=Genes), vjust=1.9, size=3.5, color = "white")+
  scale_x_discrete(labels = labels) +
  scale_fill_manual(values = c("skyblue2","firebrick")) +
  ylab("Number of genes") +
  theme(legend.position = "none", 
        axis.title.x=element_blank(), 
        panel.border = element_blank()) +
  ggtitle("mTOR-specific")
# pdf(file = "/Users/wenjunliu/RNA_seq_autophagicflux/Manuscript/Figure/new_draft/oneTreat_number.pdf",
#     height = 4)
plot_grid(EBSS_numb, AZD_numb)
# dev.off()
```

The maximum FDR of concordant starvation-specific DEGs under starvation were ranked to extract the top 25 genes. 

```{r EBSS_only_hp, fig.height = 6, fig.cap="*LogFCs of the top 25 DEGs that were unique to starvation.*"}
EBSS_top25 <- sapply(names(topTables2),function(x){
  sapply(str_subset(names(topTables2[[x]]), "EBSS15|EBSS30"), function(y){
    topTables2[[x]][[y]] %>%
      dplyr::filter(gene_name %in% EBSS_only_con) %>%
      mutate(
        coeff = paste(str_split(x,"_")[[1]][1], y, sep = "_")) %>%
      dplyr::select(gene_name, FDR, coeff) %>%
      .[!duplicated(.$gene_name),]
  }, simplify = FALSE) %>%
    bind_rows()
}, simplify = FALSE) %>%
  bind_rows() %>%
  pivot_wider(names_from = coeff,
              values_from = FDR) %>%
  mutate(max_FDR = apply(.[-1], MARGIN = 1, FUN = max, na.rm = TRUE)) %>%
  dplyr::select(gene_name,max_FDR) %>%
  .[order(.$max_FDR),] %>%
  .[1:25,] %>%
  extract2("gene_name")
ebssOrdered <- sapply(names(topTables2),function(x){
  sapply(str_subset(names(topTables2[[x]]), "EBSS15|EBSS30"), function(y){
    topTables2[[x]][[y]] %>%
      dplyr::filter(gene_name %in% EBSS_only_con) %>%
      mutate(
        coeff = paste(str_split(x,"_")[[1]][1], y, sep = "_")) %>%
      dplyr::select(gene_name, FDR, coeff) %>%
      .[!duplicated(.$gene_name),]
  }, simplify = FALSE) %>%
    bind_rows()
}, simplify = FALSE) %>%
  bind_rows() %>%
  pivot_wider(names_from = coeff,
              values_from = FDR) %>%
  mutate(max_FDR = apply(.[-1], MARGIN = 1, FUN = max, na.rm = TRUE)) %>%
  dplyr::select(gene_name,max_FDR) %>%
  .[order(.$max_FDR),] %>%
  mutate(rank = row_number()) %>%
  left_join(sapply(names(topTables2),function(x){
  sapply( str_subset(names(topTables2[[x]]), "AZD15|AZD30|EBSS15|EBSS30"), function(y){
    topTables2[[x]][[y]] %>%
      dplyr::filter(gene_name %in% EBSS_only_con) %>%
      mutate(
        time = str_remove(y, "vsContr"), 
        cell = str_split(x, "_")[[1]][1], 
        cell = case_when(cell == "Hek" ~ "HEK293", 
                         cell == "SH" ~ "SH-SY5Y", 
                         cell == "Hela" ~ "HeLa"),
        coeff = paste(cell, time, sep = "_"), 
        zscore = -sign(logFC) * qnorm(PValue/2), 
        coeff = str_replace(coeff, "_", "+"), 
        coeff = str_replace(coeff, "EBSS", "Star"),
        coeff = str_replace(coeff, "15", " (15hr)"),
        coeff = str_replace(coeff, "30", " (30hr)")
        ) %>%
      dplyr::select(gene_name,  zscore,coeff)
  }, simplify = FALSE) %>%
    bind_rows() 
}, simplify = FALSE) %>%
  bind_rows() %>%
  pivot_wider(names_from = coeff, values_from = zscore)) %>%
  left_join(DE_genes[str_subset(names(DE_genes), "EBSS")] %>%
              lapply(dplyr::filter, gene_name %in% EBSS_only) %>%
              Reduce(inner_join,.) %>%
              dplyr::select(-gene_id) %>%
              dplyr::rename(Direction = dir)
  )
# addWorksheet(wb=wb, sheetName = "Starvation-specific")
# writeData(wb, sheet = 2, ebssOrdered)

ebss_hp <- sapply(names(topTables2),function(x){
  sapply( str_subset(names(topTables2[[x]]), "AZD15|AZD30|EBSS15|EBSS30"), function(y){
    topTables2[[x]][[y]] %>%
      dplyr::filter(gene_name %in% EBSS_top25) %>%
      mutate(
        time = str_remove(y, "vsContr"), 
        cell = str_split(x, "_")[[1]][1], 
        cell = case_when(cell == "Hek" ~ "HEK293", 
                         cell == "SH" ~ "SH-SY5Y", 
                         cell == "Hela" ~ "HeLa"),
        coeff = paste(cell, time, sep = "_"), 
        zscore = -sign(logFC) * qnorm(PValue/2), 
        coeff = str_replace(coeff, "_", "+"), 
        coeff = str_replace(coeff, "EBSS", "Star"),
        coeff = str_replace(coeff, "15", " (15hr)"),
        coeff = str_replace(coeff, "30", " (30hr)")
        ) %>%
      dplyr::select(gene_name,  zscore,coeff)
  }, simplify = FALSE) %>%
    bind_rows() 
}, simplify = FALSE) %>%
  bind_rows() %>%
  pivot_wider(names_from = coeff, values_from = zscore) %>%
  column_to_rownames("gene_name") %>%
  Heatmap(
    cluster_columns = FALSE,
    column_title = "Top25 Starvation-specific DEGs",
    # show_heatmap_legend = FALSE,
    heatmap_legend_param = list(title = "Z-score"),
    column_names_rot = 90, 
    col = colorRampPalette(rev(brewer.pal(n = 7, name =
                                                          "RdYlBu")))(100),
    column_split = rep(c(1,2,3),each = 4),
  row_km = 2,
  cluster_column_slices = FALSE,
  row_title = c("", ""),
    height = unit(0.5, "cm") * nrow(.))

# pdf(file = "/Users/wenjunliu/RNA_seq_autophagicflux/Manuscript/Figure/newest/top25EBSS.pdf", 
#     width = 9, 
#     height = 9)
draw(ebss_hp , show_heatmap_legend = FALSE)
# dev.off()
```

```{r ebss25_Average, fig.height=6, fig.cap="*Average logFCs of the top 25 Starvation-specific DEGs.*"}
ebss25_Average <- sapply(names(topTables2),function(x){
  sapply( str_subset(names(topTables2[[x]]), "AZD15|AZD30|EBSS15|EBSS30"), function(y){
    topTables2[[x]][[y]] %>%
      dplyr::filter(gene_name %in% EBSS_top25) %>%
      mutate(
        time = str_remove(y, "vsContr"), 
        cell = str_split(x, "_")[[1]][1],
        coeff = paste(cell, time, sep = "_"), 
        zscore = -sign(logFC) * qnorm(PValue/2)
        ) %>%
      dplyr::select(gene_name,  zscore,coeff)
  }, simplify = FALSE) %>%
    bind_rows() %>%
    pivot_wider(names_from = coeff, values_from = zscore) %>%
    mutate(aveFC = rowMeans(.[-1]), 
           coeff = x, 
           coeff = str_replace(coeff, "_", "+"), 
           coeff = str_replace(coeff, "EBSS", "Star"),
           coeff = str_replace(coeff, "Hek", "HEK293"),
           coeff = str_replace(coeff, "SH", "SH-SY5Y"),
           coeff = str_replace(coeff, "Hela", "HeLa")) %>%
    dplyr::select(gene_name, coeff, aveFC)
}, simplify = FALSE) %>%
  bind_rows() %>%
  pivot_wider(names_from = coeff, values_from = aveFC) %>%
  column_to_rownames("gene_name") %>%
  Heatmap(
    cluster_columns = FALSE,
    column_title = NULL,
    # show_heatmap_legend = FALSE,
    heatmap_legend_param = list(title = "Average Z-score"),
    column_names_rot = 90, 
    col = colorRampPalette(rev(brewer.pal(n = 7, name =
                                                          "RdYlBu")))(100),
    column_split = rep(c(1,2,3),each = 2),
  row_km = 2,
  cluster_column_slices = FALSE,
  row_title = c("", ""),
    height = unit(0.5, "cm") * nrow(.),
    width = unit(0.5, "cm") * ncol(.))


# pdf(file = "/Users/wenjunliu/RNA_seq_autophagicflux/Manuscript/Figure/newest/ebss25(average).pdf",
#     width = 9,
#     height = 7)
draw(ebss25_Average)
# dev.off()

```


```{r AZD_only_hp, fig.height = 6, fig.cap="*LogFCs of the 25 DEGs that were unique to AZD8055.*"}
azdOrdered <- sapply(names(topTables2),function(x){
  sapply(str_subset(names(topTables2[[x]]), "AZD15|AZD30"), function(y){
    topTables2[[x]][[y]] %>%
      dplyr::filter(gene_name %in% AZD_only) %>%
      mutate(
        coeff = paste(str_split(x,"_")[[1]][1], y, sep = "_")) %>%
      dplyr::select(gene_name, FDR, coeff) %>%
      .[!duplicated(.$gene_name),]
  }, simplify = FALSE) %>%
    bind_rows()
}, simplify = FALSE) %>%
  bind_rows() %>%
  pivot_wider(names_from = coeff,
              values_from = FDR) %>%
  mutate(max_FDR = apply(.[-1], MARGIN = 1, FUN = max, na.rm = TRUE)) %>%
  dplyr::select(gene_name,max_FDR) %>%
  .[order(.$max_FDR),] %>%
  mutate(rank = row_number()) %>%
  left_join(sapply(names(topTables2),function(x){
  sapply( str_subset(names(topTables2[[x]]), "AZD15|AZD30|EBSS15|EBSS30"), function(y){
    topTables2[[x]][[y]] %>%
      dplyr::filter(gene_name %in% AZD_only) %>%
      mutate(
        time = str_remove(y, "vsContr"), 
        cell = str_split(x, "_")[[1]][1], 
        cell = case_when(cell == "Hek" ~ "HEK293", 
                         cell == "SH" ~ "SH-SY5Y", 
                         cell == "Hela" ~ "HeLa"),
        coeff = paste(cell, time, sep = "_"), 
        zscore = -sign(logFC) * qnorm(PValue/2), 
        coeff = str_replace(coeff, "_", "+"), 
        coeff = str_replace(coeff, "EBSS", "Star"),
        coeff = str_replace(coeff, "15", " (15hr)"),
        coeff = str_replace(coeff, "30", " (30hr)")
        ) %>%
      dplyr::select(gene_name,  zscore,coeff)
  }, simplify = FALSE) %>%
    bind_rows() 
}, simplify = FALSE) %>%
  bind_rows() %>%
  pivot_wider(names_from = coeff, values_from = zscore)) %>%
  left_join(DE_genes[str_subset(names(DE_genes), "AZD")] %>%
              lapply(dplyr::filter, gene_name %in% AZD_only) %>%
              Reduce(inner_join,.) %>%
              dplyr::select(-gene_id) %>%
              dplyr::rename(Direction = dir))
# addWorksheet(wb=wb, sheetName = "AZD8055-specific")
# writeData(wb, sheet = 3, azdOrdered)
azd_hp <- sapply(names(topTables2),function(x){
  sapply( str_subset(names(topTables2[[x]]), "AZD15|AZD30|EBSS15|EBSS30"), function(y){
    topTables2[[x]][[y]] %>%
      dplyr::filter(gene_name %in% AZD_only) %>%
      mutate(
        time = str_remove(y, "vsContr"), 
        cell = str_split(x, "_")[[1]][1], 
        cell = case_when(cell == "Hek" ~ "HEK293", 
                         cell == "SH" ~ "SH-SY5Y", 
                         cell == "Hela" ~ "HeLa"),
        coeff = paste(cell, time, sep = "_"), 
        zscore = -sign(logFC) * qnorm(PValue/2), 
        coeff = str_replace(coeff, "_", "+"), 
        coeff = str_replace(coeff, "EBSS", "Star"),
        coeff = str_replace(coeff, "15", " (15hr)"),
        coeff = str_replace(coeff, "30", " (30hr)")
        ) %>%
      dplyr::select(gene_name,  zscore,coeff)
  }, simplify = FALSE) %>%
    bind_rows() 
}, simplify = FALSE) %>%
  bind_rows() %>%
  pivot_wider(names_from = coeff, values_from = zscore) %>%
  column_to_rownames("gene_name") %>%
  Heatmap(
    cluster_columns = FALSE,
    column_title = "Top25 AZD8055-specific DEGs",
    # show_heatmap_legend = FALSE,
    heatmap_legend_param = list(title = "Z-score"),
    column_names_rot = 90, 
    col = colorRampPalette(rev(brewer.pal(n = 7, name =
                                                          "RdYlBu")))(100),
    column_split = rep(c(1,2,3),each = 4),
  row_km = 2,
  cluster_column_slices = FALSE,
  row_title = c("", ""),
    height = unit(0.5, "cm") * nrow(.))

# pdf(file = "/Users/wenjunliu/RNA_seq_autophagicflux/Manuscript/Figure/new_draft/top25AZD.pdf",
#     width = 9,
#     height = 9)
draw(azd_hp , show_heatmap_legend = FALSE)
# dev.off()

```
```{r azd25_Average, fig.height=6, fig.cap="*Average logFCs of the top 25 mTOR-specific DEGs.*"}
azd25_Average <- sapply(names(topTables2),function(x){
  sapply( str_subset(names(topTables2[[x]]), "AZD15|AZD30|EBSS15|EBSS30"), function(y){
    topTables2[[x]][[y]] %>%
      dplyr::filter(gene_name %in% AZD_only) %>%
      mutate(
        time = str_remove(y, "vsContr"), 
        cell = str_split(x, "_")[[1]][1],
        coeff = paste(cell, time, sep = "_"), 
        zscore = -sign(logFC) * qnorm(PValue/2)
        ) %>%
      dplyr::select(gene_name,  zscore,coeff)
  }, simplify = FALSE) %>%
    bind_rows() %>%
    pivot_wider(names_from = coeff, values_from = zscore) %>%
    mutate(aveFC = rowMeans(.[-1]), 
           coeff = x, 
           coeff = str_replace(coeff, "_", "+"), 
           coeff = str_replace(coeff, "EBSS", "Star"),
           coeff = str_replace(coeff, "Hek", "HEK293"),
           coeff = str_replace(coeff, "SH", "SH-SY5Y"),
           coeff = str_replace(coeff, "Hela", "HeLa")) %>%
    dplyr::select(gene_name, coeff, aveFC)
}, simplify = FALSE) %>%
  bind_rows() %>%
  pivot_wider(names_from = coeff, values_from = aveFC) %>%
  column_to_rownames("gene_name") %>%
  Heatmap(
    cluster_columns = FALSE,
    column_title = NULL,
    # show_heatmap_legend = FALSE,
    heatmap_legend_param = list(title = "Average Z-score"),
    column_names_rot = 90, 
    col = colorRampPalette(rev(brewer.pal(n = 7, name =
                                                          "RdYlBu")))(100),
    column_split = rep(c(1,2,3),each = 2),
  row_km = 2,
  cluster_column_slices = FALSE,
  row_title = c("", ""),
    height = unit(0.5, "cm") * nrow(.),
    width = unit(0.5, "cm") * ncol(.))


# pdf(file = "/Users/wenjunliu/RNA_seq_autophagicflux/Manuscript/Figure/newest/azd25(average).pdf",
#     width = 9,
#     height = 7)
draw(azd25_Average)
# dev.off()

```



### SH-SY5Y specific 

The three cell lines used were also known to be highly distinct therefore DEGs that were unique to SH-SY5Y cell line were also extracted. 

463 genes were differentially expressed under both treatments but only in the SH-SY5Y cell line. 449 of them showed directional consistency across two treatments. 

```{r SH_only}
SH_all <- DE_genes[str_subset(names(DE_genes), "SH")] %>%
  lapply(pull, gene_name) %>%
  Reduce(intersect,.)

otherCell_all <- DE_genes[str_subset(names(DE_genes), "SH", negate = TRUE)] %>%
  bind_rows() %>%
  extract2("gene_name")
SH_only <- setdiff(SH_all, otherCell_all)
SH_only_con <- DE_genes[str_subset(names(DE_genes), "SH")] %>%
  lapply(dplyr::filter, gene_name %in% SH_only) %>%
  Reduce(inner_join,.) %>%
  pull(gene_name)
SH_only_dis <- setdiff(SH_only, SH_only_con)
```

```{r SH_number,fig.cap="*Numbers of up- and down-regulated genes among SH-SY5Y-specific DEGS.*"}
# pdf(file = "/Users/wenjunliu/RNA_seq_autophagicflux/Manuscript/Figure/new_draft/SH_numbers.pdf", width = 4, height = 4)
labels <- c( "Discordant","Down-regulated","Up-regulated")
sapply(str_subset(names(topTables2), "SH"),function(x){
    topTables2[[x]][[str_subset(names(topTables2[[x]]), "30vsContr")]]  %>%
      dplyr::filter(gene_name %in% SH_only) %>%
      mutate(dir = ifelse(logFC < 0, -1, 1)) %>%
      dplyr::select(gene_name, dir) %>%
       .[!duplicated(.$gene_name),]

}, simplify = FALSE) %>%
  bind_rows() %>%
  chop("dir") %>%
  mutate(overDir = vapply(.$dir, sum, numeric(1)), 
         Direction = case_when(overDir == 2 ~"Upregulated",
                               overDir == -2 ~"Downregulated", 
                               abs(overDir) != 2 ~ "Discordant")) %>%
  group_by(Direction) %>%
  mutate(Genes = n()) %>%
  dplyr::select(Direction, Genes) %>%
  unique() %>%
  ggplot(aes(Direction, Genes, fill = Direction))  +
  geom_bar(stat="identity") +
  geom_text(aes(label=Genes), vjust=1.9, size=3.5, color = "white")+
  scale_x_discrete(labels = labels) +
  scale_fill_manual(values = c("mediumpurple2","skyblue2","firebrick")) +
  ylab("Number of genes") +
  theme(legend.position = "none", 
        axis.title.x=element_blank(), 
        panel.border = element_blank()) +
  ggtitle("SY5Y-specific") 
# dev.off()

```

Top 25 SH-SY5Y-unique DEGs were extracted by ranking the max_FDR (among 4 comparisions) of those genes:

```{r SH_top25, fig.height=6,fig.cap="*LogFCs of the top25 genes that were unique to SH-SY5Y.*"}
SH_top25 <- sapply(str_subset(names(topTables2), "SH"), function(x){
  sapply( str_subset(names(topTables2[[x]]), "AZD15|AZD30|EBSS15|EBSS30"), function(y){
    topTables2[[x]][[y]] %>%
      dplyr::filter(gene_name %in% SH_only_con) %>%
      mutate(
        coeff = paste(str_split(x,"_")[[1]][1], y, sep = "_")) %>%
      dplyr::select(gene_name, FDR, coeff) %>%
      .[!duplicated(.$gene_name),]
  }, simplify = FALSE) %>%
    bind_rows()
}, simplify = FALSE) %>%
  bind_rows() %>%
  pivot_wider(names_from = coeff,
              values_from = FDR) %>%
  mutate(max_FDR = apply(.[-1], MARGIN = 1, FUN = max, na.rm = TRUE)) %>%
  dplyr::select(gene_name,max_FDR) %>%
  .[order(.$max_FDR),] %>%
  .[1:25,] %>%
  extract2("gene_name")
sy5yOrdered <- sapply(str_subset(names(topTables2), "SH"), function(x){
  sapply( str_subset(names(topTables2[[x]]), "AZD15|AZD30|EBSS15|EBSS30"), function(y){
    topTables2[[x]][[y]] %>%
      dplyr::filter(gene_name %in% SH_only_con) %>%
      mutate(
        coeff = paste(str_split(x,"_")[[1]][1], y, sep = "_")) %>%
      dplyr::select(gene_name, FDR, coeff) %>%
      .[!duplicated(.$gene_name),]
  }, simplify = FALSE) %>%
    bind_rows()
}, simplify = FALSE) %>%
  bind_rows() %>%
  pivot_wider(names_from = coeff,
              values_from = FDR) %>%
  mutate(max_FDR = apply(.[-1], MARGIN = 1, FUN = max, na.rm = TRUE)) %>%
  dplyr::select(gene_name,max_FDR) %>%
  .[order(.$max_FDR),] %>%
  mutate(rank = row_number()) %>%
  left_join(sapply(names(topTables2),function(x){
  sapply( str_subset(names(topTables2[[x]]), "AZD15|AZD30|EBSS15|EBSS30"), function(y){
    topTables2[[x]][[y]] %>%
      dplyr::filter(gene_name %in% SH_only_con) %>%
      mutate(
        time = str_remove(y, "vsContr"), 
        cell = str_split(x, "_")[[1]][1], 
        cell = case_when(cell == "Hek" ~ "HEK293", 
                         cell == "SH" ~ "SH-SY5Y", 
                         cell == "Hela" ~ "HeLa"),
        coeff = paste(cell, time, sep = "_"), 
        zscore = -sign(logFC) * qnorm(PValue/2), 
        coeff = str_replace(coeff, "_", "+"), 
        coeff = str_replace(coeff, "EBSS", "Star"),
        coeff = str_replace(coeff, "15", " (15hr)"),
        coeff = str_replace(coeff, "30", " (30hr)")
      ) %>%
      dplyr::select(gene_name,  zscore,coeff)
  }, simplify = FALSE) %>%
      bind_rows() 
  }, simplify = FALSE) %>%
    bind_rows() %>%
    pivot_wider(names_from = coeff, values_from = zscore)) %>%
  left_join( DE_genes[str_subset(names(DE_genes), "SH")] %>%
               lapply(dplyr::filter, gene_name %in% SH_only) %>%
               Reduce(inner_join,.) %>%
               dplyr::select(-gene_id) %>%
               dplyr::rename(Direction = dir))
# addWorksheet(wb=wb, sheetName = "SY5Y-specific")
# writeData(wb, sheet = 4, sy5yOrdered)
# saveWorkbook(wb, here::here("output/Simplified/extracted_DEGs.xlsx"), overwrite = TRUE)
SY5Y_hp <- sapply(names(topTables2),function(x){
  sapply( str_subset(names(topTables2[[x]]), "AZD15|AZD30|EBSS15|EBSS30"), function(y){
    topTables2[[x]][[y]] %>%
      dplyr::filter(gene_name %in% SH_top25) %>%
      mutate(
        time = str_remove(y, "vsContr"), 
        cell = str_split(x, "_")[[1]][1], 
        cell = case_when(cell == "Hek" ~ "HEK293", 
                         cell == "SH" ~ "SH-SY5Y", 
                         cell == "Hela" ~ "HeLa"),
        coeff = paste(cell, time, sep = "_"), 
        zscore = -sign(logFC) * qnorm(PValue/2), 
        coeff = str_replace(coeff, "_", "+"), 
        coeff = str_replace(coeff, "EBSS", "Star"),
        coeff = str_replace(coeff, "15", " (15hr)"),
        coeff = str_replace(coeff, "30", " (30hr)")
        ) %>%
      dplyr::select(gene_name,  zscore,coeff)
  }, simplify = FALSE) %>%
    bind_rows() 
}, simplify = FALSE) %>%
  bind_rows() %>%
  pivot_wider(names_from = coeff, values_from = zscore) %>%
  column_to_rownames("gene_name") %>%
  replace(is.na(.), 0) %>%
  Heatmap(
    cluster_columns = FALSE,
    column_title = "Top25 SY5Y-specific DEGs",
    # show_heatmap_legend = FALSE,
    heatmap_legend_param = list(title = "Z-score"),
    column_names_rot = 90, 
    col = colorRampPalette(rev(brewer.pal(n = 7, name =
                                                          "RdYlBu")))(100),
    column_split = rep(c(1,2,3),each = 4),
  row_km = 2,
  cluster_column_slices = FALSE,
  row_title = c("", ""),
    height = unit(0.5, "cm") * nrow(.))

# pdf(file = "/Users/wenjunliu/RNA_seq_autophagicflux/Manuscript/Figure/new_draft/top25SY5Y.pdf",
#     width = 9,
#     height = 9)
draw(SY5Y_hp , show_heatmap_legend = FALSE)
# dev.off()

```
```{r sh25_Average, fig.height=6, fig.cap="*Average logFCs of the top 25 SY5Y-specific DEGs.*"}
sh25_Average <- sapply(names(topTables2),function(x){
  sapply( str_subset(names(topTables2[[x]]), "AZD15|AZD30|EBSS15|EBSS30"), function(y){
    topTables2[[x]][[y]] %>%
      dplyr::filter(gene_name %in% SH_top25) %>%
      mutate(
        time = str_remove(y, "vsContr"), 
        cell = str_split(x, "_")[[1]][1],
        coeff = paste(cell, time, sep = "_"), 
        zscore = -sign(logFC) * qnorm(PValue/2)
        ) %>%
      dplyr::select(gene_name,  zscore,coeff)
  }, simplify = FALSE) %>%
    bind_rows() %>%
    pivot_wider(names_from = coeff, values_from = zscore) %>%
    mutate(aveFC = rowMeans(.[-1]), 
           coeff = x, 
           coeff = str_replace(coeff, "_", "+"), 
           coeff = str_replace(coeff, "EBSS", "Star"),
           coeff = str_replace(coeff, "Hek", "HEK293"),
           coeff = str_replace(coeff, "SH", "SH-SY5Y"),
           coeff = str_replace(coeff, "Hela", "HeLa")) %>%
    dplyr::select(gene_name, coeff, aveFC)
}, simplify = FALSE) %>%
  bind_rows() %>%
  pivot_wider(names_from = coeff, values_from = aveFC) %>%
  column_to_rownames("gene_name") %>%
  replace(is.na(.), 0) %>%
  Heatmap(
    cluster_columns = FALSE,
    column_title = NULL,
    # show_heatmap_legend = FALSE,
    heatmap_legend_param = list(title = "Average Z-score"),
    column_names_rot = 90, 
    col = colorRampPalette(rev(brewer.pal(n = 7, name =
                                                          "RdYlBu")))(100),
    column_split = rep(c(1,2,3),each = 2),
  row_km = 2,
  cluster_column_slices = FALSE,
  row_title = c("", ""),
    height = unit(0.5, "cm") * nrow(.),
    width = unit(0.5, "cm") * ncol(.))


# pdf(file = "/Users/wenjunliu/RNA_seq_autophagicflux/Manuscript/Figure/newest/sh25(average).pdf",
#     width = 9,
#     height = 7)
draw(sh25_Average)
# dev.off()

```


For downstream analyses, the average logFCs of all expressed genes and DEGs under each condition were calculated. 

```{r all_fc, fig.width=14, fig.cap="*Boxplots of logFCs observed within each condition.*"}
all_fc <- sapply(names(topTables2), function(x){
  topTables2[[x]][[str_subset(names(topTables2[[x]]), "AZD15|EBSS15")]] %>%
    dplyr::rename(hr15 = logFC) %>%
  dplyr::select(gene_name, hr15) %>%
    left_join(topTables2[[x]][[str_subset(names(topTables2[[x]]), "AZD30|EBSS30")]]) %>%
    dplyr::select(gene_name, logFC, hr15) %>%
    dplyr::rename(hr30 = logFC) %>%
    mutate(aveFC = rowMeans(.[-1])) %>%
    dplyr::select(gene_name,aveFC) %>%
    .[!duplicated(.$gene_name),]
}, simplify = FALSE)
sapply(names(all_fc), function(x){
  all_fc[[x]] %>%
    mutate(group = x)
}, simplify = FALSE) %>%
  bind_rows() %>%
  ggplot(aes(group,aveFC)) +
  geom_boxplot()
```

```{r DE_fc, fig.width=14, fig.cap="*Boxplots of logFCs of DEG within each condition.*"}
DE_fc <- sapply(names(topTables2), function(x){
  topTables2[[x]][[str_subset(names(topTables2[[x]]), "AZD15|EBSS15")]] %>%
    dplyr::rename(logFC_hr15 = logFC, 
                  Pvalue_hr15 = PValue, 
                  FDR_hr15 = FDR
                  ) %>%
  dplyr::select(gene_name, logFC_hr15, Pvalue_hr15, FDR_hr15) %>%
    left_join(topTables2[[x]][[str_subset(names(topTables2[[x]]), "AZD30|EBSS30")]] %>%
    dplyr::rename(logFC_hr30 = logFC, 
                  Pvalue_hr30 = PValue, 
                  FDR_hr30 = FDR) %>%
    dplyr::select(gene_name, logFC_hr30, Pvalue_hr30, FDR_hr30)) %>%
    mutate(aveFC = rowMeans(.[c("logFC_hr15", "logFC_hr30")])) %>%
    dplyr::filter(gene_name %in% DE_genes[[x]]$gene_name) %>%
    left_join(DE_genes[[x]]) %>%
    .[!duplicated(.$gene_name),] %>%
    dplyr::rename(Direction = dir) %>%
    dplyr::select(gene_name, gene_id, logFC_hr15, Pvalue_hr15, FDR_hr15,logFC_hr30, Pvalue_hr30, FDR_hr30, aveFC, Direction )
}, simplify = FALSE) 

# wb <- createWorkbook()
# lapply(seq_along(DE_fc), function(i){
#   addWorksheet(wb=wb, sheetName = names(DE_fc[i]))
#   writeData(wb, sheet = i, DE_fc[[i]])
# })
# saveWorkbook(wb, here::here("output/Simplified/list_of_DEGs.xlsx"), overwrite = TRUE)
sapply(names(DE_fc), function(x){
  DE_fc[[x]] %>%
    mutate(group = x)
}, simplify = FALSE) %>%
  bind_rows() %>%
  ggplot(aes(group,aveFC)) +
  geom_boxplot()
# save(DE_fc, file = here::here("output/Simplified/DE_fc.Rda"))

```

## TFET

Enrichment of Transcription factor target gene-sets (TFT) among DEGs defined under each condition was tested. 
```{r tf}
entrezGenes <- dgeList$genes %>%
  unnest(entrezid) %>%
  dplyr::rename(entrez_gene = entrezid)
# save(entrezGenes, file = here::here("output/Simplified/entrezGenes.Rda"))
tf <- msigdbr("Homo sapiens", category = "C3") %>%
  dplyr::filter(gs_subcat %in% c("TFT:GTRD", "TFT:TFT_Legacy")) %>%
  dplyr::rename(gene_id = ensembl_gene) %>%
  dplyr::filter(!is.na(gene_id)) %>%
  distinct(gs_name, gene_id, .keep_all = TRUE)
tfByGene <- tf  %>%
  split(f = .$gene_id) %>%
  lapply(extract2, "gs_name")
tfByID <- tf  %>%
  split(f = .$gs_name) %>%
  lapply(extract2, "gene_symbol")
tfByIDID <- tf  %>%
  split(f = .$gs_name) %>%
  lapply(extract2, "gene_id")
```

TFT gene-sets were retrieved from msigdbr. Mappings are required from gene to pathway, and Ensembl identifiers were used to map from gene to pathway, based on the mappings in the previously used annotations (Ensembl Release 98).

A total of `r comma(length(tfByGene))` Ensembl IDs were mapped to pathways from the TFT gene sets. 

```{r formatP}
formatP <- function(p, m = 0.0001){
out <- rep("", length(p))
out[p < m] <- sprintf("%.2e", p[p<m])
out[p >= m] <- sprintf("%.4f", p[p>=m])
out <- as.numeric(out)
out
}
```


```{r gcPwf_DE}
# gcPwf_DE <- sapply(names(DE_genes), function(x){
#   grTrans %>%
#     as.data.frame() %>%
#     distinct(gene_id, .keep_all = TRUE) %>%
#     dplyr::select(gene_id,gc_content) %>%
#     mutate(Status = case_when(gene_id %in% DE_genes[[x]]$gene_id ~ 1, !gene_id %in% DE_genes[[x]]$gene_id  ~ 0)) %>%
#     with(
#       nullp(
#         DEgenes = structure(
#           Status, names = gene_id
#         ),
#         genome = "GRCh38.p16",
#         id = "ensGene",
#         bias.data = gc_content,
#         plot.fit = FALSE
#       )
#     )
# }, simplify = FALSE)
# 
# lenPwf_DE <- sapply(names(DE_genes), function(x){
#   grTrans %>%
#     as.data.frame() %>%
#     distinct(gene_id, .keep_all = TRUE) %>%
#     dplyr::select(gene_id, length) %>%
#     mutate(Status = case_when(gene_id %in% DE_genes[[x]]$gene_id  ~ 1, !gene_id %in% DE_genes[[x]]$gene_id   ~ 0)) %>%
#     with(
#       nullp(
#         DEgenes = structure(
#           Status, names = gene_id
#         ),
#         genome = "GRCh38.p16",
#         id = "ensGene",
#         bias.data =length,
#         plot.fit = FALSE
#       )
#     )
# }, simplify = FALSE)
# ```
# 
# ```{r goseqTF_DE}
# goseqTF_DE <- sapply(names(DE_genes), function(x){
#   goseq(lenPwf_DE[[x]],
#         gene2cat = tfByGene) %>%
#   as_tibble %>%
#   dplyr::filter(numDEInCat > 0) %>%
#   mutate(
#     adjP = p.adjust(over_represented_pvalue, method = "bonf"),
#     FDR = as.numeric(p.adjust(over_represented_pvalue, method = "fdr"))
#   ) %>%
#   dplyr::select(-contains("under")) %>%
#   dplyr::rename(
#     gs_name = category,
#     PValue = over_represented_pvalue,
#     nDE = numDEInCat,
#     nExpressed = numInCat
#   ) %>%
#   left_join(tf) %>%
#     dplyr::select(
#     gs_name, nExpressed, nDE, gs_id,
#     contains("P", ignore.case = FALSE),
#     FDR,
#     gene_name, gene_id
#   ) %>%
#   dplyr::filter(
#     gene_id %in% DE_genes[[x]]$gene_id
#   ) %>%
#     chop(c("gene_name", "gene_id")) %>%
#   mutate(
#     gene_name = vapply(.$gene_name, function(x){
#       paste(x,collapse = ";")
#     }, character(1)),
#     gene_id = vapply(.$gene_id, function(x){
#       paste(x,collapse = ";")
#     }, character(1))
#   ) %>%
#   dplyr::rename(DE = gene_name) %>%
#   mutate_at(
#     vars(one_of(c("PValue", "adjP", "FDR"))),
#     formatP
#   )
# }, simplify = FALSE)

# save(goseqTF_DE, file = here::here("output/Simplified/goseqTF_DE.Rda"))
load(here::here("output/Simplified/goseqTF_DE.Rda"))
names(goseqTF_DE) <- c("HEK293_AZD8055", "HEK293_Starvation", "SH-SY5Y_AZD8055", "SH-SY5Y_Starvation", "HeLa_AZD8055",
                 "HeLa_Starvation")
goseqTF_DE_sig <- goseqTF_DE %>%
  lapply(dplyr::filter, FDR < 0.01) %>%
  lapply(pull, gs_name)
 # goseqTF_DE %>%
 #  lapply(dplyr::filter, gs_name == "BARX1_TARGET_GENES")
```

TFT gene-sets with FDR smaller than 0.05 were defined to be significantly enriched.  Enrichment testing results (includign non-sig) were written into an excel spreadsheet: "TFT_ORA_results.xlsx"
```{r}
# wb <- createWorkbook()
# lapply(seq_along(goseqTF_DE), function(i){
#   temp <- goseqTF_DE[[i]] %>%
#     mutate(Significant = ifelse(FDR < 0.05, TRUE, FALSE))
#   addWorksheet(wb=wb, sheetName = names(goseqTF_DE)[[i]])
#   writeData(wb, sheet = i, temp)
# })
# saveWorkbook(wb, here::here("output/Simplified/TFT_ORA_results.xlsx"), overwrite = TRUE)
```

```{r goseq_DEupset, fig.cap="*Overlap between TFT gene-sets enriched under each condition.From the plot, there are 9 gene-sets enriched under all condition (red), 22 that were unique to SY5Y cell line (blue), and 1 that was unique to starvation treatment (orange). *" }
# pdf(file = "/Users/wenjunliu/RNA_seq_autophagicflux/Manuscript/Figure/newest/TFT_goseq_upset.pdf", onefile = FALSE, 
#     height = 5, 
#     width = 8)
goseqTF_DE %>%
  lapply(dplyr::filter, FDR < 0.05) %>%
  lapply(pull, gs_name) %>%
  fromList() %>%
  upset(sets = colnames(.), 
        nintersects = NA, 
        keep.order = TRUE, 
        # queries = list(
        #   list(query = intersects, 
        #        params = list("HEK293_Starvation",  "SH-SY5Y_Starvation", "HeLa_Starvation"),
        #        color = "orange",
        #        active = T), 
        # list(query = intersects, 
        #      params = list("HEK293_AZD8055","HEK293_Starvation", "SH-SY5Y_AZD8055", "SH-SY5Y_Starvation", "HeLa_AZD8055", 
        #                    "HeLa_Starvation"),
        #      color = "red",
        #      active = T),
        # list(query = intersects,
        #      params = list( "SH-SY5Y_Starvation", "SH-SY5Y_AZD8055"),
        #      color = "blue",
        #      active = T)
        )

# dev.off()
```
```{r goseq_summary}
goseq_all <- goseqTF_DE_sig %>%
  Reduce(intersect,.)
goseq_SY5Y <- goseqTF_DE_sig[str_subset(names(goseqTF_DE_sig), "SH")] %>%
  Reduce(intersect,.) %>%
  setdiff(., goseqTF_DE_sig[str_subset(names(goseqTF_DE_sig), "SH", negate = TRUE)] %>%
            unname() %>%
            unlist())
goseq_EBSS <- goseqTF_DE_sig[str_subset(names(goseqTF_DE_sig), "Starvation")] %>%
  Reduce(intersect,.) %>%
  setdiff(., goseqTF_DE_sig[str_subset(names(goseqTF_DE_sig), "Starvation", negate = TRUE)] %>%
            unname() %>%
            unlist())
```

```{r orTFT_3_hp, fig.width=12, fig.cap="-log10(Pvalue) of TFT pathways enriched (a) only in SY5Y cell line (b) in all conditions, and (c) only under starvation.*"}
temp1 <- sapply(names(goseqTF_DE), function(x){
  goseqTF_DE[[x]] %>%
    dplyr::filter(gs_name %in% goseq_all, 
                  FDR < 0.05) %>%
    mutate(Condition = x, 
           PValue = -log10(PValue)) %>%
    dplyr::select(gs_name, PValue , Condition)
}, simplify = FALSE)
hp1 <- temp1 %>%
  bind_rows() %>%
  ggplot(aes(Condition, 
             gs_name, fill = PValue)) +
  geom_tile(aes(fill = PValue), fontsize =3) +
  scale_fill_viridis(option = "magma", 
                     name = "-log10(P value)") +
  theme(
    axis.title=element_blank(),
    panel.grid = element_blank(),
    # axis.ticks = element_blank(),
    legend.title = element_text(size= 8),
    legend.key.size = unit(3, 'mm'),
    axis.text.x = element_text(angle = 45,hjust=1)
    #   axis.text.y = element_text(colour = col)
  ) +
  ggtitle("Over-represented in all conditions")
temp2 <- sapply(str_subset(names(goseqTF_DE), "SH"), function(x){
  goseqTF_DE[[x]] %>%
    dplyr::filter(gs_name %in% goseq_SY5Y, 
                  FDR < 0.05) %>%
    mutate(Condition = x, 
           PValue = -log10(PValue)) %>%
    dplyr::select(gs_name, PValue, Condition)
}, simplify = FALSE)
hp2 <- temp2 %>%
  bind_rows() %>%
  ggplot(aes(Condition, 
             gs_name, fill = PValue)) +
  geom_tile(aes(fill = PValue), fontsize =3) +
  scale_fill_viridis(option = "magma", 
                     name = "-log10(P value)") +
  theme(
    axis.title=element_blank(),
    panel.grid = element_blank(),
    # axis.ticks = element_blank(),
    legend.title = element_text(size= 8),
    legend.key.size = unit(3, 'mm'),
    axis.text.x = element_text(angle = 45, hjust=1)
    #   axis.text.y = element_text(colour = col)
  ) +
  ggtitle("Over-represented only in SY5Y")

temp3 <- sapply(str_subset(names(goseqTF_DE), "Starvation"), function(x){
  goseqTF_DE[[x]] %>%
    dplyr::filter(gs_name %in% goseq_EBSS, 
                  FDR < 0.05) %>%
    mutate(Condition = x, 
           PValue = -log10(PValue)) %>%
    dplyr::select(gs_name, PValue, Condition)
}, simplify = FALSE)
hp3 <- temp3 %>%
  bind_rows() %>%
  ggplot(aes(Condition, 
             gs_name, fill = PValue)) +
  geom_tile(aes(fill = PValue), fontsize =3) +
  scale_fill_viridis(option = "magma", 
                     name = "-log10(P value)") +
  theme(
    axis.title=element_blank(),
    panel.grid = element_blank(),
    # axis.ticks = element_blank(),
    legend.title = element_text(size= 8),
    legend.key.size = unit(3, 'mm'),
    #   axis.text.y = element_text(colour = col)
  ) +
  ggtitle("Over-represented only under starvation")
# pdf(file = "/Users/wenjunliu/RNA_seq_autophagicflux/Manuscript/Figure/newest/summary_TFT(3hp).pdf",
#     width = 12,
#     height = 7)
plot_grid(hp2, 
          plot_grid(hp1, hp3, 
                    nrow = 2, 
                    rel_heights = c(3,1), 
                    labels = c("b", "c")), 
          ncol = 2, 
          rel_widths  = c(3,5), 
          labels = c("a", ""))
# dev.off()
```


```{r orTFT_2_hp, fig.width=14, fig.height=8, fig.cap="-log10(Pvalue) of TFT pathways enriched (a) in all conditions (b)only in SY5Y cell line.*"}
temp1 <- sapply(names(goseqTF_DE), function(x){
  goseqTF_DE[[x]] %>%
    dplyr::filter(gs_name %in% goseq_all, 
                  FDR < 0.05) %>%
    mutate(Condition = x, 
           logFDR = -log10(FDR), 
           FDR = paste(round(FDR,3)*100, "%", sep = ""),
           FDR = ifelse(FDR == "0%", "<0.1%", FDR), 
           FDR = paste(FDR, "*", sep = "")) %>%
    dplyr::select(gs_name,logFDR, Condition, FDR)
}, simplify = FALSE)%>%
  bind_rows()
temp1_hcl <- temp1 %>%
  dplyr::select(gs_name, logFDR, Condition) %>%
  pivot_wider(names_from = Condition, values_from = logFDR) %>%
  replace(is.na(.),0) %>%
  column_to_rownames("gs_name") %>%
  dist() %>%
  hclust() 
hp1 <- temp1  %>%
  ggplot(aes(factor(gs_name, levels = temp1_hcl$labels[temp1_hcl$order]), Condition)) +
  geom_tile(aes(fill = logFDR)) +
  geom_text(aes(label = FDR), size = 2.5) +
  scale_fill_distiller(palette = "YlOrRd", 
                       name = "-log10(FDR)", 
                       direction = 1, 
                       limits = c(0, max( temp1 %>%
                                            bind_rows() %>%
                                            pull(logFDR)))) +
  theme(
    axis.title=element_blank(),
    panel.grid = element_blank(),
    # axis.ticks = element_blank(),
    legend.title = element_text(size= 12),
    legend.key.size = unit(7, 'mm'),
    axis.text.x = element_text(angle = 45,hjust=1)
    #   axis.text.y = element_text(colour = col)
  ) +
  panel_border(remove = TRUE) +
  ggtitle("Over-represented in all conditions") + 
  coord_fixed(ratio=1)
temp2 <- sapply(str_subset(names(goseqTF_DE), "SH"), function(x){
  goseqTF_DE[[x]] %>%
    dplyr::filter(gs_name %in% goseq_SY5Y, 
                  FDR < 0.05) %>%
     mutate(Condition = x, 
           logFDR = -log10(FDR), 
           FDR = paste(round(FDR,3)*100, "%", sep = ""),
           FDR = ifelse(FDR == "0%", "<0.1%", FDR), 
           FDR = paste(FDR, "*", sep = "")) %>%
    dplyr::select(gs_name,logFDR, Condition, FDR)
}, simplify = FALSE) %>%
  bind_rows()
temp2_hcl <- temp2 %>%
  dplyr::select(gs_name, logFDR, Condition) %>%
  pivot_wider(names_from = Condition, values_from = logFDR) %>%
  replace(is.na(.),0) %>%
  column_to_rownames("gs_name") %>%
  dist() %>%
  hclust() 
hp2 <- temp2  %>%
  ggplot(aes(factor(gs_name, levels = temp2_hcl$labels[temp2_hcl$order]), Condition)) +
  geom_tile(aes(fill = logFDR)) +
  geom_text(aes(label = FDR), size =3) +
  scale_fill_distiller(palette = "YlOrRd", 
                       name = "-log10(FDR)", 
                       direction = 1, 
                       limits = c(0, max( temp2 %>%
                                            bind_rows() %>%
                                            pull(logFDR)))) +
  theme(
    axis.title=element_blank(),
    panel.grid = element_blank(),
    # axis.ticks = element_blank(),
    legend.title = element_text(size= 8),
    legend.key.size = unit(3, 'mm'),
    axis.text.x = element_text(angle = 45, hjust=1)
    #   axis.text.y = element_text(colour = col)
  ) +
  panel_border(remove = TRUE) +
  ggtitle("Over-represented only in SH-SY5Y") + 
  coord_fixed(ratio=1)

plot <- plot_grid(
  plot_grid(hp1 + theme(legend.position = "none"), 
            hp2 + theme(legend.position = "none"), 
            nrow = 2, 
            # rel_heights  = c(3,1), 
            labels = c("a", "b")), 
  get_legend(hp1),
   nrow = 1, 
  rel_widths = c(8,1))
# ggsave(filename = "/Users/wenjunliu/RNA_seq_autophagicflux/Manuscript/Figure/newest/figure4_tft.svg",
#        plot = plot,
#        width = 14,
#        height = 8)
plot

```

### Consistent

Although the 9 TFT gene-sests were enriched in all conditions, the DE TFT targets are not consistent across conditions. In the first network plot below, only the TF-gene edges that were consistent across all conditions were included. 
```{r Consist_aveFC}
Consist_aveFC <- sapply(names(DE_fc), function(x){
  DE_fc[[x]] %>%
    # dplyr::filter(gene_name %in% DE) %>%
    mutate(group = x) %>%
    dplyr::select(gene_name, aveFC, group)
}, simplify = FALSE) %>%
  bind_rows() %>%
  pivot_wider(names_from = group, values_from = aveFC) %>%
  mutate(aveFC = rowMeans(.[-1]), 
         Direction = case_when(rowSums(.[-1] > 0) == 6 ~ "Up", 
                            rowSums(.[-1] > 0) == 0 ~ "Down", 
                            !rowSums(.[-1] > 0) %in% c(0,6) ~ "Mixed")) 
```

```{r consTF_merge, fig.height= 12, fig.width=16, fig.cap="*Regulatory relationships between 9 consistently enriched TFs and DEGs that were perserved across the entire dataset.*"}
temp <- sapply(names(goseqTF_DE), function(x){
  goseqTF_DE[[x]] %>%
  dplyr::filter(gs_name %in% goseq_all) %>%
  mutate(DE = strsplit(as.character(DE), ";"), 
         condition = x, 
         gs_name = str_replace(gs_name, "_TARGET_GENES", "\nTargets")) %>%
  unnest(DE) %>%
  dplyr::rename(gene_name = DE) %>%
  dplyr::select(gs_name,gene_name) %>%
  unique() 
}, simplify = FALSE) %>%
  Reduce(inner_join,.) %>%
  left_join(Consist_aveFC) %>%
  mutate(color = case_when(Direction == "Up" ~"tomato3",
                   Direction == "Down" ~ "steelblue3",
                   Direction == "Mixed" ~ "mediumpurple3"),
         type = gs_name) %>%
  dplyr::rename(from = gs_name, 
                to = gene_name) 
g <- temp %>%
  .[,c("from", "to", "type")] %>%
  graph.data.frame()
TFtoGene <- temp %>%
  #dplyr::select(gs_name, gene_name) %>%
    split(f = .$from)
n <- length(TFtoGene)
totV <- length(V(g))
V(g)$type <- "gene"
V(g)$type[1:n] <- "TF"
V(g)$color[1:n] <- RColorBrewer::brewer.pal(n,"Set3")
V(g)$color[n+1:totV] <- temp$color
gs_col <- RColorBrewer::brewer.pal(n,"Set1")
names(gs_col) <- V(g)$name[1:n]
E(g)$color <- gs_col[E(g)$type ]
pdf(file = "/Users/wenjunliu/RNA_seq_autophagicflux/Manuscript/Figure/newest/consTF_merge.pdf",
    width = 18,
    height = 14,
    bg = "transparent")
network1 <- g %>%
  ggraph(layout = "kk") +
  geom_edge_arc(alpha = 1.3,
                aes_(color = ~I(color)),
                show.legend = FALSE,
                strength = 0.7, 
                linemitre = 5, 
                width = 1.3) +
  geom_node_point(
    aes_(fill  = ~I(color)),
    data = . %>% dplyr::filter(type == "TF"),
    size = 5,
    shape = 21,
    repel = TRUE,
    stroke = 0.5,
    show.legend = FALSE) + 
  geom_node_label(
    aes_(label = ~name, 
         fill  = ~I(color)),
    data = . %>% dplyr::filter(type == "TF"),
    repel = TRUE,
    size = 6,
    force = 0.2,
    label.padding = unit(0.35, "lines"), 
    show.legend = FALSE
  ) +
  geom_node_point(
    aes_(color = ~I(color)),
    data = . %>% dplyr::filter(type == "gene"),
    shape = 17,
    size = 4,
    stroke = 0.5,
    repel = TRUE,
    show.legend = FALSE,
    alpha = 0.7, 
  ) +
  geom_node_text(
    aes(label = name),
    size = 6,
    data = . %>% dplyr::filter(type == "gene"),
    repel = TRUE,
    colour = "black"
  ) +
  theme(panel.border = element_blank(), 
        panel.background = element_blank()
  )
network1
dev.off()
# saveForGEPHI(g, here::here("output/consTF_nodeList.csv"), 
#              here::here("output/consTF_edgeList.csv"))
```

```{r saveForGEPHI}
# Converts the given igraph object to edge and node list reqruied for Gephi and save them at the given filepath locations
#     g: input igraph object to be converted 
#     NODEfilepath: file location where the output node list should be saved
#     EDGEfilepath: file location where the output edge list should be saved
saveForGEPHI <- function(g, NODEfilepath="node_list.csv",  EDGEfilepath ="edge_list.csv")
{
  # gexf nodes require two column data frame (id, label)
  # check if the input vertices has label already present
  # if not, just have the ids themselves as the label
  if(is.null(V(g)$label))
    V(g)$label <- as.character(V(g))
  
  # similarily if edges does not have weight, add default 1 weight
  if(is.null(E(g)$weight))
    E(g)$weight <- rep.int(1, ecount(g))
  
  nodes <- data.frame(cbind(V(g), V(g)$label))
  edges <- t(Vectorize(get.edge, vectorize.args='id')(g, 1:ecount(g)))
  
  # combine all node attributes into a matrix (and take care of & for xml)
  vAttrNames <- setdiff(list.vertex.attributes(g), "label") 
  nodesAtt <- data.frame(sapply(vAttrNames, function(attr) sub("&", "&",get.vertex.attribute(g, attr))))
  
  # combine all edge attributes into a matrix (and take care of & for xml)
  eAttrNames <- setdiff(list.edge.attributes(g), "weight") 
  edgesAtt <- data.frame(sapply(eAttrNames, function(attr) sub("&", "&",get.edge.attribute(g, attr))))
  
  nodes %>%
  cbind(., nodesAtt) %>%
  dplyr::rename(id = X1, 
                label = name) %>%
  dplyr::select(-X2) %>%
  mutate(color = col2hcl(color)) %>%
  write_csv(file = NODEfilepath)
edges %>%
  cbind(., edgesAtt) %>%
  dplyr::rename(Source= 1, 
                Target = 2) %>%
  write_csv(file = EDGEfilepath)
} 



```

### SH-SY5Y

22 TFT gene-sets were enriched under both treatments in and only in SH5Y cell line.  In the network plot below, only the TF-gene edges that appeared under both treatments were included. 

```{r SY5Y_aveFC}
SY5Y_aveFC <- sapply(c("SH-SY5Y_AZD8055", "SH-SY5Y_Starvation"), function(x){
  DE_fc[[x]] %>%
    # dplyr::filter(gene_name %in% DE) %>%
    mutate(group = x) %>%
    dplyr::select(gene_name, aveFC, group)
}, simplify = FALSE) %>%
  bind_rows() %>%
  pivot_wider(names_from = group, values_from = aveFC) %>%
  mutate(aveFC = rowMeans(.[-1]), 
         Direction = case_when(rowSums(.[-1] > 0) == 2 ~ "Up", 
                            rowSums(.[-1] > 0) == 0 ~ "Down", 
                            !rowSums(.[-1] > 0) %in% c(0,2) ~ "Mixed")) 
```


From the network plot, it's clear that the majority of TFTs belonged to the E2F family and their target genes hugely overlapped. Therefore, those 18 gene-sets were merged together in the downstream analysis. 

```{r shTF_merge, fig.height=18, fig.width=16, fig.cap="*Regulatory relationships between 21 consistently enriched TFs and DEGs that were identical under both conditions.*"}
temp <- sapply(c("SH-SY5Y_AZD8055", "SH-SY5Y_Starvation"), function(x){
  goseqTF_DE[[x]] %>%
  dplyr::filter(gs_name %in% goseq_SY5Y) %>%
  mutate(DE = strsplit(as.character(DE), ";"), 
         condition = x) %>%
  unnest(DE) %>%
  dplyr::rename(gene_name = DE) %>%
  dplyr::select(gs_name,gene_name) %>%
  unique() 
}, simplify = FALSE) %>%
  Reduce(inner_join,.) %>%
  left_join(SY5Y_aveFC) %>%
  mutate(color = case_when(Direction == "Up" ~"firebrick",
                   Direction == "Down" ~ "midnightblue",
                   Direction == "Mixed" ~ "forestgreen"),
         type = gs_name) %>%
  dplyr::rename(from = gs_name, 
                to = gene_name) 
g <- temp %>%
  .[,c("from", "to", "type")] %>%
  graph.data.frame()
TFtoGene <- temp %>%
  #dplyr::select(gs_name, gene_name) %>%
    split(f = .$from)
n <- length(TFtoGene)
totV <- length(V(g))
V(g)$type <- "gene"
V(g)$type[1:n] <- "TF"
gs_col <- c(RColorBrewer::brewer.pal(4,"Set3"), rep(RColorBrewer::brewer.pal(5,"Set3")[5], 18))
names(gs_col) <- c(str_subset(V(g)$name[1:n], "E2F", negate = TRUE), str_subset(V(g)$name[1:n], "E2F"))
V(g)$color[1:n] <- gs_col[V(g)$name[1:n]]
V(g)$color[n+1:totV] <- temp$color
E(g)$color <- gs_col[E(g)$type ]
# saveForGEPHI(g, NODEfilepath = here::here("output/sy5yTF_node.csv"),
#              EDGEfilepath = here::here("output/sy5yTF_edge.csv"))
# pdf(file = "/Users/wenjunliu/RNA_seq_autophagicflux/Manuscript/Figure/newest/sy5yTF_merge.pdf",
# 
#     width = 12,
#     height = 15)
network2 <- g %>%
  ggraph(layout = "kk") +
  geom_edge_arc(alpha = 0.4,
                aes_(color = ~I(color)),
                show.legend = FALSE,
                strength = 0.9, 
                linemitre = 5, 
                width = 0.5) +
  geom_node_point(
    aes_(fill  = ~I(color)),
    data = . %>% dplyr::filter(type == "TF"),
    size = 3,
    shape = 21,
    repel = TRUE,
    stroke = 0.5,
    show.legend = FALSE) + 
  geom_node_label(
    aes_(label = ~name, 
         fill  = ~I(color)),
    data = . %>% dplyr::filter(type == "TF"),
    repel = TRUE,
    size = 4,
    force = 0.2,
    label.padding = unit(0.35, "lines"), 
    show.legend = FALSE
  ) +
  geom_node_point(
    aes_(color = ~I(color)),
    data = . %>% dplyr::filter(type == "gene"),
    shape = 17,
    size = 2,
    stroke = 0.5,
    repel = TRUE,
    show.legend = FALSE,
    alpha = 0.7, 
  ) +
  geom_node_text(
    aes(label = name),
    size = 3,
    data = . %>% dplyr::filter(type == "gene"),
    repel = TRUE,
    colour = "black"
  ) +
  theme(panel.border = element_blank(), 
        panel.background = element_blank()
  )
network2
# dev.off()
```

```{r fig.height= 16, fig.width=28}
plot <- plot_grid(network1, network2,  
          nrow = 1, 
          scale  = c(0.9, 1))
ggsave(filename = "/Users/wenjunliu/RNA_seq_autophagicflux/Manuscript/Figure/newest/figure4(alternative).svg",
       plot = plot,
       width = 28,
       height = 16)
```

### EBSS

1 TFT gene-sets was enriched under under all cell lines when starved but no in any cell line when treated with AZD8055.  In the network plot below, only the TF-gene edges that appeared in all cell lines were included. 

```{r EBSS_aveFC}
EBSS_aveFC <- sapply(str_subset(names(DE_fc),"Starvation"), function(x){
  DE_fc[[x]] %>%
    # dplyr::filter(gene_name %in% DE) %>%
    mutate(group = x) %>%
    dplyr::select(gene_name, aveFC, group)
}, simplify = FALSE) %>%
  bind_rows() %>%
  pivot_wider(names_from = group, values_from = aveFC) %>%
  mutate(aveFC = rowMeans(.[-1]), 
         Direction = case_when(rowSums(.[-1] > 0) == 3 ~ "Up", 
                               rowSums(.[-1] > 0) == 0 ~ "Down", 
                               !rowSums(.[-1] > 0) %in% c(0,3) ~ "Mixed")) 
```

```{r ebssTF_merge, fig.height=4, fig.width=4, fig.cap="*Regulatory relationships between 1 consistently enriched TF and DEGs that were perserved across all cell lines.*"}
temp <- sapply(str_subset(names(DE_fc),"Starvation"), function(x){
  goseqTF_DE[[x]] %>%
  dplyr::filter(gs_name %in% goseq_EBSS) %>%
  mutate(DE = strsplit(as.character(DE), ";"), 
         condition = x) %>%
  unnest(DE) %>%
  dplyr::rename(gene_name = DE) %>%
  dplyr::select(gs_name,gene_name) %>%
  unique() 
}, simplify = FALSE) %>%
  Reduce(inner_join,.) %>%
  left_join(EBSS_aveFC) %>%
  mutate(color = case_when(Direction == "Up" ~"firebrick",
                   Direction == "Down" ~ "midnightblue",
                   Direction == "Mixed" ~ "forestgreen"),
         type = gs_name) %>%
  dplyr::rename(from = gs_name, 
                to = gene_name) 
g <- temp %>%
  .[,c("from", "to", "type")] %>%
  graph.data.frame()
TFtoGene <- temp %>%
  #dplyr::select(gs_name, gene_name) %>%
    split(f = .$from)
n <- length(TFtoGene)
totV <- length(V(g))
V(g)$type <- "gene"
V(g)$type[1:n] <- "TF"
V(g)$color <- "black"
V(g)$color[n+1:totV] <- temp$color
# pdf(file = "/Users/wenjunliu/RNA_seq_autophagicflux/Manuscript/Figure/newest/ebssTF_merge.pdf",
# 
#     width = 5,
#     height = 5)
# saveForGEPHI(g, NODEfilepath = here::here("output/EBSSTF_node.csv"),
#              EDGEfilepath = here::here("output/EBSSTF_edge.csv"))
g %>%
  ggraph(layout = "kk") +
  geom_edge_arc(alpha = 0.4,
                aes_(color = ~as.character(type)),
                show.legend = FALSE,
                strength = 0.7, 
                linemitre = 5, 
                width = 2) +
  geom_node_point(
    aes_(fill = ~as.character(name)),
    data = . %>% dplyr::filter(type == "TF"),
    size = 5,
    shape = 21,
    repel = TRUE,
    stroke = 0.5,
    show.legend = FALSE
  ) + 
  geom_node_label(
    aes_(label = ~name, fill = ~as.character(name)
         ),
    data = . %>% dplyr::filter(type == "TF"),
    repel = TRUE,
    size = 4,
    force = 0.2,
    label.padding = unit(0.35, "lines"), 
    show.legend = FALSE
  ) +
  geom_node_point(
    aes_(color = ~I(color)),
    data = . %>% dplyr::filter(type == "gene"),
    shape = 17,
    size = 3,
    stroke = 0.5,
    repel = TRUE,
    show.legend = FALSE,
    alpha = 0.7, 
  ) +
  geom_node_text(
    aes(label = name),
    size = 4.5,
    data = . %>% dplyr::filter(type == "gene"),
    repel = TRUE,
    colour = "black"
  ) +
  theme(panel.border = element_blank(), 
        panel.background = element_blank()
  )
# dev.off()
```

## Regulted KEGG

Significant regulation of KEGG pathways were then determined by combining evidence derived from modified SPIA and goseq. All DEGs defined under each condition were used in this type. 

312 KEGG pathways were extracted using `graphite`

```{r prepareSPIA}
# kegg <- pathways("hsapiens", "kegg")
# kegg <- convertIdentifiers(kegg, "ENTREZID")
# prepareSPIA(kegg, "keggEx")
# kgToGene <- sapply(seq_len(length(kegg)), function(x){
#   kegg[[x]] %>%
#     nodes() %>%
#     as.data.frame() %>%
#   set_colnames("entrez_gene") %>%
#   mutate(entrez_gene = gsub("ENTREZID:", "", entrez_gene),
#          entrez_gene = as.integer(entrez_gene),
#     gs_name = pathwayTitle(kegg[[x]])) %>%
#   left_join(entrezGenes %>%
#               dplyr::select(gene_name, entrez_gene)) %>%
#     dplyr::select(-"entrez_gene")
# }, simplify = FALSE)  %>%
#   bind_rows() %>%
#   drop_na()
# save(kgToGene, file = here::here("output/kgToGene.rds"))
load(here::here("output/kgToGene.rds"))
all <- entrezGenes %>%
  dplyr::filter(!is.na(entrez_gene)) %>%
  mutate(entrez_gene = paste("ENTREZID:", entrez_gene, sep = "")) %>%
  extract2("entrez_gene")
```


```{r kgByGs}
kg <- kgToGene %>%
  left_join(grGenes %>%
              as.data.frame() %>%
              distinct(gene_id, .keep_all = TRUE) %>%
              dplyr::select(gene_id, gene_name))
kgByGene <- kg %>%
  split(f = .$gene_id) %>%
  lapply(pull, gs_name)
```

### Over-representation

KEGG pathways with FDR < 0.05 in `goseq` output were defined to be significantly over-represented. 

```{r goseq_merge}
# goseq_merge <- sapply(names(lenPwf_DE), function(x){
#   goseq(lenPwf_DE[[x]],
#         gene2cat = kg) %>%
#     as_tibble %>%
#     dplyr::filter(numDEInCat > 0) %>%
#     mutate(
#       adjP = p.adjust(over_represented_pvalue, method = "bonf"),
#       FDR = as.numeric(p.adjust(over_represented_pvalue, method = "fdr"))
#     ) %>%
#     dplyr::select(-contains("under")) %>%
#     dplyr::rename(
#       gs_name = category,
#       PValue = over_represented_pvalue,
#       nDE = numDEInCat,
#       nExpressed = numInCat
#     ) %>%
#     left_join(kg) %>%
#     dplyr::select(
#       gs_name, nExpressed, nDE,
#       contains("P", ignore.case = FALSE),
#       FDR,
#       gene_name, gene_id
#     ) %>%
#     dplyr::filter(
#       gene_id %in% DE_genes[[x]]$gene_id
#     ) %>%
#     chop(c("gene_name", "gene_id")) %>%
#     mutate(
#       gene_name = vapply(.$gene_name, function(x){
#         paste(x,collapse = ";")
#       }, character(1)),
#       gene_id = vapply(.$gene_id, function(x){
#         paste(x,collapse = ";")
#       }, character(1))
#     ) %>%
#     mutate_at(
#       vars(one_of(c("PValue", "adjP", "FDR"))),
#       formatP
#     )
# }, simplify = FALSE)
# names(goseq_merge) <- c("HEK293_AZD8055", "HEK293_Starvation","SH-SY5Y_AZD8055", "SH-SY5Y_Starvation", "HeLa_AZD8055",
#                   "HeLa_Starvation")
# save(goseq_merge, file = here::here("output/Simplified/goseq_merge.Rda"))
load(here::here("output/Simplified/goseq_merge.Rda"))

goseq_merge_sig <- goseq_merge %>%
lapply(dplyr::filter, FDR < 0.05)
```

Enrichment testing results were written into an excel spreadsheet: "KEGG_ORA_results.xlsx"
```{r}
# wb <- createWorkbook()
# lapply(seq_along(goseq_merge), function(i){
#   temp <- goseq_merge[[i]] %>%
#      mutate(Significant = ifelse(FDR < 0.05, TRUE, FALSE))
#   addWorksheet(wb=wb, sheetName = names(goseq_merge)[[i]])
#   writeData(wb, sheet = i, temp)
# })
# saveWorkbook(wb, here::here("output/Simplified/KEGG_ORA_results.xlsx"), overwrite = TRUE)
```

Overlap between enriched KEGG pathways defined under each condition was visualized through the upset plot. 

```{r goseq_merge_sig, fig.cap="*Overlap between KEGG pathways defined to be over-represented under each condition.There wasn't any KEGG pathways that were over-represented in all conditions but there were 4 that were OR in 4 conditions (colored red) and 1 pathway that was unique to SY5Y cell line (colored blue).*"}
# pdf(file = "/Users/wenjunliu/RNA_seq_autophagicflux/Manuscript/Figure/newest/KEGG_goseq_upset.pdf",
#     onefile = FALSE,
#     width = 9,
#     height = 5)
goseq_merge %>%
  lapply(dplyr::filter, FDR < 0.05) %>%
  lapply(pull, gs_name) %>%
  fromList() %>%
  upset(sets = colnames(.), 
        nintersects = NA, 
        keep.order = TRUE, 
        queries = list(
          list(query = intersects, 
             params = list("HEK293_AZD8055", "HeLa_AZD8055", 
                           "HeLa_Starvation", "SH-SY5Y_AZD8055"),
             color = "red",
             active = T),
          list(query = intersects, 
             params = list("HeLa_AZD8055", 
                           "HeLa_Starvation", "SH-SY5Y_AZD8055","SH-SY5Y_Starvation"),
             color = "red",
             active = T),
        list(query = intersects, 
             params = list("HEK293_AZD8055","HEK293_Starvation", "HeLa_AZD8055", 
                           "HeLa_Starvation"),
             color = "red",
             active = T),
        list(query = intersects,
             params = list( "SH-SY5Y_Starvation", "SH-SY5Y_AZD8055"),
             color = "blue",
             active = T)
        )
  )
# dev.off()
```


```{r orKEGG_2hp, fig.width=12, fig.cap="-log10(Pvalue) of KEGG pathways enriched (a) in 4 conditions (b) only in SY5Y cell line*"}
orKEGG_4 <- goseq_merge %>%
  lapply(dplyr::filter, FDR < 0.05) %>%
  bind_rows() %>%
  group_by(gs_name) %>%
  filter(n() == 4) %>%
  pull(gs_name) %>%
  unique()
orKEGG_1 <- goseq_merge[str_subset(names(goseq_merge), "SH")] %>%
  lapply(dplyr::filter, FDR < 0.05) %>%
  lapply(pull, gs_name) %>%
  Reduce(intersect,.) %>%
  setdiff(., goseq_merge[str_subset(names(goseq_merge), "SH", negate = TRUE)] %>%
            lapply(dplyr::filter, FDR < 0.05) %>%
            lapply(pull, gs_name) %>%
            unname() %>%
            unlist())
temp1 <- sapply(names(goseq_merge), function(x){
  goseq_merge[[x]] %>%
    dplyr::filter(gs_name %in% orKEGG_4, 
                  FDR < 0.05) %>%
    mutate(Condition = x, 
           PValue = -log10(PValue)) %>%
    dplyr::select(gs_name, PValue , Condition)
}, simplify = FALSE)
hp1 <- temp1 %>%
  bind_rows() %>%
  # rbind(data.frame(gs_name = orKEGG_4, PValue = NA, Condition = "HeLa_Starvation")) %>%
  ggplot(aes(factor(Condition, levels = names(goseq_merge)), 
             gs_name, fill = PValue)) +
  geom_tile(aes(fill = PValue), fontsize =3) +
  scale_fill_viridis(option = "magma", 
                     name = "-log10(P value)") +
  theme(
    axis.title=element_blank(),
    panel.grid = element_blank(),
    # axis.ticks = element_blank(),
    legend.title = element_text(size= 8),
    legend.key.size = unit(3, 'mm'),
    axis.text.x = element_text(angle = 45,hjust=1)
    #   axis.text.y = element_text(colour = col)
  ) +
  ggtitle("Over-represented in at least 4 conditions")
temp2 <- sapply(str_subset(names(goseq_merge), "SH"), function(x){
  goseq_merge[[x]] %>%
    dplyr::filter(gs_name %in% orKEGG_1, 
                  FDR < 0.05) %>%
    mutate(Condition = x, 
           PValue = -log10(PValue)) %>%
    dplyr::select(gs_name, PValue, Condition)
}, simplify = FALSE)
hp2 <- temp2 %>%
  bind_rows() %>%
  ggplot(aes(Condition, 
             gs_name, fill = PValue)) +
  geom_tile(aes(fill = PValue), fontsize =3) +
  scale_fill_viridis(option = "magma", 
                     name = "-log10(P value)") +
  theme(
    axis.title=element_blank(),
    panel.grid = element_blank(),
    # axis.ticks = element_blank(),
    legend.title = element_text(size= 8),
    legend.key.size = unit(3, 'mm'),
    axis.text.x = element_text(angle = 45,hjust=1)
    # axis.text.x = element_text(angle = 45, hjust=1)
    #   axis.text.y = element_text(colour = col)
  ) +
  ggtitle("Over-represented only in SY5Y")
# pdf(file = "/Users/wenjunliu/RNA_seq_autophagicflux/Manuscript/Figure/newest/summary_KEGG_OR(2hp).pdf",
#     width = 10,
#     height = 5)
plot_grid(hp1, hp2, 
          ncol = 2, 
          rel_heights = c(3,1), 
          scale = c(1, 0.5),
          labels = c("a", "b"))
# dev.off()
```

ALl KEGG pathways that were over-represented in at least one condition were extracted and their FDR were visualised. 
```{r keggOR_hp, fig.height= 10, fig.width=16}
allKEGGor <- goseq_merge %>%
  lapply(dplyr::filter, FDR < 0.05) %>%
  bind_rows() %>%
  pull(gs_name) %>%
  unique()
allKEGGor <- setdiff(allKEGGor, "Metabolic pathways")
temp <- sapply(names(goseq_merge), function(x){
  goseq_merge[[x]] %>%
    dplyr::filter(gs_name %in% allKEGGor) %>%
    replace(is.na(.),0) %>%
     mutate(Condition = x, 
           logFDR = -log10(FDR), 
           Sig = ifelse(FDR < 0.05, TRUE, FALSE),
           roundedFDR = paste(round(FDR,3)*100, "%", sep = ""),
           roundedFDR = ifelse(roundedFDR == "0%", "<0.1%", roundedFDR), 
           roundedFDR = ifelse(FDR < 0.05, paste(roundedFDR, "*", sep = ""), roundedFDR)) %>%
    dplyr::select(gs_name,logFDR, Condition, roundedFDR, Sig)
}, simplify = FALSE) %>%
  bind_rows()
allKEGGor_hcl <- temp %>%
  dplyr::select(gs_name, logFDR, Condition) %>%
  pivot_wider(names_from = Condition, values_from = logFDR) %>%
  replace(is.na(.),0) %>%
  column_to_rownames("gs_name") %>%
  dist() %>%
  hclust() 
keggOR_hp <- temp %>%
  ggplot(aes(Condition,factor(gs_name, levels = allKEGGor_hcl$labels[allKEGGor_hcl$order]))) +
  geom_tile(aes(fill = logFDR)) +
  geom_text(data = . %>% dplyr::filter(!Sig), 
            aes(label = roundedFDR), size = 3, color = "azure4") +
  geom_text(data = . %>% dplyr::filter(Sig), 
            aes(label = roundedFDR), size = 3, color = "black") +
  scale_fill_continuous_sequential(palette = "YlOrRd",
                                   name = "-log10(FDR)",
                                   limits = c(0, max( temp %>%
                                                        bind_rows() %>%
                                                        pull(logFDR)))) +
  theme(
    axis.title=element_blank(),
    panel.grid = element_blank(),
    # axis.ticks = element_blank(),
    legend.title = element_text(size= 8),
    legend.key.size = unit(3, 'mm'),
    axis.text.x = element_text(angle = 45, hjust=1)
    #   axis.text.y = element_text(colour = col)
  ) +
  panel_border(remove = TRUE) +
  coord_fixed(ratio= 0.5)
keggOR_hp
```

### Perturbation

The observed PSs of KEGG pathways under each condition were firstly derived. 

```{r net_spia}
# load("/Users/wenjunliu/RNA_seq_autophagicflux/output/Simplified/BminsI.Rda")
# merge_DE_spia <- sapply(names(DE_fc), function(x){
#   DE_fc[[x]] %>%
#     left_join(entrezGenes) %>%
#     dplyr::select(aveFC, entrez_gene) %>%
#     unique() %>%
#     drop_na() %>%
#     mutate(entrez_gene = paste("ENTREZID:", entrez_gene, sep = ""))
# }, simplify = FALSE)
# merge_DE_spia <- sapply(names(merge_DE_spia), function(x){
#   merge_DE_spia[[x]] %>%
#     extract2("aveFC") %>%
#     set_names(merge_DE_spia[[x]]$entrez_gene)
# }, simplify = FALSE)
# spia_netPer <- sapply(names(merge_DE_spia), function(x){
#   spia_my(de = merge_DE_spia[[x]], BminsI = BminsI)
# }, simplify = FALSE)
# save(spia_netPer, file = here::here("output/Simplified/spia_netPer.Rda"))
load(here::here("output/Simplified/spia_netPer.Rda"))
```

To get ready for permutation, DGELists were split by cells, undetectable genes were removed and cqn was applied to add offset terms just like how DE analyses were performed before. 

DGELists were then further split to 6 conditions (cell_treat), and only the baseline control and 15 and 30 hrs treated were kept.

```{r dgeList_byCell}
# metaByCell <- sapply(c("HeLa.tfLC3", "SH-SY5Y.tfLC3","HEK293.tfLC3"), function(x){
#   metadata %>%
#     dplyr::filter(CELL == x)
# }, simplify = FALSE) %>%
#   lapply(mutate, group = paste(TREAT, TIME, sep = "_"))
# sampleByCell <- sapply(names(metaByCell), function(x){
#   grepl(paste(unique(metaByCell[[x]]$sample), collapse="|"),dgeList$samples$sample)
# }, simplify = FALSE)
# dgeList_byCell <- sapply(names(metaByCell), function(x){
#   dgeList[,sampleByCell[[x]]] %>%
#     .[rowSums(cpm(.) >= minCPM) >= minSamples,]
# }, simplify = FALSE)
# cqn_byCell <- sapply(names(dgeList_byCell), function(x){
#   cqn(
#     counts = dgeList_byCell[[x]]$counts,
#     x = dgeList_byCell[[x]]$genes$gc_content,
#     lengths = dgeList_byCell[[x]]$genes$length,
#     sizeFactors = dgeList_byCell[[x]]$samples$lib.size
#   )
# }, simplify = FALSE)
# set_offset <- function(dgelist, offset){
# 	dgelist$offset <- offset
# 	dgelist
# }
# dgeList_byCell <- sapply(names(dgeList_byCell), function(x){
#   set_offset(dgeList_byCell[[x]],cqn_byCell[[x]]$glm.offset)
# }, simplify = FALSE)
# ```
# 
# ```{r dgeByTreat}
# dgeList_AZD <- sapply(names(metaByCell), function(x){
#   sampleToKepp <- grepl(paste(dplyr::filter(metaByCell[[x]], group %in% c("AZD8055_15", "AZD8055_30","DMSO_0"))$sample, collapse="|"),dgeList_byCell[[x]]$samples$sample)
#   dgeList_byCell[[x]][,sampleToKepp]
# }, simplify = FALSE)
# names(dgeList_AZD) <- vapply(names(dgeList_AZD), function(x){
#   paste(str_remove(x, ".tfLC3"), "AZD", sep = "_")
# }, character(1))
# # write_rds(
# #   x = dgeList_AZD,
# #   path = here::here("data/dgeList_AZD.rds"),
# #   compress = "gz"
# # )
# dgeList_EBSS <- sapply(names(metaByCell), function(x){
#     sampleToKepp <- grepl(paste(dplyr::filter(metaByCell[[x]], group %in% c("EBSS_15", "EBSS_30","DMEM_0"))$sample, collapse="|"),dgeList_byCell[[x]]$samples$sample)
#   dgeList_byCell[[x]][,sampleToKepp]
# }, simplify = FALSE)
# names(dgeList_EBSS) <- vapply(names(dgeList_EBSS), function(x){
#   paste(str_remove(x, ".tfLC3"), "EBSS", sep = "_")
# }, character(1))
# dgeByCon <- c(dgeList_AZD, dgeList_EBSS)
# dgeByCon <- sapply(names(dgeByCon), function(x){
#   dgeByCon[[x]] %>%
#     estimateDisp()
# }, simplify = FALSE)
# write_rds(
#   x = dgeByCon,
#   path = here::here("data/dgeByCon.rds"),
#   compress = "gz"
# )

dgeByCon <- here::here("data/dgeByCon.rds") %>% read_rds()

```


To create the empirical null distribution of perturbation scores, sample labels were permuted to generate permuted logFCs and genes were ranked by the permuted p-values. Same number of top ranked genes as the number of DEGs defined under a certain condition was taken for each permutation and used in SPIA. This step was performed on Phoenix.

SPIA algorithm was applied 1000 times for each condition. This step was performed on Phoenix and outputs were imported back.

Looking at the distribution of permtued tAs for 10 KEGG pathways in HeLa_AZD8055. Most of the distributions are roughly normally distributed. Significance of KEGG pathway perturbation was defined by FDR derived from robust z-scores of observed perturbation scores, which was calculated by ((obs tA) - median(permuted tA))/mad(permuted tA). KEGG pathways with FDR < 0.05 were defined to be significantly perturbed concensusly. 

```{r spia_net_sum}
load(here::here("output/Simplified/spia_net_permut.Rda"))
# spia_net_permut$HeLa_AZD8055 %>%
#       split(f = .$gs_name) %>%
#   .[1:10] %>%
#   lapply(pull, tA) %>%
#   lapply(hist)
spia_net_sum <- sapply(names(spia_net_permut), function(x){
  spia_net_permut[[x]] %>%
    split(f = .$gs_name) %>%
    lapply(dplyr::mutate, 
           MAD = mad(tA), 
           MEDIAN = median(tA)) %>%
    lapply(dplyr::select,gs_name, MAD, 
           MEDIAN) %>%
    lapply(unique) %>%
    bind_rows() %>%
    dplyr::filter(MAD != 0) %>%
    left_join(spia_netPer[[x]]) %>%
    mutate(
      Z_mad = (tA - MEDIAN)/MAD) %>% 
    mutate(
      p_mad = 2*pnorm(-abs(Z_mad)),
      bonf_mad = p.adjust(p_mad, method = "bonferroni"),
      FDR_mad = p.adjust(p_mad, method = "fdr"),
      Sig = ifelse(FDR_mad < 0.05, TRUE, FALSE)) %>%
    .[order(.$FDR_mad),] %>%
    mutate(Rank_spia = row_number()) 
}, simplify = FALSE) 
```


KEGG pathways that were significantly perturbed were extracted. Intersection between conditions were visualized. 

```{r spia_NETupset, fig.cap="*Overlap between KEGG pathways perturbed under each condition.From the plot, there are 2 pathways enriched under all condition (red), 7 that were unique to SY5Y cell line (blue), and 1 that were unique to starvation treatment (orange). *" }

spia_net_sig <- sapply(names(spia_net_permut), function(x){
    spia_net_sum[[x]] %>%
      dplyr::filter(FDR_mad < 0.05) %>%
    pull(gs_name)  
  }, simplify = FALSE)
# pdf(file = "/Users/wenjunliu/RNA_seq_autophagicflux/Manuscript/Figure/newest/KEGG_spia_upset.pdf",
#     onefile = FALSE,
#     height = 5,
#     width = 8)
spia_net_sig %>%
  fromList() %>%
  .[,c("HEK293_AZD8055", "HEK293_Starvation", "SH-SY5Y_AZD8055", "SH-SY5Y_Starvation", "HeLa_AZD8055",
                 "HeLa_Starvation")] %>%
  upset(sets = colnames(.), 
        nintersects = NA, 
        keep.order = TRUE, 
        queries = list(
          list(query = intersects, 
               params = list("HEK293_Starvation",  "SH-SY5Y_Starvation", "HeLa_Starvation"),
               color = "orange",
               active = T), 
        list(query = intersects, 
             params = list("HEK293_AZD8055","HEK293_Starvation", "SH-SY5Y_AZD8055", "SH-SY5Y_Starvation", "HeLa_AZD8055", 
                           "HeLa_Starvation"),
             color = "red",
             active = T),
        list(query = intersects,
             params = list( "SH-SY5Y_Starvation", "SH-SY5Y_AZD8055"),
             color = "blue",
             active = T)
        ))
# dev.off()
```
Enrichment testing results were written into an excel spreadsheet: "KEGG_spia_results.xlsx"
```{r}
# wb <- createWorkbook()
# lapply(seq_along(spia_net_sum), function(i){
#   temp <- spia_net_sum[[i]] %>%
#     mutate(Significant = ifelse(FDR_mad < 0.05, TRUE, FALSE))
#   addWorksheet(wb=wb, sheetName = names(spia_net_sum)[[i]])
#   writeData(wb, sheet = i, temp)
# })
# saveWorkbook(wb, here::here("output/Simplified/KEGG_spia_results.xlsx"), overwrite = TRUE)
```

Consensus perturbation scores of KEGG pathways mentioned above were visualised. 

```{r perturbKEGG_3_hp, fig.width=10, fig.cap="Consensus PCs of KEGG pathways significant perturbed (a) in all conditions (b) only in SY5Y cell line, and (c) only under starvation.*"}
hp1_gs <- spia_net_sig %>%
  Reduce(intersect,.)
hp2_gs <- spia_net_sig[str_subset(names(spia_net_sig), "SH")] %>%
  Reduce(intersect,.) %>%
  setdiff(., spia_net_sig[str_subset(names(spia_net_sig), "SH", negate = TRUE)] %>%
            unname() %>%
            unlist())
hp3_gs <- spia_net_sig[str_subset(names(spia_net_sig), "Starvation")] %>%
  Reduce(intersect,.) %>%
  setdiff(., spia_net_sig[str_subset(names(spia_net_sig), "Starvation", negate = TRUE)] %>%
            unname() %>%
            unlist())
temp1 <- sapply(names(spia_net_permut), function(x){
  spia_net_sum[[x]] %>%
    dplyr::filter(gs_name %in% hp1_gs, 
                  FDR_mad < 0.05) %>%
    mutate(Condition = x) %>%
    dplyr::select(gs_name, Z_mad, Condition)
}, simplify = FALSE)
limit <- temp1 %>%
  bind_rows() %>%
  pull(Z_mad) %>%
  max(abs(.))* c(-1, 1)
hp1 <- temp1 %>%
  bind_rows() %>%
  ggplot(aes(Condition, 
             gs_name, fill = Z_mad)) +
  geom_tile(aes(fill = Z_mad), fontsize =3) +
  geom_text(aes(label = round(Z_mad,2))) +
  scale_fill_distiller(palette = "RdBu", 
                       limit = limit,
                       name = "Normalised Perturbation Score"
                       ) +
  theme_dark()+
  theme(
    axis.title=element_blank(),
    panel.grid = element_blank(),
    # axis.ticks = element_blank(),
    legend.title = element_text(size= 8),
    legend.key.size = unit(3, 'mm'),
    axis.text.x = element_text(angle = 45,hjust=1)
    #   axis.text.y = element_text(colour = col)
  ) +
  ggtitle("Perturbed in all conditions")

temp2 <- sapply(str_subset(names(spia_net_sig), "SH"), function(x){
  spia_net_sum[[x]] %>%
    dplyr::filter(gs_name %in% hp2_gs, 
                  FDR_mad < 0.05) %>%
    mutate(Condition = x) %>%
    dplyr::select(gs_name, Z_mad, Condition)
}, simplify = FALSE)
limit <- temp2 %>%
  bind_rows() %>%
  pull(Z_mad) %>%
  max(abs(.))* c(-1, 1)
hp2 <- temp2 %>%
  bind_rows() %>%
  ggplot(aes(Condition, 
             gs_name, fill = Z_mad)) +
  geom_tile(aes(fill = Z_mad), fontsize =3) +
  geom_text(aes(label = round(Z_mad,2))) +
  scale_fill_distiller(palette = "RdBu", 
                       limit = limit,
                       name = "Normalised Perturbation Score"
                       ) +
  theme_dark()+
  theme(
    axis.title=element_blank(),
    panel.grid = element_blank(),
    # axis.ticks = element_blank(),
    legend.title = element_text(size= 8),
    legend.key.size = unit(3, 'mm'),
    axis.text.x = element_text(angle = 45,hjust=1)
    #   axis.text.y = element_text(colour = col)
  ) +
  ggtitle("Perturbed only in SY5Y")

temp3 <- sapply(str_subset(names(spia_net_sig), "Starvation"), function(x){
  spia_net_sum[[x]] %>%
    dplyr::filter(gs_name %in% hp3_gs, 
                  FDR_mad < 0.05) %>%
    mutate(Condition = x) %>%
    dplyr::select(gs_name, Z_mad, Condition)
}, simplify = FALSE)
limit <- temp3 %>%
  bind_rows() %>%
  pull(Z_mad) %>%
  max(abs(.))* c(-1, 1)
hp3 <- temp3 %>%
  bind_rows() %>%
  ggplot(aes(Condition, 
             gs_name, fill = Z_mad)) +
  geom_tile(aes(fill = Z_mad), fontsize =3) +
  geom_text(aes(label = round(Z_mad,2))) +
  scale_fill_distiller(palette = "RdBu", 
                       limit = limit,
                       name = "Normalised Perturbation Score"
                       ) +
  theme_dark()+
  theme(
    axis.title=element_blank(),
    panel.grid = element_blank(),
    # axis.ticks = element_blank(),
    legend.title = element_text(size= 8),
    legend.key.size = unit(3, 'mm'),
    axis.text.x = element_text(angle = 45,hjust=1)
    #   axis.text.y = element_text(colour = col)
  ) +
  ggtitle("Perturbed only under starvation")
# pdf(file = "/Users/wenjunliu/RNA_seq_autophagicflux/Manuscript/Figure/newest/summary_KEGG_spia(3hp).pdf",
#     width = 14,
#     height = 7)
plot_grid(hp1, 
          plot_grid(hp2, hp3, 
                    ncol = 2, 
                    rel_heights = c(2,1),
                    scale = c(1, 0.8),
                    labels = c("b", "c")), 
          nrow = 2, 
          rel_heights = c(2,1), 
          labels = c("a", ""),
          scale = c(0.8,1))
# dev.off()
```

There are many KEGG pathways that were significantly perturbed in multiple conditions so the top 35 signatures (highly significant in many cell lines) were chosen by firstly ranking pathways by their FDR within each condition, taking average of the individual rankings and rank the average rankings to derive the final rankings. 

```{r perturbKEGG_30}
perturbKEGG_rank <- sapply(names(spia_net_sum), function(x){
  spia_net_sum[[x]] %>%
    mutate(Condition = x)
}, simplify = FALSE) %>%
  lapply(dplyr::select, gs_name, Rank_spia, Condition) %>%
  bind_rows() %>%
  group_by(gs_name) %>%
  mutate(aveRank = sum(Rank_spia)/6) %>%
  dplyr::select(gs_name, aveRank) %>%
  unique() %>%
  .[order(.$aveRank),]
perturbKEGG_35 <- perturbKEGG_rank[1:35,] %>%
  pull(gs_name)
#is.nan.data.frame <- function(x){do.call(cbind, lapply(x, is.nan))}
# temp[is.nan(temp)] <- NA
# temp[is.na(temp)] <- 1
# temp %>%
#   group_by(gs_name) %>%
#   # filter(sum(!is.na(p_mad)) >2) %>%
#   mutate(combinedP = wilkinsonp(p = p_mad, r= 2)$p) %>% 
#   ungroup() %>%
#   dplyr::select(gs_name, combinedP) %>%
#   unique() %>%
#   .[order(.$combinedP),] %>%
#   mutate(FDR = p.adjust(.$combinedP, "fdr"))
```

```{r, fig.width=16, fig.height=8}
PerKEGG5 <- spia_net_sum %>%
  lapply(dplyr::filter, FDR_mad < 0.05) %>%
  bind_rows() %>%
  group_by(gs_name) %>%
  filter(n() > 4) %>%
  pull(gs_name) %>%
  unique()
PerKEGG5_top15<- perturbKEGG_rank %>%
  dplyr::filter(gs_name %in% PerKEGG5)%>%
  .[1:15,] %>%
  pull(gs_name)
temp <- sapply(names(spia_net_permut), function(x){
  spia_net_sum[[x]] %>%
    dplyr::filter(gs_name %in% PerKEGG5_top15) %>%
    mutate(Condition = x) %>%
    dplyr::select(gs_name, Z_mad, Condition, Sig)
}, simplify = FALSE) %>%
  bind_rows() %>%
  mutate(
    Z_mad = round(Z_mad, 2),
    Z_label = ifelse(Sig, paste(Z_mad, "*", sep = ""), Z_mad),
    Extreme = ifelse(abs(Z_mad) > 10, TRUE, FALSE), 
    Z_label = ifelse(Extreme, ">10", Z_mad))
PerKEGG5_ro <- temp %>%
  mutate(
    nameLength = vapply(.$gs_name, 
                        function(x){str_split(x, " ")[[1]] %>%
                            length},
                        integer(1)),
    gs_name = ifelse(nameLength > 3,
                     vapply(.$gs_name, 
                            function(x){str_replace_nth(x, " ", "\n", 3)},
                            character(1)), 
                     gs_name) ) %>%
  dplyr::select(gs_name, Z_mad, Condition) %>%
  pivot_wider(names_from = Condition, values_from = Z_mad) %>%
  replace(is.na(.),0) %>%
  column_to_rownames("gs_name") %>%
  # dist() %>%
  # hclust() 
  mutate(overall = rowSums(.)) %>%
  .[order(.$overall, decreasing = FALSE),] %>%
  rownames(.)
keggPer_hp <- temp %>%
  mutate(
    nameLength = vapply(.$gs_name, 
                        function(x){str_split(x, " ")[[1]] %>%
                            length},
                        integer(1)),
    gs_name = ifelse(nameLength > 3,
                     vapply(.$gs_name, 
                            function(x){str_replace_nth(x, " ", "\n", 3)},
                            character(1)), 
                     gs_name) ) %>%
  ggplot(aes(Condition,factor(gs_name, levels = PerKEGG5_ro))) +
  geom_tile(data = . %>% dplyr::filter(!Extreme),
            aes(fill = Z_mad)) +
  # scale_fill_continuous_diverging(palette = "Blue-Red 3",
  #                                  name = "Normalised Perturbation Score",
  #                                  limits = c(-10,10)) +
  scale_fill_distiller(palette = "RdYlBu",
                       limit = c(-10,10),
                       name = "Normalised\nPerturbation\nScore"
                       ) +
  geom_tile(data = . %>% dplyr::filter(Extreme),
            fill = "tomato3") +
  geom_text(data = . %>% dplyr::filter(!Sig),
            aes(label = Z_label), size = 4, color = "azure4") +
  geom_text(data = . %>% dplyr::filter(Sig),
            aes(label = Z_label), size = 4, color = "black") +
  theme(
    axis.title=element_blank(),
    panel.grid = element_blank(),
    axis.text.y =element_text(size=14),
    axis.text.x.bottom  =element_text(size=12),
    axis.ticks = element_blank(),
    legend.title = element_text(size= 12),
    legend.key.size = unit(1, 'cm'),
    axis.text.x = element_text(angle = 45, hjust=1),
    plot.margin = unit(c(0, 0, 0, 0), "null"),
    panel.margin = unit(c(0, 0, 0, 0), "null"),
    #   axis.text.y = element_text(colour = col)
  ) +
  # ggtitle("Perturbation") +
  coord_fixed(ratio= 0.5) +
  panel_border(remove = TRUE)

pdf(file = "/Users/wenjunliu/RNA_seq_autophagicflux/Manuscript/Figure/newest/summary_KEGG_allButOne.pdf",
    width = 12,
    height = 8)
keggPer_hp
dev.off()
```

```{r, fig.width=18, fig.height=10}
plot <- plot_grid(keggOR_hp, keggPer_hp, 
          ncol  = 2, 
          # rel_widths = c(0.9, 1),
          labels =  c("A", "B"))
plot
ggsave(filename = "/Users/wenjunliu/RNA_seq_autophagicflux/Manuscript/Figure/newest/figure3_pathways.svg",
       plot = plot,
       width = 18,
       height = 10)
```

##  TF-KEGG network

Knowing what TFs and KEGG pathways were enriched among the data-set. We then hypothesized that the enriched TFs were the driver of gene signatures detected while the significantly regulated KEGG were the reflection of gene signature on pathway level. 

We wanted to create a regulatory network between the TFs and KEGG, hence combining the two pieces of information.

To achieve that, KEGG pathways perturbed due to the deferentially expressed target genes of enriched TFs were tested through modified SPIA. 

### consistent TFs

30 TFs were consistently enriched under all conditions with an FDR < 0.05. DEGs that were also target genes of those TFs were extracted and used as input of SPIA. Observed PCs were derived locally while the permutation was performed on phoenix

```{r consTF_DE}
names(DE_fc) <- c("HEK293_AZD8055","HEK293_Starvation","SH-SY5Y_AZD8055", "SH-SY5Y_Starvation","HeLa_AZD8055","HeLa_Starvation")
consTF_DE <- sapply(goseq_all, function(y){
  sapply(names(DE_fc), function(x){
    DE_fc[[x]] %>%
      dplyr::filter(gene_name %in% tfByID[[y]]) %>%
      left_join(grGenes %>%
                  as.data.frame() %>%
                  distinct(gene_id, .keep_all = TRUE) %>%
                  dplyr::select(gene_id, gene_name))
    
  }, simplify = FALSE)
}, simplify = FALSE)
# save(consTF_DE, file = here::here("output/Simplified/consTF_DE.Rda"))
```

```{r  spia_consTF}
# load("/Users/wenjunliu/RNA_seq_autophagicflux/output/Simplified/BminsI.Rda")
# consTF_expressed <- sapply(names(consTF_DE), function(y){
#   sapply(names(DE_fc), function(x){
#     DE_fc[[x]] %>%
#       dplyr::filter(gene_name %in% consTF_DE[[y]][[x]]$gene_name) %>%
#       left_join(entrezGenes) %>%
#       drop_na() %>%
#       mutate(entrez_gene = paste("ENTREZID:", entrez_gene, sep = "")) %>%
#       dplyr::select(entrez_gene, aveFC, gene_name) %>%
#       .[!duplicated(.$entrez_gene),]
#   }, simplify = FALSE)
# }, simplify = FALSE)
# consTF_expressed <- sapply(names(consTF_expressed), function(y){
#   sapply(names(consTF_expressed[[y]]), function(x){
#     consTF_expressed[[y]][[x]] %>%
#       extract2("aveFC") %>%
#       set_names(consTF_expressed[[y]][[x]]$entrez_gene)
#   }, simplify = FALSE)
# }, simplify = FALSE)
# spia_consTF <- sapply(names(consTF_expressed), function(y){
#   sapply(names(consTF_expressed[[y]]), function(x){
#     spia_my(de = consTF_expressed[[y]][[x]], BminsI = BminsI)
#   }, simplify = FALSE)
# }, simplify = FALSE)
# save(spia_consTF, file = here::here("output/Simplified/spia_consTF.Rda"))
load(here::here("output/Simplified/spia_consTF.Rda"))
load(here::here("output/Simplified/spia_consTF_permut.Rda"))
```

```{r spia_consTF_sum}
 spia_consTF_sum <- sapply(names(spia_consTF), function(x){
  sapply(names(spia_consTF_permut[[x]]), function(y){
    spia_consTF_permut[[x]][[y]] %>%
      split(f = .$gs_name) %>%
      lapply(dplyr::mutate, 
             MAD = mad(tA), 
             MEDIAN = median(tA)) %>%
      lapply(dplyr::select,gs_name, MAD, 
             MEDIAN) %>%
      lapply(unique) %>%
      bind_rows() %>%
      dplyr::filter(MAD != 0) %>%
      left_join(spia_consTF[[x]][[y]]) %>%
      mutate(
             Z_mad = (tA - MEDIAN)/MAD) %>% 
      mutate(
             p_mad = 2*pnorm(-abs(Z_mad)),
             bonf_mad = p.adjust(p_mad, method = "bonferroni"),
             FDR_mad = p.adjust(p_mad, method = "fdr"),
             Sig = ifelse(FDR_mad < 0.05, TRUE, FALSE)) %>%
      .[order(.$FDR_mad),] %>%
      mutate(Rank_spia = row_number()) 
  # %>%
  #     dplyr::filter(!Z_mad %in% c("-Inf", "Inf"))
  }, simplify = FALSE) 
}, simplify = FALSE)
spia_consTF_sig <- sapply(names(spia_consTF_sum), function(x){
  sapply(names(spia_consTF_sum[[x]]), function(y){
    spia_consTF_sum[[x]][[y]] %>%
      dplyr::filter(Sig) %>%
      mutate(TF = str_split(x, "_")[[1]][1], 
             condition = y) %>%
      dplyr::select(TF, gs_name, condition, tA)
  }, simplify = FALSE) %>%
    bind_rows()
}, simplify = FALSE) %>%
    bind_rows() %>%
  split(f = .$condition)
```


### SY5Y TFs

The same method was applied onto SY5Y-specific TFs, while the only difference was that DE TFTs of the 18 TFs belonged to E2F family and highly overlapped were merged into a "E2F_merge" group. 
```{r sy5yTF_DE}
sy5yTF_DE <- sapply(goseq_SY5Y, function(y){
  sapply(str_subset(names(DE_fc), "SH"), function(x){
    DE_fc[[x]] %>%
      dplyr::filter(gene_name %in% tfByID[[y]]) %>%
      left_join(grGenes %>%
                  as.data.frame() %>%
                  distinct(gene_id, .keep_all = TRUE) %>%
                  dplyr::select(gene_id, gene_name))
    
  }, simplify = FALSE)
}, simplify = FALSE)
notCombine <- c("HSD17B8_TARGET_GENES", "MIER1_TARGET_GENES")
temp_ls <- list()
temp_ls[["E2F_merge"]] <- sapply(setdiff(goseq_SY5Y, notCombine), function(y){
  sapply(names(sy5yTF_DE[[y]]), function(x){
    sy5yTF_DE[[y]][[x]] %>%
      mutate(condition = x)
  }, simplify = FALSE) %>%
    bind_rows()
}, simplify = FALSE) %>%
    bind_rows() %>%
  split(f =.$condition)
sy5yTF_DE <- c(temp_ls, 
               sy5yTF_DE[notCombine])

# save(sy5yTF_DE, file = here::here("output/Simplified/sy5yTF_DE.Rda"))
```

```{r  spia_sy5yTF}
# sy5yTF_expressed <- sapply(names(sy5yTF_DE), function(y){
#   sapply(str_subset(names(DE_fc), "SH"), function(x){
#     DE_fc[[x]] %>%
#       dplyr::filter(gene_name %in% sy5yTF_DE[[y]][[x]]$gene_name) %>%
#       left_join(entrezGenes) %>%
#       drop_na() %>%
#       mutate(entrez_gene = paste("ENTREZID:", entrez_gene, sep = "")) %>%
#       dplyr::select(entrez_gene, aveFC, gene_name) %>%
#       .[!duplicated(.$entrez_gene),]
#   }, simplify = FALSE)
# }, simplify = FALSE)
# sy5yTF_expressed <- sapply(names(sy5yTF_expressed), function(y){
#   sapply(names(sy5yTF_expressed[[y]]), function(x){
#     sy5yTF_expressed[[y]][[x]] %>%
#       extract2("aveFC") %>%
#       set_names(sy5yTF_expressed[[y]][[x]]$entrez_gene)
#   }, simplify = FALSE)
# }, simplify = FALSE)
# spia_sy5yTF <- sapply(names(sy5yTF_expressed), function(y){
#   sapply(names(sy5yTF_expressed[[y]]), function(x){
#     spia_my(de = sy5yTF_expressed[[y]][[x]], BminsI = BminsI)
#   }, simplify = FALSE)
# }, simplify = FALSE)
# save(spia_sy5yTF, file = here::here("output/Simplified/spia_sy5yTF.Rda"))
load(here::here("output/Simplified/spia_sy5yTF.Rda"))
load(here::here("output/Simplified/spia_sy5yTF_permut.Rda"))
```

```{r spia_sy5yTF_sum}
spia_sy5yTF_sum <- sapply(names(spia_sy5yTF), function(x){
  sapply(names(spia_sy5yTF_permut[[x]]), function(y){
    spia_sy5yTF_permut[[x]][[y]] %>%
      split(f = .$gs_name) %>%
      lapply(dplyr::mutate, 
             MAD = mad(tA), 
             MEDIAN = median(tA)) %>%
      lapply(dplyr::select,gs_name, MAD, 
             MEDIAN) %>%
      lapply(unique) %>%
      bind_rows() %>%
      dplyr::filter(MAD != 0) %>%
      left_join(spia_sy5yTF[[x]][[y]]) %>%
      mutate(
        Z_mad = (tA - MEDIAN)/MAD) %>% 
      mutate(
        p_mad = 2*pnorm(-abs(Z_mad)),
        bonf_mad = p.adjust(p_mad, method = "bonferroni"),
        FDR_mad = p.adjust(p_mad, method = "fdr"),
        Sig = ifelse(FDR_mad < 0.05, TRUE, FALSE)) %>%
      .[order(.$FDR_mad),] %>%
      mutate(Rank_spia = row_number()) 
    # %>%
    #     dplyr::filter(!Z_mad %in% c("-Inf", "Inf"))
  }, simplify = FALSE) 
}, simplify = FALSE)
spia_sy5yTF_sig <- sapply(names(spia_sy5yTF_sum), function(x){
  sapply(names(spia_sy5yTF_sum[[x]]), function(y){
    spia_sy5yTF_sum[[x]][[y]] %>%
      dplyr::filter(Sig) %>%
      mutate(TF = str_split(x, "_")[[1]][1], 
             condition = y) %>%
      dplyr::select(TF, gs_name, condition, tA)
  }, simplify = FALSE) %>%
    bind_rows()
}, simplify = FALSE) %>%
    bind_rows() %>%
  split(f = .$condition)
```



### Summary

Significantly perturbed KEGG pathways and TFs regulating them were combined into regulatory pairs and intersections between pairs defined under each condition was visualized. 

```{r spia_sum}
spia_sum <- c(spia_consTF_sum %>%
                lapply(function(x){lapply(x, mutate, category = "Consistent")}), 
              spia_sy5yTF_sum %>%
                lapply(function(x){lapply(x, mutate, category = "SY5Y-specific")}))
TF_KEGG <- sapply(names(spia_sum), function(x){
  sapply(names(spia_sum[[x]]), function(y){
    spia_sum[[x]][[y]] %>%
      # dplyr::filter(gs_name %in%  spia_net_sig[[y]]) %>%
      mutate(TF = str_split(x, "_")[[1]][1], 
             condition = y, 
             Regulation = ifelse(Z_mad < 0, "Inhibition", "Activation")) %>%
      dplyr::select(TF, gs_name, condition, Z_mad, Sig,p_mad, FDR_mad)
  }, simplify = FALSE) %>%
    bind_rows()
}, simplify = FALSE) %>%
    bind_rows() %>%
  split(f = .$condition)
TF_KEGG2 <- sapply(names(spia_sum), function(x){
  sapply(names(spia_sum[[x]]), function(y){
    spia_sum[[x]][[y]] %>%
      dplyr::filter(gs_name %in%  spia_net_sig[[y]]) %>%
      mutate(TF = str_split(x, "_")[[1]][1], 
             condition = y, 
             Regulation = ifelse(Z_mad < 0, "Inhibition", "Activation")) %>%
      dplyr::select(TF, gs_name, condition, Z_mad, Sig,p_mad, FDR_mad)
  }, simplify = FALSE) %>%
    bind_rows()
}, simplify = FALSE) %>%
    bind_rows() %>%
  split(f = .$condition)
# pdf(file = "/Users/wenjunliu/RNA_seq_autophagicflux/Manuscript/Figure/newest/TF-KEGG_upset.pdf",
#     onefile = FALSE,
#     height = 5,
#     width = 8)
TF_KEGG2%>%
  lapply(dplyr::filter, Sig) %>%
  lapply(mutate, pair = paste(TF, gs_name, sep = "-")) %>%
  lapply(pull, pair) %>%
  fromList() %>%
  upset(sets = colnames(.), 
        nintersects = NA, 
        keep.order = TRUE)
TF_KEGG %>%
  lapply(dplyr::filter, Sig) %>%
  lapply(mutate, pair = paste(TF, gs_name, sep = "-")) %>%
  lapply(pull, pair) %>%
  fromList() %>%
  upset(sets = colnames(.), 
        nintersects = NA, 
        keep.order = TRUE, 
         queries = list(
        list(query = intersects,
             params = list("HEK293_AZD8055","HEK293_Starvation", "SH-SY5Y_AZD8055", "SH-SY5Y_Starvation", "HeLa_AZD8055","HeLa_Starvation"),
             color = "red",
             active = T),
        list(query = intersects,
             params = list("SH-SY5Y_AZD8055", "SH-SY5Y_Starvation"),
             color = "blue",
             active = T)
        )
        )
# dev.off()
```

The regulatory relationship inference results were  were written into an excel spreadsheet: "TF_KEGG_pair.xlsx"
```{r}
# wb <- createWorkbook()
# lapply(seq_along(TF_KEGG), function(i){
#   addWorksheet(wb=wb, sheetName = names(TF_KEGG)[[i]])
#   writeData(wb, sheet = i, TF_KEGG[[i]])
# })
# saveWorkbook(wb, here::here("output/Simplified/TF_KEGG_pair.xlsx"), overwrite = TRUE)
```

10 pairs of regulatory relationships were preserved in all conditions. 
```{r tfKEGG4}
tfKEG4 <- TF_KEGG2 %>%
  lapply(dplyr::filter, Sig) %>%
  lapply(mutate, pair = paste(TF, gs_name, sep = "-")) %>%
  bind_rows() %>%
  group_by(pair) %>%
  filter(n() > 4) %>%
  pull(pair) %>%
  unique()
tfKEGG10 <- TF_KEGG %>%
  lapply(dplyr::filter, Sig) %>%
  lapply(mutate, pair = paste(TF, gs_name, sep = "-")) %>%
  lapply(pull, pair) %>%
  Reduce(intersect,.)
```


```{r tfKEGG10_hp, fig.height= 6, fig.width= 12}
temp <- TF_KEGG %>%
  lapply(mutate, pair = paste(TF, gs_name, sep = "-")) %>%
  lapply(dplyr::filter, pair %in% tfKEGG10) %>%
  bind_rows() %>%
  mutate(
    Z_mad = round(Z_mad, 2),
    Z_label = ifelse(Sig, paste(Z_mad, "*", sep = ""), Z_mad),
    neg_extreme = ifelse(Z_mad < -10, TRUE, FALSE), 
    pos_extreme = ifelse(Z_mad > 10, TRUE, FALSE), 
    Z_label = ifelse(pos_extreme, ">10", Z_label),
    )
# pdf(file = "/Users/wenjunliu/RNA_seq_autophagicflux/Manuscript/Figure/newest/TF-KEGG_allButOne.pdf",
#     width = 15,
#     height = 3)
plot <- temp %>%
  ggplot(aes(condition, pair, fill = Z_mad)) +
  geom_tile(data = .%>%
              dplyr::filter(!neg_extreme, !pos_extreme),
            aes(fill = Z_mad), fontsize =3) +
  scale_fill_continuous_diverging(palette = "Blue-Red 3",
                                   name = "Normalised Perturbation Score",
                                   limits = c(-10,10)) +
  geom_tile(data = .%>%
              dplyr::filter(pos_extreme),
            fill = "firebrick", fontsize =3) +
  geom_text(aes(label = Z_label)) +
  theme(
    axis.title=element_blank(),
    panel.grid = element_blank(),
    # axis.ticks = element_blank(),
    legend.title = element_text(size= 8),
    legend.key.size = unit(3, 'mm')
    #   axis.text.y = element_text(colour = col)
  ) +
  ggtitle("TF-KEGG regulatory pairs conserverd across all conditions")
# ggsave(filename = "/Users/wenjunliu/RNA_seq_autophagicflux/Manuscript/Figure/newest/TF-KEGG_consist.svg",
#        plot = plot,
#        width = 12,
#        height =6)

# dev.off()
```



```{r fig.height= 10 , fig.width= 12}
temp <- TF_KEGG %>%
  lapply(mutate, pair = paste(TF, gs_name, sep = "-")) %>%
  lapply(dplyr::filter, pair %in% tfKEGG10) %>%
  bind_rows() %>%
  dplyr::select(TF, gs_name, Z_mad) %>%
  group_by(TF, gs_name) %>%
  summarise(Mean_mad = mean(Z_mad)) %>%
  mutate(color = ifelse(Mean_mad > 0 ,"firebrick", "midnightblue"),
         type = TF, 
         gs_name = str_replace_all(gs_name, " ", "\n") %>%
           str_to_title(), 
         gs_name = ifelse(gs_name == "Pi3k-Akt\nSignaling\nPathway", "PI3K-AKT\nSignaling\nPathway", gs_name)) %>%
  dplyr::rename(from = TF, 
                to = gs_name) 
g <- temp %>%
  .[,c("from", "to", "color", "Mean_mad")] %>%
  graph.data.frame()
TFtoGS <- temp %>%
  #dplyr::select(gs_name, gene_name) %>%
    split(f = .$from)
n <- length(TFtoGS)
totV <- length(V(g))
V(g)$type <- "kegg"
V(g)$type[1:n] <- "TF"
plot <- g %>%
  ggraph(layout = "lgl") +
  geom_edge_arc(alpha = 0.9,
                color = "tomato3", 
                show.legend = FALSE,
                strength = 0.4, 
                linemitre = 5, 
                width = 2.3,
                arrow = arrow(type = "closed", length = unit(3, 'mm'))
                ) +
  geom_node_point(
    fill = "lightyellow1",
    data = . %>% dplyr::filter(type == "TF"),
    size = 5,
    shape = 21,
    repel = TRUE,
    stroke = 0.5,
    show.legend = FALSE
  ) + 
  geom_node_label(
    aes_(label = ~name
         ),
    fill = "lightyellow1",
    data = . %>% dplyr::filter(type == "TF"),
    repel = TRUE,
    size = 7,
    force = 0.2,
    label.padding = unit(0.8, "lines"), 
    label.size = 0,
    label.r = 0.3,
    show.legend = FALSE
  ) +
  geom_node_point(
    fill = "goldenrod1",
    data = . %>% dplyr::filter(type == "kegg"),
    shape = 21,
    size = 2,
    stroke = 0.5,
    repel = TRUE,
    show.legend = FALSE,
    alpha = 0.7, 
  ) +
  # geom_node_text(
  #   aes(label = name),
  #   size = 12,
  #   data = . %>% dplyr::filter(type == "kegg"),
  #   repel = TRUE
  # ) +
  theme(panel.border = element_blank(), 
         panel.background = element_rect(fill = "transparent"), # bg of the panel
  plot.background = element_rect(fill = "transparent", color = NA)
  )

pdf("/Users/wenjunliu/RNA_seq_autophagicflux/Manuscript/Figure/newest/tfKEGG_nw.pdf",
    height = 6,
    width = 8, 
    )
plot
dev.off()
```

```{r}
TF_KEGG$HEK293_AZD8055 %>%
  mutate(pair = paste(TF, gs_name, sep = "-")) %>%
  dplyr::filter(pair %in% tfKEGG10) %>%
  left_join(kg) %>%
  dplyr::filter(gene_name %in% DE_genes$Hek_AZD$gene_name) %>%
  split(f = .$gs_name) %>%
  lapply(pull, gene_name) %>%
  fromList() %>%
  upset(sets = colnames(.), 
        nintersects = NA)
```

From the upset plot, it's found that 32 regulatory relationship were exclusive to SY5Y under both treatments. Those were extracted. 

```{r tf-keggSH, fig.width=12, fig.height=8}
pairToExtract <- TF_KEGG[str_subset(names(TF_KEGG), "SH")] %>%
  lapply(dplyr::filter, Sig) %>%
  lapply(dplyr::select, TF, gs_name) %>%
  Reduce(inner_join,.) %>%
  mutate(pair = paste(TF, gs_name, sep = "-")) %>%
  pull(pair) %>%
  setdiff(.,TF_KEGG[str_subset(names(TF_KEGG), "SH", negate = TRUE)] %>%
            lapply(dplyr::filter, Sig) %>%
            lapply(dplyr::select, TF, gs_name) %>%
            Reduce(full_join,.) %>%
            mutate(pair = paste(TF, gs_name, sep = "-")) %>%
            pull(pair)
  )  
temp2 <- TF_KEGG[str_subset(names(TF_KEGG), "SH")] %>%
  lapply(mutate, pair = paste(TF, gs_name, sep = "-")) %>%
  lapply(dplyr::filter, pair %in% pairToExtract) %>%
  bind_rows() %>%
  mutate(
    Z_mad = round(Z_mad, 2),
    Z_label = ifelse(Sig, paste(Z_mad, "*", sep = ""), Z_mad),
    neg_extreme = ifelse(Z_mad < -16, TRUE, FALSE), 
    Z_label = ifelse(neg_extreme, "<16", Z_label),
    )
# pdf(file = "/Users/wenjunliu/RNA_seq_autophagicflux/Manuscript/Figure/newest/TF-KEGG_SHonly.pdf",
#     width = 10,
#     height = 8)
plot <- temp2 %>%
  bind_rows() %>%
  ggplot(aes(condition, pair, fill = Z_mad)) +
  geom_tile(data = . %>% dplyr::filter(!neg_extreme),
            aes(fill = Z_mad), fontsize =3) +
  scale_fill_continuous_diverging(palette = "Blue-Red 3",
                                   name = "Normalised Perturbation Score",
                                   limits = c(-16,16)) +
  geom_tile(data = .%>%
              dplyr::filter(neg_extreme),
            fill = "royalblue3", fontsize =3) +
  geom_text(aes(label = Z_label)) +
  theme(
    axis.title=element_blank(),
    panel.grid = element_blank(),
    # axis.ticks = element_blank(),
    legend.title = element_text(size= 8),
    legend.key.size = unit(3, 'mm'),
    #   axis.text.y = element_text(colour = col)
  )  +
  ggtitle("TF-KEGG regulatory pairs unique to SY5Y cell line")
# ggsave(filename = "/Users/wenjunliu/RNA_seq_autophagicflux/Manuscript/Figure/newest/TF-KEGG_sy5y.svg",
#        plot = plot,
#        width = 10,
#        height = 10)
# dev.off()
```


```{r tf-keggSH_np, fig.height=8, fig.width=16}
# pairToExtract <- temp[str_subset(names(temp), "SH")] %>%
#   lapply(dplyr::select, -c("condition", "Z_mad")) %>%
#   Reduce(inner_join,.) %>%
#   mutate(pair = paste(TF, gs_name, sep = "-")) %>%
#   pull(pair) %>%
#   setdiff(.,temp[str_subset(names(temp), "SH", negate = TRUE)] %>%
#   lapply(dplyr::select, -c("condition", "Z_mad")) %>%
#   Reduce(full_join,.) %>%
#   mutate(pair = paste(TF, gs_name, sep = "-")) %>%
#   pull(pair)
#   )  
# temp2 <- temp[c("SH_AZD", "SH_EBSS")] %>%
#   lapply(mutate, pair = paste(TF, gs_name, sep = "-")) %>%
#   lapply(dplyr::filter, pair %in% pairToExtract) %>%
#   bind_rows() 
# temp2$Z_mad[is.infinite(temp2$Z_mad) & temp2$Z_mad < 0] <- min(temp2$Z_mad[!is.infinite(temp2$Z_mad)])
# temp2$Z_mad[is.infinite(temp2$Z_mad) & temp2$Z_mad > 0] <- max(temp2$Z_mad[!is.infinite(temp2$Z_mad)])
# temp2 <- temp2 %>%
#   split(f = .$condition) %>%
#   lapply(mutate, color = ifelse(Z_mad < 0, "midnightblue", "firebrick") ,
#          type = TF) %>%
#   lapply(dplyr::rename, from = TF, 
#                 to = gs_name)
# 
# np <- sapply(names(temp2), function(x){
#   g <- temp2[[x]] %>%
#     .[,c("from", "to", "color", "Z_mad")] %>%
#     graph.data.frame()
#   TFtoKEGG <- temp2[[x]] %>%
#     split(f = .$from)
#   n <- length(TFtoKEGG)
#   V(g)$type <- "kegg"
#   V(g)$type[1:n] <- "TF"
#   g_net <- spia_net_sum[[x]] %>%
#     mutate(color = ifelse(Z_mad < 0 ,  "midnightblue", "firebrick")) %>%
#     dplyr::rename(net_e = Z_mad, 
#                   to = gs_name) %>%
#     dplyr::filter(to %in% V(g)$name) %>%
#     unique() %>%
#     .[match(V(g)$name, .$to),]
#   V(g)$color <- g_net$color
#   # V(g)$Category <- kegg_shebss[,c("from", "Category")] %>%
#   #   unique() %>%
#   #   .[match(V(g)$name, .$from),] %>%
#   #   pull(Category)
#   name <- V(g)$name
#   V(g)$name <- paste0(V(g)$name, "(",as.character(round(g_net$net_e, 2)),")")
#   V(g)$name[1:n] <- name[1:n]
#   # pdf(file = "/Users/wenjunliu/RNA_seq_autophagicflux/Manuscript/Figure/new_draft/SHEBSS_ef.pdf",
#   #     width = 16,
#   #     height = 13)
#   g %>%
#     ggraph(layout = "fr") +
#     geom_edge_arc(
#       aes_(color = ~as.character(color), 
#            alpha = ~(abs(Z_mad))),
#       show.legend = FALSE,
#       strength = 0.25, 
#       width = 4) +
#     # geom_edge_link(alpha = 0.8,
#     #               aes_(color = ~as.character(color)),
#     #               show.legend = FALSE,
#     #               strength = 0.5, 
#     #               length = unit(2, 'mm')) +
#     geom_node_point(
#       aes_(fill = ~name),
#       data = . %>% dplyr::filter(type == "TF"),
#       size = 5,
#       repel = TRUE,
#       shape = 21,
#       stroke = 0.5,
#       show.legend = FALSE
#     ) + 
#     geom_node_label(
#       aes_(label = ~name, fill = ~name
#       ),
#       data = . %>% dplyr::filter(type == "TF"),
#       repel = TRUE,
#       size = 5,
#       force = 0.2,
#       label.padding = unit(0.3, "lines"), 
#       show.legend = FALSE
#     ) +
#     geom_node_point(
#       aes_(color = ~I(color)),
#       # color = "black",
#       data = . %>% dplyr::filter(type == "kegg"),
#       shape =18,
#       size = 3,
#       stroke = 0.5,
#       show.legend = FALSE,
#       alpha = 0.7
#     ) +
#     geom_node_text(
#       aes(label = name),
#       size = 4,
#       data = . %>% dplyr::filter(type == "kegg"),
#       repel = TRUE,
#       colour = "black"
#     )  +
#     theme(panel.border = element_blank(), 
#           panel.background = element_blank(), 
#           legend.position="top", 
#           legend.background = element_rect(fill="lightgrey", 
#                                            size=0.5, linetype="solid")
#     )
# }, simplify = FALSE)
# plot_grid(plotlist = np)

```



2 regulatory relationships appeared in all cell lines but were exlusive to starvation treatment. Those were extracted. -Inf anf Inf were converted to minimum and maximum non-infinity values respectively. 

```{r tf-keggEBSS, fig.width=12, fig.height=4}
# pairToExtract <- temp[str_subset(names(temp), "EBSS")] %>%
#   lapply(dplyr::select, -c("condition", "Z_mad")) %>%
#   Reduce(inner_join,.) %>%
#   mutate(pair = paste(TF, gs_name, sep = "-")) %>%
#   pull(pair) %>%
#   setdiff(.,temp[str_subset(names(temp), "EBSS", negate = TRUE)] %>%
#   lapply(dplyr::select, -c("condition", "Z_mad")) %>%
#   Reduce(full_join,.) %>%
#   mutate(pair = paste(TF, gs_name, sep = "-")) %>%
#   pull(pair)
#   )  
# temp2 <- temp[str_subset(names(temp), "EBSS")] %>%
#   lapply(mutate, pair = paste(TF, gs_name, sep = "-")) %>%
#   lapply(dplyr::filter, pair %in% pairToExtract) %>%
#   bind_rows() 
# temp2$Z_mad[is.infinite(temp2$Z_mad) & temp2$Z_mad < 0] <- min(temp2$Z_mad[!is.infinite(temp2$Z_mad)])
# temp2$Z_mad[is.infinite(temp2$Z_mad) & temp2$Z_mad > 0] <- max(temp2$Z_mad[!is.infinite(temp2$Z_mad)])
# limit <- temp2 %>%
#   pull(Z_mad) %>%
#   max(abs(.))* c(-1, 1)
# temp2 %>%
#   bind_rows() %>%
#   ggplot(aes(TF, gs_name, fill = Z_mad)) +
#   geom_tile(aes(fill = Z_mad))  +
#   geom_text(aes(label = round(Z_mad,2))) +
#   scale_fill_distiller(palette = "RdBu", 
#                        limit = limit,
#                        name = "Perturbatino Score"
#                        ) +
#   theme(
#     axis.title=element_blank(),
#     panel.grid = element_blank(),
#     # axis.ticks = element_blank(),
#     legend.title = element_text(size= 8),
#     legend.key.size = unit(3, 'mm'),
#     #   axis.text.y = element_text(colour = col)
#   ) +
#   facet_wrap(~condition)
# ```
# 
# The 4 consistently perturbed KEGG pathways were regulated by different TFs under different conditions. 
# 
# ```{r tf-kegg4, fig.height=18, fig.width=16}
# tf_kegg4 <- sapply(names(spia_sum), function(x){
#   sapply(names(spia_sum[[x]]), function(y){
#     spia_sum[[x]][[y]] %>%
#       dplyr::filter(Sig,
#                     gs_name %in%  hp1_gs) %>%
#       mutate(TF = str_split(x, "_")[[1]][1], 
#              condition = y) %>%
#       dplyr::select(TF, gs_name, condition, Z_mad, category)
#   }, simplify = FALSE) %>%
#     bind_rows()
# }, simplify = FALSE) %>%
#   bind_rows() 
# tf_kegg4$Z_mad[is.infinite(tf_kegg4$Z_mad) & tf_kegg4$Z_mad < 0] <- min(tf_kegg4$Z_mad[!is.infinite(tf_kegg4$Z_mad)])
# tf_kegg4$Z_mad[is.infinite(tf_kegg4$Z_mad) & tf_kegg4$Z_mad > 0] <- max(tf_kegg4$Z_mad[!is.infinite(tf_kegg4$Z_mad)])
# tf_kegg4 <- tf_kegg4 %>%
#   split(f = .$condition) %>%
#   lapply(mutate, color = ifelse(Z_mad < 0, "midnightblue", "firebrick") ,
#          type = TF) %>%
#   lapply(dplyr::rename, from = TF, 
#                 to = gs_name)
# 
# np2 <- sapply(names(tf_kegg4), function(x){
#   g <- tf_kegg4[[x]] %>%
#     .[,c("from", "to", "color", "Z_mad")] %>%
#     graph.data.frame()
#   TFtoKEGG <- tf_kegg4[[x]] %>%
#     split(f = .$from)
#   n <- length(TFtoKEGG)
#   V(g)$type <- "kegg"
#   V(g)$type[1:n] <- "TF"
#   g_net <- spia_net_sum[[x]] %>%
#     mutate(color = ifelse(Z_mad < 0 ,  "midnightblue", "firebrick")) %>%
#     dplyr::rename(net_e = Z_mad, 
#                   to = gs_name) %>%
#     dplyr::filter(to %in% V(g)$name) %>%
#     unique() %>%
#     .[match(V(g)$name, .$to),]
#   V(g)$color <- g_net$color
#   V(g)$Category <-tf_kegg4[[x]][,c("from", "category")] %>%
#     unique() %>%
#     .[match(V(g)$name, .$from),] %>%
#     pull(category)
#   name <- V(g)$name
#   V(g)$name <- paste0(V(g)$name, "(",as.character(round(g_net$net_e, 2)),")")
#   V(g)$name[1:n] <- name[1:n]
#   # pdf(file = "/Users/wenjunliu/RNA_seq_autophagicflux/Manuscript/Figure/new_draft/SHEBSS_ef.pdf",
#   #     width = 16,
#   #     height = 13)
#   g %>%
#     ggraph(layout = "fr") +
#     geom_edge_arc(
#       aes_(color = ~as.character(color), 
#            alpha = ~(abs(Z_mad))),
#       show.legend = FALSE,
#       strength = 0.25, 
#       width = 4) +
#     # geom_edge_link(alpha = 0.8,
#     #               aes_(color = ~as.character(color)),
#     #               show.legend = FALSE,
#     #               strength = 0.5, 
#     #               length = unit(2, 'mm')) +
#     geom_node_point(
#       aes_(fill = ~Category ),
#       data = . %>% dplyr::filter(type == "TF"),
#       size = 5,
#       repel = TRUE,
#       shape = 21,
#       stroke = 0.5,
#       # show.legend = FALSE
#     ) + 
#     geom_node_label(
#       aes_(label = ~name, fill = ~Category 
#       ),
#       data = . %>% dplyr::filter(type == "TF"),
#       repel = TRUE,
#       size = 5,
#       force = 0.2,
#       label.padding = unit(0.3, "lines"), 
#       show.legend = FALSE
#     ) +
#     geom_node_point(
#       aes_(color = ~I(color)),
#       # color = "black",
#       data = . %>% dplyr::filter(type == "kegg"),
#       shape =18,
#       size = 3,
#       stroke = 0.5,
#       show.legend = FALSE,
#       alpha = 0.7
#     ) +
#     geom_node_text(
#       aes(label = name),
#       size = 4,
#       data = . %>% dplyr::filter(type == "kegg"),
#       repel = TRUE,
#       colour = "black"
#     )  +
#     theme(panel.border = element_blank(), 
#           panel.background = element_blank(), 
#           legend.position="top", 
#           legend.background = element_rect(fill="lightgrey", 
#                                            size=0.5, linetype="solid")
#     )
# }, simplify = FALSE)
# lg <- get_legend(np2$SH_EBSS)
# np2 <- np2 %>%
#   lapply(function(x){
#     x +
#       theme(legend.position = "none")
#   })
# plot_grid(
#   lg,
#   plot_grid(plotlist = np2, 
#           labels = names(np2), 
#           ncol = 2), 
#   nrow = 2, 
#   rel_heights  = c(1,10))
# ```
# 
# ### By Condition
# 
# Under each condition, KEGG pathways that were universally significantly regulated were extracted. TFs that significantly perturbed those selected pathways were also extracted and their relationships were visualized through network plots. 
# 
# In the network plots, names in squares are transcription factors and each diamond shape is a KEGG pathway. An edge between a TF and a KEGG pathway shows significant perturbation between that pair and it's color reflects the direction of perturbation and width reflect the strength. 
# 
# Diamond shapes were colored by the consensus tA of pathways. 
# 
# #### SH_EBSS
# 
# 
# ```{r KEGG_SHEBSS_ep, fig.width=12, fig.height= 10}
# kegg_shebss  <-  spia_consTF_sig$SH_EBSS %>%
#   mutate(Category = "Consistent") %>%
#   rbind(spia_sy5yTF_sig$SH_EBSS %>%
#           mutate(Category = "SY5Y-specific")) %>%
#   rbind(spia_ebssTF_sig$SH_EBSS %>%
#           mutate(Category = "EBSS-specific")) %>%
#   dplyr::select(gs_name, tA, TF, Category) %>%
#   dplyr::filter(gs_name %in% kegg_net_sig$SH_EBSS) %>%
#   mutate(color = ifelse(tA < 0, "midnightblue", "firebrick"), 
#          type = TF)  %>%
#   dplyr::rename(from = TF, 
#                 to = gs_name)
# g <- kegg_shebss %>%
#   .[,c("from", "to", "color", "tA")] %>%
#   graph.data.frame()
# TFtoKEGG <- kegg_shebss %>%
#   #dplyr::select(gs_name, gene_name) %>%
#     split(f = .$from)
# n <- length(TFtoKEGG)
# V(g)$type <- "kegg"
# V(g)$type[1:n] <- "TF"
# g_net <- spia_netPer$SH_EBSS %>%
#   mutate(color = ifelse(tA < 0 ,  "midnightblue", "firebrick")) %>%
#   dplyr::rename(net_e = tA, 
#                 to = gs_name) %>%
#   dplyr::filter(to %in% V(g)$name) %>%
#   unique() %>%
#   .[match(V(g)$name, .$to),]
# V(g)$color <- g_net$color
# V(g)$Category <- kegg_shebss[,c("from", "Category")] %>%
#   unique() %>%
#   .[match(V(g)$name, .$from),] %>%
#   pull(Category)
# name <- V(g)$name
# V(g)$name <- paste0(V(g)$name, "(",as.character(round(g_net$net_e, 2)),")")
# V(g)$name[1:n] <- name[1:n]
# # pdf(file = "/Users/wenjunliu/RNA_seq_autophagicflux/Manuscript/Figure/new_draft/SHEBSS_ef.pdf",
# #     width = 16,
# #     height = 13)
# g %>%
#   ggraph(layout = "kk") +
#   geom_edge_arc(
#                 aes_(color = ~as.character(color), 
#                      alpha = ~(abs(tA))),
#                 show.legend = FALSE,
#                 strength = 0.25, 
#                 width = 4) +
#   # geom_edge_link(alpha = 0.8,
#   #               aes_(color = ~as.character(color)),
#   #               show.legend = FALSE,
#   #               strength = 0.5, 
#   #               length = unit(2, 'mm')) +
#   geom_node_point(
#     aes_(fill = ~Category),
#     data = . %>% dplyr::filter(type == "TF"),
#     size = 7,
#     repel = TRUE,
#     shape = 21,
#     stroke = 0.5,
#     # show.legend = FALSE
#   ) + 
#   geom_node_label(
#     aes_(label = ~name, fill = ~Category
#          ),
#     data = . %>% dplyr::filter(type == "TF"),
#     repel = TRUE,
#     size = 7,
#     force = 0.2,
#     label.padding = unit(0.3, "lines"), 
#     show.legend = FALSE
#   ) +
#   geom_node_point(
#     aes_(color = ~I(color)),
#     # color = "black",
#     data = . %>% dplyr::filter(type == "kegg"),
#     shape =18,
#     size = 6,
#     stroke = 0.5,
#     show.legend = FALSE,
#     alpha = 0.7
#   ) +
#   geom_node_text(
#     aes(label = name),
#     size = 6,
#     data = . %>% dplyr::filter(type == "kegg"),
#     repel = TRUE,
#     colour = "black"
#   )  +
#   theme(panel.border = element_blank(), 
#         panel.background = element_blank(), 
#         legend.position="top", 
#         legend.background = element_rect(fill="lightgrey", 
#                                   size=0.5, linetype="solid")
#   )
# dev.off()
```
