---
title: "Differential Expression Analysis"
author: "Wenjun Liu"
date: "12/11/2020"
output: html_document
---


```{r setup, include=FALSE}
knitr::opts_chunk$set(
    autodep = TRUE,
	echo = TRUE,
	warning = FALSE,
	message = FALSE
)
```

#Set libaries
```{r loadPackages}
library(ngsReports)
library(tidyverse)
library(magrittr)
library(edgeR)
library(AnnotationHub)
library(ensembldb)
library(scales)
library(pander)
library(cowplot)
library(corrplot)
library(ggrepel)
library(RColorBrewer)
library(pheatmap)
library(UpSetR)
library(msigdbr)
library(GO.db)
library(goseq)
library(kableExtra)
library(plotly)
library(reshape2)
library(org.Hs.eg.db)
library(cqn)
library(Numero)
```

```{r setOpts}
theme_set(theme_bw())
panderOptions("table.split.table", Inf)
panderOptions("table.style", "rmarkdown")
panderOptions("big.mark", ",")
```

```{r annotationSetup}
ah <- AnnotationHub() %>%
	subset(species == "Homo sapiens") %>%
	subset(rdataclass == "EnsDb")
ensDB <- ah[["AH75011"]]
genes(ensDB)
grTrans <- transcripts(ensDB)
trLengths <- exonsBy(ensDB, "tx") %>%
	width() %>%
	vapply(sum, integer(1))
mcols(grTrans)$length <- trLengths[names(grTrans)]
```

```{r geneAnnotation}
gcGene <- grTrans %>%
  mcols() %>%
  as.data.frame() %>%
  dplyr::select(gene_id, tx_id, gc_content, length) %>%
  as_tibble() %>%
  group_by(gene_id) %>%
  summarise(
    gc_content = sum(gc_content*length) / sum(length),
    length = ceiling(median(length))
  )
grGenes <- genes(ensDB)
mcols(grGenes) %<>%
  as.data.frame() %>%
  left_join(gcGene) %>%
  as.data.frame() %>%
  DataFrame()
```

Metadata of each fastq file and each sample was loaded. 

```{r metaData}
metadata <- here::here("data", "metadata.txt") %>%
  read.table(header = TRUE, sep = "\t",stringsAsFactors = FALSE) %>% 
  as_tibble()%>%
  mutate_at(vars(one_of(c("CELL", "TREAT","CONTR", "TIME"))), as.factor)
filename <- here::here("data", "filename.txt") %>%
  read.table(header = TRUE, sep = "\t",stringsAsFactors = FALSE) %>% 
  as_tibble()%>%
  mutate_at(vars(one_of(c("CELL", "TREAT","CONTR", "TIME"))), as.factor) %>%
  mutate(TREATED = ifelse(CONTR == 0,1,0))
labels <- structure(
  filename$sample,
  names = filename$Filename
)
#extract metadata for each cell line
hela_metadata <- metadata %>%
  dplyr::filter(CELL == "HeLa.tfLC3")
SH_metadata <- metadata %>%
  dplyr::filter(CELL == "SH-SY5Y.tfLC3")  
hek_metadata <- metadata %>%
  dplyr::filter(CELL == "HEK293.tfLC3")  
```

Raw read counts were read in and a separate `DGElist` object was created for each cell line. As observed in the [PCA plot of all control samples ](HeLa_QC.html## Cell line labelling checking), the cell-line effect was dominating the variability in this dataset. Therefore, all the downstream analyses were performed individually within each cell line, before results were integrated. 

```{r dgeList}
minCPM <- 1.5
minSamples <- 3
dgeList <- here::here("data", "2_alignedData", "featureCounts", "genes.out") %>%
  read_delim(delim = "\t") %>%
  set_names(basename(names(.))) %>%
  as.data.frame() %>%
  column_to_rownames("Geneid") %>%
  as.matrix() %>% 
  set_colnames(str_remove(colnames(.), "Aligned.sortedByCoord.out.bam")) %>%
  DGEList(
    samples = tibble(sample = colnames(.)) %>%
      left_join(dplyr::filter(filename, read == "R1")),
    genes = grGenes[rownames(.)] %>%
      as.data.frame() %>%
      dplyr::select(
        chromosome = seqnames, start, end, 
        gene_id, gene_name, gene_biotype, description, 
        entrezid, gc_content, length
      )
  ) %>%
  .[!grepl("rRNA", .$genes$gene_biotype),] %>%
  calcNormFactors()
```


## HeLa 

A `DGEList` was created for HeLa samples and conditional quantile normalisation (cqn) was applied to create an offset for noises introduced by systematic artefacts: GC content and gene length.  

```{r hela_dgeList}
hela_cell <- grepl(paste(unique(hela_metadata$sample), collapse="|"),dgeList$samples$sample)  
hela_dgeList <- dgeList[,hela_cell]
hela_dgeList %<>% .[rowSums(cpm(.) >= minCPM) >= minSamples,]
hela_thrownGene <- setdiff(dgeList$genes$gene_id, hela_dgeList$genes$gene_id)
```

```{r hela_gcCqn}
hela_gcCqn <- cqn(
  counts = hela_dgeList$counts,
  x = hela_dgeList$genes$gc_content,
  lengths = hela_dgeList$genes$length,
  sizeFactors = hela_dgeList$samples$lib.size
)
# Offset was added to original dgelist
hela_dgeList$offset <- hela_gcCqn$glm.offset 
```

SINCE samples treated between by two different autophagy-inducing treatments fell into two distinct cluster in PCA plots across all cell lines, analyses were also performed separately for each treatment.

### AZD8055

Hence there are two factors that are of interests: treatment and time, those two factors were combined into one to create model matrix (visualised as a heatmap shown below).

```{r d_hela_AZD}
hela_dgeList$samples$timeGroups <- paste(hela_dgeList$samples$TREAT, hela_dgeList$samples$TIME, sep = "_") %>%
  as.factor()
hela_dgeList$samples$timeGroups <- relevel(hela_dgeList$samples$timeGroups, ref = "DMSO_0")
d_hela_AZD <- model.matrix(~ timeGroups, data = hela_dgeList$samples %>%
                             dplyr::filter(TREAT %in% c("DMSO", "AZD8055")) %>%
                             droplevels())
d_hela_AZD <- d_hela_AZD %>%
  set_colnames(c("Intercept","AZD1", "AZD15", "AZD30","DMSO30"))
hela_AZD <- grepl(paste(dplyr::filter(hela_metadata, TREAT %in% c("AZD8055", "DMSO"))$sample, collapse="|"),hela_dgeList$samples$sample)
helaAZD_dgeList <- hela_dgeList[,hela_AZD]
helaAZD_dgeList %<>% estimateDisp(design = d_hela_AZD)
```

```{r dmVisualisation, fig.height=3, fig.width=6, fig.cap = "*Visualisation of the pairwise design matrix for AZD8055 treatment*"}
d_hela_AZD %>%
  set_colnames(c("Intercept","Treat_1hr", "Treat_15hrs", "Treat_30hrs","Control_30hrs")) %>%
  as.data.frame() %>%
  rownames_to_column("sample") %>%
  left_join(metadata %>%
              mutate(group = paste(TREAT, TIME, sep= "_")) %>%
              dplyr::select(group, sample)) %>%
  mutate(rownames = paste(sample, group, sep = ":")) %>%
  dplyr::select(-c("group", "sample")) %>%
  column_to_rownames("rownames") %>%
  pheatmap(cluster_cols = FALSE, 
         cluster_rows = TRUE,
         legend = FALSE, 
         show_rownames = TRUE, 
         fontsize_col = 12, 
         angle_col = "0")
```

```{r MakeContrast_AZD}
contrast_AZD <- makeContrasts(AZD1vsContr = AZD1,
                              AZD15vsContr = AZD15,
                              AZD30vsContr = AZD30,
                              DMSO30vs0 = DMSO30,
                              levels = d_hela_AZD)
```

Models were fit using the negative-binomial approaches of `glmQLFit()`.
Top Tables were then obtained using quasi-likelihood (QL) F-test tests against the default fold-change threshold in `glmTreat()`.

```{r topTables_helaAZD}
fit_helaAZD <- glmQLFit(helaAZD_dgeList)
topTables_helaAZD  <- colnames(contrast_AZD) %>%
    sapply(function(x){
      glmTreat(fit_helaAZD, contrast = contrast_AZD[,x]) %>%
      topTags(n = Inf) %>%
      .[["table"]] %>%
      as_tibble() %>%
      arrange(PValue) %>%
      dplyr::select(
        gene_id, gene_name, logFC, logCPM, PValue, FDR, everything()) %>%
      mutate(
        coef = x,
        bonfP = p.adjust(PValue, "bonf")
      )
  }, simplify = FALSE)
# write_rds(topTables_helaAZD,
#           file= here::here("output/topTables/topTables_helaAZD.rds"),
#           compress = "gz")
```

The numbers of genes with FDR < 0.05 in each group were:

```{r DE_helaAZD}
topTables_helaAZD %>%
  lapply(dplyr::filter, FDR < 0.05) %>%
  vapply(nrow, integer(1)) %>%
  pander()
```

### EBSS

The same steps described above were also applied on starved HeLa samples, using control samples cultured in DMEM media as the baseline control this step. 

```{r d_hela_EBSS}
hela_EBSS <- grepl(paste(dplyr::filter(hela_metadata, TREAT %in% c("EBSS", "DMEM"))$sample, collapse="|"),hela_dgeList$samples$sample)
helaEBSS_dgeList <- hela_dgeList[,hela_EBSS]
d_hela_EBSS <- model.matrix(~ timeGroups, data = hela_dgeList$samples %>%
                             dplyr::filter(TREAT %in% c("DMEM", "EBSS")) %>%
                             droplevels())
d_hela_EBSS <- d_hela_EBSS %>%
  set_colnames(c("Intercept","DMEM30","EBSS1", "EBSS15", "EBSS30"))
helaEBSS_dgeList %<>% estimateDisp(design = d_hela_EBSS)
```

```{r dmVisualisation1, fig.height=3, fig.width=6, fig.cap = "*Visualisation of the pairwise design matrix for EBSS8055 treatment*"}
pheatmap(d_hela_EBSS,
         cluster_cols = FALSE, 
         cluster_rows = FALSE,
         legend = FALSE)
```

```{r MakeContrast_EBSS}
contrast_EBSS <- makeContrasts(EBSS1vsContr = EBSS1,
                               EBSS15vsContr = EBSS15,
                               EBSS30vsContr = EBSS30,
                               DMEM30vs0 = DMEM30,
                               levels = d_hela_EBSS)
```

The numbers of DEGs detected in each comparison are:

```{r topTables_helaEBSS}
fit_helaEBSS <- glmQLFit(helaEBSS_dgeList)
topTables_helaEBSS  <- colnames(contrast_EBSS) %>%
  sapply(function(x){
    glmTreat(fit_helaEBSS, contrast = contrast_EBSS[,x]) %>%
      topTags(n = Inf) %>%
      .[["table"]] %>%
      as_tibble() %>%
      arrange(PValue) %>%
      dplyr::select(
        gene_id, gene_name, logFC, logCPM, PValue, FDR, everything()) %>%
      mutate(
        coef = x,
        bonfP = p.adjust(PValue, "bonf")
      )
  }, simplify = FALSE)
# write_rds(topTables_helaEBSS,
#           file= here::here("output/topTables/topTables_helaEBSS.rds"),
#           compress = "gz")
topTables_helaEBSS %>%
  lapply(dplyr::filter, FDR < 0.05) %>% 
  vapply(nrow, integer(1)) %>%
  pander()
```

## SH-SY5Y

Same pipeline developed using HeLa cell line was also applied on SH-SY5Y cell line. 

```{r SH_dgeList}
SH_cell <- grepl(paste(unique(SH_metadata$sample), collapse="|"),dgeList$samples$sample)  
SH_dgeList <- dgeList[,SH_cell]
SH_dgeList %<>% .[rowSums(cpm(.) >= minCPM) >= minSamples,]
SH_thrownGene <- setdiff(dgeList$genes$gene_id, SH_dgeList$genes$gene_id)
```

```{r SH_gcCqn}
SH_gcCqn <- cqn(
  counts = SH_dgeList$counts,
  x = SH_dgeList$genes$gc_content,
  lengths = SH_dgeList$genes$length,
  sizeFactors = SH_dgeList$samples$lib.size
)
# Offset was added to original dgelist
SH_dgeList$offset <- SH_gcCqn$glm.offset 
```

### AZD8055
```{r d_SH_AZD}
SH_dgeList$samples$timeGroups <- paste(SH_dgeList$samples$TREAT, SH_dgeList$samples$TIME, sep = "_") %>%
  as.factor()
SH_dgeList$samples$timeGroups <- relevel(SH_dgeList$samples$timeGroups, ref = "DMSO_0")
d_SH_AZD <- model.matrix(~ timeGroups, data = SH_dgeList$samples %>%
                             dplyr::filter(TREAT %in% c("DMSO", "AZD8055")) %>%
                             droplevels())
d_SH_AZD <- d_SH_AZD %>%
  set_colnames(c("Intercept","AZD1", "AZD15", "AZD30","DMSO15","DMSO30"))
SH_AZD <- grepl(paste(dplyr::filter(SH_metadata, TREAT %in% c("AZD8055", "DMSO"))$sample, collapse="|"),SH_dgeList$samples$sample)
SHAZD_dgeList <- SH_dgeList[,SH_AZD]
SHAZD_dgeList %<>% estimateDisp(design = d_SH_AZD)
```


```{r MakeContrast_AZD2}
contrast_AZD <- makeContrasts(AZD1vsContr = AZD1,
                              AZD15vsContr = AZD15,
                              AZD30vsContr = AZD30,
                              DMSO15vs0 = DMSO15,
                              DMSO30vs0 = DMSO30,
                              levels = d_SH_AZD)
```

We also collected 15 hrs control samples for SH-SY5Y cell line. The numbers of DEGs detected in each comparison are:

```{r topTables_SHAZD}
fit_SHAZD <- glmQLFit(SHAZD_dgeList)
topTables_SHAZD  <- colnames(contrast_AZD) %>%
  sapply(function(x){
    glmTreat(fit_SHAZD, contrast = contrast_AZD[,x]) %>%
      topTags(n = Inf) %>%
      .[["table"]] %>%
      as_tibble() %>%
      arrange(PValue) %>%
      dplyr::select(
        gene_id, gene_name, logFC, logCPM, PValue, FDR, everything()) %>%
      mutate(
        coef = x,
        bonfP = p.adjust(PValue, "bonf")
      )
  }, simplify = FALSE)
topTables_SHAZD %>%
  lapply(dplyr::filter, FDR < 0.05) %>% 
  vapply(nrow, integer(1)) %>%
  pander()
# write_rds(topTables_SHAZD, 
#           path = here::here("output/topTables/topTables_SHAZD.rds"), 
#           compress = "gz")
```



### EBSS
```{r d_SH_EBSS}
SH_dgeList$samples$timeGroups <- relevel(SH_dgeList$samples$timeGroups, ref = "DMEM_0")
d_SH_EBSS <- model.matrix(~ timeGroups, data = SH_dgeList$samples %>%
                              dplyr::filter(TREAT %in% c("DMEM", "EBSS")) %>%
                              droplevels())
d_SH_EBSS <- d_SH_EBSS %>%
  set_colnames(c("Intercept","DMEM15","DMEM30","EBSS1", "EBSS15", "EBSS30"))
SH_EBSS <- grepl(paste(dplyr::filter(SH_metadata, TREAT %in% c("EBSS", "DMEM"))$sample, collapse="|"),SH_dgeList$samples$sample)
SHEBSS_dgeList <- SH_dgeList[,SH_EBSS]
SHEBSS_dgeList %<>% estimateDisp(design = d_SH_EBSS)
```


```{r MakeContrast_EBSS2}
contrast_EBSS <- makeContrasts(EBSS1vsContr = EBSS1,
                               EBSS15vsContr = EBSS15,
                               EBSS30vsContr = EBSS30,
                               DMEM30vs0 = DMEM30,
                               DMEM15vs0 = DMEM15,
                               levels = d_SH_EBSS)
```

```{r topTables_SHEBSS}
fit_SHEBSS <- glmQLFit(SHEBSS_dgeList)
topTables_SHEBSS  <- colnames(contrast_EBSS) %>%
  sapply(function(x){
    glmTreat(fit_SHEBSS, contrast = contrast_EBSS[,x]) %>%
      topTags(n = Inf) %>%
      .[["table"]] %>%
      as_tibble() %>%
      arrange(PValue) %>%
      dplyr::select(
        gene_id, gene_name, logFC, logCPM, PValue, FDR, everything()) %>%
      mutate(
        coef = x,
        bonfP = p.adjust(PValue, "bonf")
      )
  }, simplify = FALSE)
topTables_SHEBSS %>%
  lapply(dplyr::filter, FDR < 0.05) %>% 
  vapply(nrow, integer(1)) %>%
  pander()
# write_rds(topTables_SHEBSS, 
#           path = here::here("output/topTables/topTables_SHEBSS.rds"), 
#           compress = "gz")
```

## HEK293

Same pipeline developed using HeLa cell line was also applied on HEK293 cell line. However, 

```{r hek_dgeList}
hek_cell <- grepl(paste(unique(hek_metadata$sample), collapse="|"),dgeList$samples$sample)  
hek_dgeList <- dgeList[,hek_cell]
hek_dgeList %<>% .[rowSums(cpm(.) >= minCPM) >= minSamples,]
hek_thrownGene <- setdiff(dgeList$genes$gene_id, hek_dgeList$genes$gene_id)
```

```{r hek_gcCqn}
hek_gcCqn <- cqn(
  counts = hek_dgeList$counts,
  x = hek_dgeList$genes$gc_content,
  lengths = hek_dgeList$genes$length,
  sizeFactors = hek_dgeList$samples$lib.size
)
# Offset was added to original dgelist
hek_dgeList$offset <- hek_gcCqn$glm.offset 
```

### AZD8055
```{r d_hek_AZD}
hek_dgeList$samples$timeGroups <- paste(hek_dgeList$samples$TREAT, hek_dgeList$samples$TIME, sep = "_") %>%
  as.factor()
hek_dgeList$samples$timeGroups <- relevel(hek_dgeList$samples$timeGroups, ref = "DMSO_0")
d_hek_AZD <- model.matrix(~ timeGroups, data = hek_dgeList$samples %>%
                             dplyr::filter(TREAT %in% c("DMSO", "AZD8055")) %>%
                             droplevels())
d_hek_AZD <- d_hek_AZD %>%
  set_colnames(c("Intercept","AZD1", "AZD15", "AZD30","DMSO30"))
hek_AZD <- grepl(paste(dplyr::filter(hek_metadata, TREAT %in% c("AZD8055", "DMSO"))$sample, collapse="|"),hek_dgeList$samples$sample)
hekAZD_dgeList <- hek_dgeList[,hek_AZD]
hekAZD_dgeList %<>% estimateDisp(design = d_hek_AZD)
```

```{r MakeContrast_AZD3}
contrast_AZD <- makeContrasts(AZD1vsContr = AZD1,
                              AZD15vsContr = AZD15,
                              AZD30vsContr = AZD30,
                              DMSO30vs0 = DMSO30,
                              levels = d_hek_AZD)
```

```{r topTables_hekAZD}

fit_hekAZD <- glmQLFit(hekAZD_dgeList)
topTables_hekAZD  <- colnames(contrast_AZD) %>%
  sapply(function(x){
    glmTreat(fit_hekAZD, contrast = contrast_AZD[,x]) %>%
      topTags(n = Inf) %>%
      .[["table"]] %>%
      as_tibble() %>%
      arrange(PValue) %>%
      dplyr::select(
        gene_id, gene_name, logFC, logCPM, PValue, FDR, everything()) %>%
      mutate(
        coef = x,
        bonfP = p.adjust(PValue, "bonf")
      )
  }, simplify = FALSE)
# write_rds(topTables_hekAZD, 
#           path = here::here("output/topTables/topTables_hekAZD.rds"), 
#           compress = "gz")
topTables_hekAZD %>%
  lapply(dplyr::filter, FDR < 0.05) %>%
  vapply(nrow, integer(1)) %>%
  pander()
```

### EBSS

```{r d_hek_EBSS}
hek_dgeList$samples$timeGroups <- relevel(hek_dgeList$samples$timeGroups, ref = "DMEM_0")
d_hek_EBSS <- model.matrix(~ timeGroups, data = hek_dgeList$samples %>%
                              dplyr::filter(TREAT %in% c("DMEM", "EBSS")) %>%
                              droplevels())
d_hek_EBSS <- d_hek_EBSS %>%
  set_colnames(c("Intercept","DMEM30","EBSS1", "EBSS15", "EBSS30"))
hek_EBSS <- grepl(paste(dplyr::filter(hek_metadata, TREAT %in% c("EBSS", "DMEM"))$sample, collapse="|"),hek_dgeList$samples$sample)
hekEBSS_dgeList <- hek_dgeList[,hek_EBSS]
hekEBSS_dgeList %<>% estimateDisp(design = d_hek_EBSS)
```

```{r MakeContrast_EBSS3}
contrast_EBSS <- makeContrasts(EBSS1vsContr = EBSS1,
                               EBSS15vsContr = EBSS15,
                               EBSS30vsContr = EBSS30,
                               DMEM30vs0 = DMEM30,
                               levels = d_hek_EBSS)
```

A large number of DEGs were detected betwee 0 and 30hrs control samples, aligning with the problem on 30hr HEK293 control samples cultured in DMEM that was detected at the [quality control stage](SHSY5Y_QC.html).

```{r topTables_hekEBSS}
fit_hekEBSS <- glmQLFit(hekEBSS_dgeList)
topTables_hekEBSS  <- colnames(contrast_EBSS) %>%
  sapply(function(x){
    glmTreat(fit_hekEBSS, contrast = contrast_EBSS[,x]) %>%
      topTags(n = Inf) %>%
      .[["table"]] %>%
      as_tibble() %>%
      arrange(PValue) %>%
      dplyr::select(
         gene_id, gene_name, logFC, logCPM, PValue, FDR, everything()) %>%
      mutate(
        coef = x,
        bonfP = p.adjust(PValue, "bonf")
        )
  }, simplify = FALSE)
topTables_hekEBSS %>%
  lapply(dplyr::filter, FDR < 0.05) %>% 
  vapply(nrow, integer(1)) %>%
  pander()
# write_rds(topTables_hekEBSS,
#           path = here::here("output/newTop/topTables_hekEBSS.rds"),
#           compress = "gz")
```
